---
title: "data_analysis_bp1"
author: "Leire A. Murua"
date: "2025-03-07"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(DECIPHER); packageVersion("DECIPHER")
library(phangorn); packageVersion("phangorn")
library(phyloseq); packageVersion("phyloseq")
library(Biostrings); packageVersion("Biostrings")
library(ggplot2); packageVersion("ggplot2")
library(tidyverse); packageVersion("tidyverse")
library(vegan); packageVersion("vegan")
library(microbiome); packageVersion("microbiome")
library(dplyr); packageVersion("dplyr")
library(tidyr); packageVersion("tidyr")
library(microViz); packageVersion("microViz")
library(dada2); packageVersion("dada2")
library(ggtree); packageVersion("ggtree")
library(biomeUtils); packageVersion("biomeUtils")
library(kableExtra); packageVersion("kableExtra")
library(patchwork); packageVersion("patchwork")
library(ggsci); packageVersion("ggsci")
library(RColorBrewer); packageVersion("RColorBrewer")
library(rstatix); packageVersion("rstatix")
library(ggpubr); packageVersion("ggpubr")
library(Maaslin2); packageVersion("Maaslin2")
library(multcompView); packageVersion("multcompView")

library(beepr)

```

## BIOPROJECT 1 WORKFLOW
```{r}

ps_1 <- ps %>% ps_filter(bioproject == "1")

meta_1 <- sample_data(ps_1)

sample_names_1 <- rownames(meta_1)
  
sample_names_1 

seqtab.nochim_737472 <- readRDS("../seqtab.nochim_737472.rds")

rownames(seqtab.nochim_737472) <- sub("^\\d+\\.", "", rownames(seqtab.nochim_737472))

seqtab_filtered_1 <- seqtab.nochim_737472[sample_names_1, ]

seqs_1 <- getSequences(seqtab_filtered_1)
names(seqs_1) <- seqs_1

# alignment_1 <- AlignSeqs(DNAStringSet(seqs_1), anchor=NA)
# saveRDS(alignment_1, "alignment_1.rds")
alignment_1 <- readRDS("alignment_1.rds")


# phang_align_1 <- phyDat(as(alignment_1, "matrix"), type="DNA")
# saveRDS(phang_align_1, "phang_align_1.rds")
phang_align_1 <- readRDS("phang_align_1.rds")


## Check kind of distance
# dm_1 <- dist.ml(phang_align_1)

# saveRDS(dm_1, "dm_1.rds")
dm_1 <- readRDS("dm_1.rds")


# treeNJ_1 <- NJ(dm_1)
# saveRDS(treeNJ_1, "treeNJ_1.rds")
treeNJ_1 <- readRDS("treeNJ_1.rds")

```

```{r}

ps_sum_1 <- summarize_phyloseq(ps_1)
summary_ps_table_1 <- data.frame(do.call(rbind, ps_sum_1[c(1:10)]))

write.csv(summary_ps_table_1, "../Results/summary_ps_table_1.csv", row.names = FALSE)

summary_ps_table_1

```

```{r}

head(ps_1@otu_table, n=10) %>% kbl() %>% 
  kableExtra::kable_paper(font_size = 12) %>% 
  scroll_box(width = "800px", height = "300px")

head(ps_1@tax_table, n=10) %>% kbl() %>%
  kableExtra::kable_paper(font_size = 12) %>%
  scroll_box(width = "800px", height = "300px")

metadata_x_1 <- ps_1 %>% meta()
head(metadata_x_1, n=10) %>% kbl() %>%
  kableExtra::kable_paper(font_size = 12) %>%
  scroll_box(width = "800px", height = "300px")

ggtree(ps_1@phy_tree, options(ignore.negative.edge=TRUE))

```


```{r}
Chloroplast_1 <- ntaxa(subset_taxa(ps_1, Order == "Chloroplast"))

Mitochondria_1 <- ntaxa(subset_taxa(ps_1, Family == "Mitochondria"))

chl_mt_df_1 <- data.frame(Taxa = c("Chloroplast", "Mitochondria"),
                       nASVs = c(Chloroplast_1, Mitochondria_1)) %>%
  mutate(fraction = round(nASVs/ntaxa(ps_1), 3))
chl_mt_df_1 

ps_filt_1 <- subset_taxa(ps_1, Order != "Chloroplast")
ps_filt_1 <- subset_taxa(ps_filt_1, Family != "Mitochondria")

ps_filt_1@tax_table %>% 
  data.frame() %>% 
  group_by(Kingdom) %>% 
  dplyr::count() %>%
  mutate(fraction = round(n/ntaxa(ps_filt_1), 3))

ps_filt_1 <- subset_taxa(ps_filt_1, Kingdom == "Bacteria")

phylum_filt_table_1 <- ps_filt_1@tax_table %>% 
  data.frame() %>% 
  group_by(Phylum) %>% 
  dplyr::count() %>% 
  arrange(desc(n))

phylum_filt_table_1

write.csv(phylum_filt_table_1, "../Results/phylum_filt_table_1.csv", row.names = FALSE)

ps_filt_sum_1 <- summarize_phyloseq(ps_filt_1)
summary_ps_filt_table_1 <- data.frame(do.call(rbind, ps_filt_sum_1[c(1:10)]))
write.csv(summary_ps_filt_table_1, "../Results/summary_ps_filt_table_1.csv", row.names = FALSE)

summary_ps_filt_table_1

```

```{r}

ps_filt_fixed_1 <- ps_filt_1 %>%
 tax_fix(
  min_length = 4,
  sep="_", anon_unique=TRUE, 
  suffix_rank="classified")

ps_filt_fixed_1@tax_table

ggtree(ps_filt_fixed_1@phy_tree, options(ignore.negative.edge=TRUE))

  
```


```{r}

sum_of_reads_1 <- as.data.frame(sample_sums(ps_filt_fixed_1@otu_table)) %>% rownames_to_column(var = "samples") %>%
 dplyr::rename(sum_of_reads_1 = `sample_sums(ps_filt_fixed_1@otu_table)`)%>%
  left_join((metadata_x_1 %>% rownames_to_column(var = "samples")), by = "samples")
head((sum_of_reads_1 %>% arrange(sum_of_reads_1)), n=20)%>% kbl() %>%
  kableExtra::kable_paper(font_size = 12) %>%
  scroll_box(width = "800px", height = "300px")

range(sum_of_reads_1$sum_of_reads)

```

```{r}
sum_of_reads_plot_1 <- sum_of_reads_1 %>% arrange(sum_of_reads_1) %>% 
  ggplot(aes(x = reorder(samples, sum_of_reads_1), y = sum_of_reads_1, fill = group)) +
  geom_point(shape = 21, size = 2, color = "black")+
  scale_fill_manual(values = c(
    "active-active" = "darkred",
    "active-remission" = "orange", 
    "healthy-healthy" = "darkgreen"
  ))+
  labs(x = "samples")+
  theme_bw()+
  theme(axis.text.x=element_text(angle=90))

ggsave("../Results/SumOfReads_1_plot.png", sum_of_reads_plot_1, device = png, width = 10, height = 8)

sum_of_reads_plot_1
```


```{r}
ps_filt_fixed_pruned_1 <- prune_taxa(taxa_sums(ps_filt_fixed_1) > 0, ps_filt_fixed_1) # Remove absent taxa


ps_filt_fixed_gen_rel_1 <- ps_filt_fixed_pruned_1 %>% 
  tax_transform("compositional", rank = "Genus")

ps_filt_fixed_gen_rel_1@sam_data$group <- as.factor(ps_filt_fixed_gen_rel_1@sam_data$group)

ps_ff_gen_rel_end_1 <- ps_filt_fixed_gen_rel_1 %>% ps_filter(!is.na(disease_status_end))
top_taxa_end_1 <- tax_top(ps_filt_fixed_gen_rel_1, n=40)

melt_genus_rel_end_1 <- ps_filt_fixed_gen_rel_1 %>% psmelt() %>% 
  mutate(Genus_top = case_when(Genus %in% top_taxa_end_1 ~ Genus, 
                               TRUE ~ "other")) %>% 
  mutate(Genus_top = as.factor(Genus_top)) %>% 
  mutate(Genus_top = fct_relevel(Genus_top, c(top_taxa_end_1, "other"))) %>% 
  filter(!is.na(disease_status_end))

main_genus_end_1 <- melt_genus_rel_end_1 %>%
  ggplot(aes(x = Sample, y = Abundance, fill = Genus_top)) +
  geom_bar(stat = "identity", color = "black")+
  scale_fill_manual(values = distinct_palette(40, pal = "brewerPlus", add = "grey70"))+
  theme_bw()+
  labs(title= "Genus Bar plot for end samples") +
  theme(axis.title.x = element_blank(), legend.position = "right")

ps_ff_gen_rel_base_1 <- ps_filt_fixed_gen_rel_1 %>% ps_filter(!is.na(baseline_status))
top_taxa_base_1 <- tax_top(ps_ff_gen_rel_base_1, n=40)

melt_genus_rel_base_1 <- ps_filt_fixed_gen_rel_1 %>% psmelt() %>% 
  mutate(Genus_top = case_when(Genus %in% top_taxa_base_1~ Genus, 
                               TRUE ~ "other")) %>% 
  mutate(Genus_top = as.factor(Genus_top)) %>% 
  mutate(Genus_top = fct_relevel(Genus_top, c(top_taxa_base_1, "other"))) %>% 
  filter(!is.na(baseline_status))

main_genus_base_1 <- melt_genus_rel_base_1 %>%
  ggplot(aes(x = Sample, y = Abundance, fill = Genus_top)) +
  geom_bar(stat = "identity", color = "black")+
  scale_fill_manual(values = distinct_palette(40, pal = "brewerPlus", add = "grey70"))+
  theme_bw()+
  labs(title= "Genus Bar plot for baseline samples") +
  theme(axis.title.x = element_blank(), legend.position = "right")

ps_ff_pat_gen_rel_end_1 <- ps_filt_fixed_gen_rel_1 %>% ps_filter(!is.na(disease_status_end) & donor == "no")
top_taxa_pat_end_1 <- tax_top(ps_ff_pat_gen_rel_end_1, n=40)

melt_genus_rel_end_pat_1 <- ps_filt_fixed_gen_rel_1 %>% psmelt() %>% 
  mutate(Genus_top = case_when(Genus %in% top_taxa_pat_end_1 ~ Genus, 
                               TRUE ~ "other")) %>% 
  mutate(Genus_top = as.factor(Genus_top)) %>% 
  mutate(Genus_top = fct_relevel(Genus_top, c(top_taxa_pat_end_1, "other"))) %>% 
  filter(!is.na(disease_status_end)) %>% 
  filter(donor == "no")

main_genus_end_pat_1 <-melt_genus_rel_end_pat_1 %>%
  ggplot(aes(x = Sample, y = Abundance, fill = Genus_top)) +
  geom_bar(stat = "identity", color = "black")+
  scale_fill_manual(values = distinct_palette(40, pal = "brewerPlus", add = "grey70"))+
  theme_bw()+
  labs(title= "Genus Bar plot for end Patient samples") +
  theme(axis.title.x = element_blank(), legend.position = "right")

ps_ff_pat_gen_rel_base_1 <- ps_filt_fixed_gen_rel_1 %>% ps_filter(!is.na(baseline_status) & donor == "no")
top_taxa_pat_base_1 <- tax_top(ps_ff_pat_gen_rel_base_1, n=40)
melt_genus_rel_base_pat_1 <- ps_filt_fixed_gen_rel_1 %>% psmelt() %>% 
  mutate(Genus_top = case_when(Genus %in% top_taxa_pat_base_1 ~ Genus, 
                               TRUE ~ "other")) %>% 
  mutate(Genus_top = as.factor(Genus_top)) %>% 
  mutate(Genus_top = fct_relevel(Genus_top, c(top_taxa_pat_base_1, "other"))) %>% 
  filter(!is.na(baseline_status)) %>% 
  filter(donor == "no")

main_genus_base_pat_1 <- melt_genus_rel_base_pat_1 %>%
  ggplot(aes(x = Sample, y = Abundance, fill = Genus_top)) +
  geom_bar(stat = "identity", color = "black")+
  scale_fill_manual(values = distinct_palette(40, pal = "brewerPlus", add = "grey70"))+
  theme_bw()+
  labs(title= "Genus Bar plot for baseline Patient samples") +
  theme(axis.title.x = element_blank(), legend.position = "right")


ggsave("../Results/main_genus_end_1_barplot.png", main_genus_end_1, device = png, width = 10, height = 8)
ggsave("../Results/main_genus_base_1_barplot.png", main_genus_base_1, device = png, width = 10, height = 8)
ggsave("../Results/main_genus_end_pat_1_barplot.png", main_genus_end_pat_1, device = png, width = 10, height = 8)
ggsave("../Results/main_genus_base_pat_1_barplot.png", main_genus_base_pat_1, device = png, width = 10, height = 8)

main_genus_end_1
main_genus_base_1
main_genus_end_pat_1
main_genus_base_pat_1

```

## Group by donor, plot end  
```{r}
ps_ff_don_gen_rel_1 <- ps_filt_fixed_gen_rel_1 %>% ps_filter(donor == "yes")
top_taxa_don_1 <- tax_top(ps_ff_don_gen_rel_1, n=40)

melt_genus_rel_healthy_1 <- ps_filt_fixed_gen_rel_1 %>% psmelt() %>% 
  mutate(Genus_top = case_when(Genus %in% top_taxa_don_1 ~ Genus, 
                               TRUE ~ "other")) %>% 
  mutate(Genus_top = as.factor(Genus_top)) %>% 
  mutate(Genus_top = fct_relevel(Genus_top, c(top_taxa_don_1, "other"))) %>% 
  filter(donor == "yes")

genus_healthy_barplot_donorid <- melt_genus_rel_healthy_1 %>%
  ggplot(aes(x = Sample, y = Abundance, fill = Genus_top)) +
  geom_bar(stat = "identity", color = "black")+
  facet_wrap("host_subject_id", scales = "free_x") + 
  scale_fill_manual(values = distinct_palette(40, pal = "brewerPlus", add = "grey70"))+
  theme_bw()+
  labs(title= "Genus Bar plot for Healthy Donor samples grouped by DonorID") +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank(), legend.position = "right")

melt_genus_rel_end_pat_1 <- ps_filt_fixed_gen_rel_1 %>% psmelt() %>% 
  mutate(Genus_top = case_when(Genus %in% top_taxa_pat_end_1 ~ Genus, 
                               TRUE ~ "other")) %>% 
  mutate(Genus_top = as.factor(Genus_top)) %>% 
  mutate(host_assigned_donor_subject_id = as.factor(host_assigned_donor_subject_id)) %>% 
  mutate(Genus_top = fct_relevel(Genus_top, c(top_taxa_pat_end_1, "other"))) %>% 
  filter(!is.na(disease_status_end)) %>% 
  filter(donor == "no")

genus_pat_barplot_donorpaired <- melt_genus_rel_end_pat_1 %>%
  ggplot(aes(x = Sample, y = Abundance, fill = Genus_top)) +
  geom_bar(stat = "identity", color = "black")+
  facet_wrap("host_assigned_donor_subject_id", scales = "free_x") + 
  scale_fill_manual(values = distinct_palette(40, pal = "brewerPlus", add = "grey70"))+
  scale_x_discrete(guide = guide_axis(n.dodge = 2)) +
  theme_bw()+
  labs(title= "Genus Bar plot for end Patient samples grouped by Donor") +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank(), legend.position = "right")

ggsave("../Results/genus_healthy_barplot_donorid.png", genus_healthy_barplot_donorid, width = 13, height = 8, dpi = 300)
ggsave("../Results/genus_pat_barplot_donorpaired.png", genus_pat_barplot_donorpaired, width = 15, height = 8, dpi = 300)

genus_healthy_barplot_donorid
genus_pat_barplot_donorpaired

```




```{r}
create_donorpair_barplot <- function(ps_obj, donor_id) {
  ps_subset <- ps_obj %>% ps_filter(host_assigned_donor_subject_id == donor_id)

  metadata <- sample_data(ps_subset) %>% as.data.frame()

  metadata$facet_label <- paste(metadata$host_subject_id, "(", metadata$group, ")", sep = "")

  sample_data(ps_subset) <- metadata

  barplot <- ps_subset %>%
    ps_mutate(host_sex = as.factor(host_sex)) %>%
    ps_mutate(host_assigned_donor_subject_id = as.factor(host_assigned_donor_subject_id)) %>%
    ps_arrange(host_assigned_donor_subject_id, group, week) %>%
    comp_barplot(group_by = "host_assigned_donor_subject_id", tax_level = "Genus",
                 tax_transform_for_plot = "compositional", n_taxa = 11,
                 bar_width = 0.8, merge_other = FALSE, label = "week",
                 sample_order = "default")

  wrapped_plot <- wrap_plots(barplot) +
    plot_layout(guides = "collect") &
    facet_wrap(~facet_label, scales = "free_x") &  
    theme(axis.text.x = element_text(angle = 60, hjust = 1, size = 10),
          axis.ticks.y = element_blank(), legend.title = element_blank(),
          legend.position = "bottom", strip.text = element_text(size = 12),
          legend.key.size = unit(0.4, "cm"),
          legend.text = element_text(size = 10))

  return(wrapped_plot)
}

plot_donorpaired_0044 <- create_donorpair_barplot(ps_filt_fixed_gen_rel_1, "D0044")
plot_donorpaired_0097 <- create_donorpair_barplot(ps_filt_fixed_gen_rel_1, "D0097")
plot_donorpaired_0485 <- create_donorpair_barplot(ps_filt_fixed_gen_rel_1, "D0485")

ggsave("../Results/plot_donorpaired_0044_barplot.png", plot_donorpaired_0044, width = 10, height = 8, dpi = 300)
ggsave("../Results/plot_donorpaired_0097_barplot.png", plot_donorpaired_0097, width = 10, height = 8, dpi = 300)
ggsave("../Results/plot_donorpaired_0485_barplot.png", plot_donorpaired_0485, width = 10, height = 8, dpi = 300)

plot_donorpaired_0044
plot_donorpaired_0097
plot_donorpaired_0485

```


## Check for singletons and doubletons
```{r}
length(which(taxa_sums(ps_filt_fixed_pruned_1) <= 1))

length(which(taxa_sums(ps_filt_fixed_pruned_1) <= 2))

```


```{r}

taxa_sums(ps_filt_fixed_pruned_1) %>%
  data.frame() %>%
  dplyr::rename(sum = ".") %>%
  arrange(sum)%>%
  tail(n=20) %>% kbl() %>%
  kableExtra::kable_paper(font_size = 12) %>%
  scroll_box(width = "200px", height = "300px")

```

## Normalization and check distribution
```{r}

ps_filt_fixed_rel_1 <- tax_transform(ps_filt_fixed_pruned_1, trans = "compositional")
head(ps_filt_fixed_rel_1@otu_table, n=10) 

```


## Transformation
```{r}

ps_filt_fixed_rel_clr_1 <- tax_transform(ps_filt_fixed_rel_1, trans = "clr", chain = TRUE)
head(ps_filt_fixed_rel_clr_1@otu_table, n=10)


```

## Aggregate data at specific taxonomic rank: Phylum and Genus
```{r}

ps_filt_fixed_phylum_1 <- tax_agg(ps_filt_fixed_pruned_1, rank = "Phylum")
ps_filt_fixed_class_1 <- tax_agg(ps_filt_fixed_pruned_1, rank = "Class")
ps_filt_fixed_order_1 <- tax_agg(ps_filt_fixed_pruned_1, rank = "Order")
ps_filt_fixed_family_1 <- tax_agg(ps_filt_fixed_pruned_1, rank = "Family")
ps_filt_fixed_genus_1 <- tax_agg(ps_filt_fixed_pruned_1, rank = "Genus")
#ps_filt_fixed_species_1 <- tax_agg(ps_filt_fixed_pruned_1, rank = "Species")

print(ps_filt_fixed_phylum_1)
print(ps_filt_fixed_genus_1)


summary_agg_1 <- data.frame(Taxa = c(colnames(ps_filt_fixed_pruned_1@tax_table)[c(2:6)], "ASVs"),
                 number_of_taxa=c(dim(ps_filt_fixed_phylum_1@otu_table)[2], dim(ps_filt_fixed_class_1@otu_table)[2], dim(ps_filt_fixed_order_1@otu_table)[2], dim(ps_filt_fixed_family_1@otu_table)[2], dim(ps_filt_fixed_genus_1@otu_table)[2], dim(ps_filt_fixed_pruned_1@otu_table)[2]))
print(summary_agg_1)

summary_agg_1

```


## Data exploration
```{r}
barplots_group <- ps_filt_fixed_pruned_1 %>%
  ps_mutate(host_sex = as.factor(host_sex)) %>%
  ps_mutate(group = as.factor(group)) %>%
  ps_arrange(group, week) %>% 
  comp_barplot(group_by = "group", tax_level = "Phylum", 
               tax_transform_for_plot = "compositional", n_taxa = 11, 
               bar_width = 0.8, merge_other = FALSE, label = "week", 
               sample_order = "default")

barplots_group_wrapped <- wrap_plots(barplots_group) +
  plot_layout(guides = "collect", nrow=2)

groups_barplot <- barplots_group_wrapped&
  facet_wrap("host_subject_id", scales = "free_x")& 
  theme(axis.text.x = element_text(angle = 60, hjust = 1, size = 10), 
        axis.ticks.y = element_blank(), leg.end.title = element_blank(), 
        legend.position = "bottom", strip.text = element_text(size=12),
        legend.key.size = unit(0.4, "cm"),
        legend.text = element_text(size=10))

groups_barplot

ggsave("../Results/groups_barplot.png", groups_barplot, width = 10, height = 8, dpi = 300)
```


```{r}

barplots_genus_group <- ps_filt_fixed_pruned_1 %>%
  ps_mutate(host_sex = as.factor(host_sex)) %>%
  ps_mutate(group = as.factor(group)) %>%
  ps_arrange(group, week) %>% 
  comp_barplot(group_by = "group", tax_level = "Genus", 
               tax_transform_for_plot = "compositional", n_taxa = 11, 
               bar_width = 0.8, merge_other = FALSE, label = "week", 
               sample_order = "default")

barplots_genus_group_wrapped <- wrap_plots(barplots_genus_group) +
  plot_layout(guides = "collect", nrow=2)

groups_genus_barplot <- barplots_genus_group_wrapped&
  facet_wrap("host_subject_id", scales = "free_x")& 
  theme(axis.text.x = element_text(angle = 60, hjust = 1, size = 10), 
        axis.ticks.y = element_blank(), leg.end.title = element_blank(), 
        legend.position = "bottom", strip.text = element_text(size=12),
        legend.key.size = unit(0.4, "cm"),
        legend.text = element_text(size=10))

groups_genus_barplot

ggsave("../Results/groups_genus_barplot.png", groups_genus_barplot, width = 10, height = 8, dpi = 300)

```


```{r}
create_group_barplot <- function(ps_obj, group_name) {
  ps_subset <- ps_obj %>% ps_filter(group == group_name)
  
  barplot <- ps_subset %>%
    ps_mutate(host_sex = as.factor(host_sex)) %>%
    ps_mutate(group = as.factor(group)) %>%
    ps_arrange(group, week) %>%
    comp_barplot(group_by = "group", tax_level = "Phylum",
                 tax_transform_for_plot = "compositional", n_taxa = 11,
                 bar_width = 0.8, merge_other = FALSE, label = "week",
                 sample_order = "default")
  
  wrapped_plot <- wrap_plots(barplot) +
    plot_layout(guides = "collect") &
    facet_wrap("host_subject_id", scales = "free_x") &
    theme(axis.text.x = element_text(angle = 60, hjust = 1, size = 10),
          axis.ticks.y = element_blank(), legend.title = element_blank(),
          legend.position = "bottom", strip.text = element_text(size = 12),
          legend.key.size = unit(0.4, "cm"),
          legend.text = element_text(size = 10))
  
  return(wrapped_plot)
}

plot_1 <- create_group_barplot(ps_filt_fixed_pruned_1, "active-active")
plot_2 <- create_group_barplot(ps_filt_fixed_pruned_1, "active-remission")
plot_3 <- create_group_barplot(ps_filt_fixed_pruned_1, "healthy-healthy")

# Save plots as separate images
ggsave("../Results/ac-ac_barplot.png", plot_1, width = 10, height = 8, dpi = 300)
ggsave("../Results/ac-rem_barplot.png", plot_2, width = 10, height = 8, dpi = 300)
ggsave("../Results/healthy_barplot.png", plot_3, width = 10, height = 8, dpi = 300)

# If you want to visualize them:
plot_1
plot_2
plot_3

```


```{r}
create_group_barplots <- function(ps_obj, group_name, max_per_plot = 6, output_prefix = "plot") {
  ps_subset <- ps_obj %>% ps_filter(group == group_name)
  
  subject_ids <- unique(sample_data(ps_subset)$host_subject_id)
  
  subject_batches <- split(subject_ids, ceiling(seq_along(subject_ids) / max_per_plot))
  
  plots <- list()
  
  for (i in seq_along(subject_batches)) {
    subjects <- subject_batches[[i]]
    
    ps_batch <- ps_subset %>% ps_filter(host_subject_id %in% subjects)
    
    barplot <- ps_batch %>%
      ps_mutate(host_sex = as.factor(host_sex)) %>%
      ps_mutate(group = as.factor(group)) %>%
      ps_arrange(group, week) %>%
      comp_barplot(group_by = "group", tax_level = "Phylum",
                   tax_transform_for_plot = "compositional", n_taxa = 11,
                   bar_width = 0.8, merge_other = FALSE, label = "week",
                   sample_order = "default")
    
    wrapped_plot <- wrap_plots(barplot) +
      plot_layout(guides = "collect") &
      facet_wrap("host_subject_id", scales = "free_x") &
      theme(axis.text.x = element_text(angle = 60, hjust = 1, size = 10),
            axis.ticks.y = element_blank(), legend.title = element_blank(),
            legend.position = "bottom", strip.text = element_text(size = 12),
            legend.key.size = unit(0.4, "cm"),
            legend.text = element_text(size = 10))
    
    plots[[i]] <- wrapped_plot
    
    ggsave(filename = paste0(output_prefix, "_1_batch_", i, ".png"), plot = wrapped_plot, 
           path = "../Results/", width = 10, height = 8, dpi = 300)
  }
  
  return(plots)
}

plots_ac_ac_1 <- create_group_barplots(ps_filt_fixed_pruned_1, "active-active", max_per_plot = 6, output_prefix = "ac_ac_barplot")
plots_ac_rem_1 <- create_group_barplots(ps_filt_fixed_pruned_1, "active-remission", max_per_plot = 6, output_prefix = "ac_rem_barplot")

plots_ac_ac_1

plots_ac_rem_1
```



```{r}
create_genus_group_barplot <- function(ps_obj, group_name) {
  ps_subset <- ps_obj %>% ps_filter(group == group_name)
  
  barplot <- ps_subset %>%
    ps_mutate(host_sex = as.factor(host_sex)) %>%
    ps_mutate(group = as.factor(group)) %>%
    ps_arrange(group, week) %>%
    comp_barplot(group_by = "group", tax_level = "Genus",
                 tax_transform_for_plot = "compositional", n_taxa = 11,
                 bar_width = 0.8, merge_other = FALSE, label = "week",
                 sample_order = "default")
  
  wrapped_plot <- wrap_plots(barplot) +
    plot_layout(guides = "collect") &
    facet_wrap("host_subject_id", scales = "free_x") &
    theme(axis.text.x = element_text(angle = 60, hjust = 1, size = 10),
          axis.ticks.y = element_blank(), legend.title = element_blank(),
          legend.position = "bottom", strip.text = element_text(size = 12),
          legend.key.size = unit(0.4, "cm"),
          legend.text = element_text(size = 10))
  
  return(wrapped_plot)
}

plot_genus_1 <- create_genus_group_barplot(ps_filt_fixed_pruned_1, "active-active")
plot_genus_2 <- create_genus_group_barplot(ps_filt_fixed_pruned_1, "active-remission")
plot_genus_3 <- create_genus_group_barplot(ps_filt_fixed_pruned_1, "healthy-healthy")

ggsave("../Results/ac-ac_genus_barplot.png", plot_genus_1, width = 10, height = 8, dpi = 300)
ggsave("../Results/ac-rem_genus_barplot.png", plot_genus_2, width = 10, height = 8, dpi = 300)
ggsave("../Results/healthy_genus_barplot.png", plot_genus_3, width = 10, height = 8, dpi = 300)

plot_genus_1
plot_genus_2
plot_genus_3


```

```{r}
create_group_barplots <- function(ps_obj, group_name, max_per_plot = 6, output_prefix = "plot") {
  ps_subset <- ps_obj %>% ps_filter(group == group_name)
  
  subject_ids <- unique(sample_data(ps_subset)$host_subject_id)
  
  subject_batches <- split(subject_ids, ceiling(seq_along(subject_ids) / max_per_plot))
  
  plots <- list()
  
  for (i in seq_along(subject_batches)) {
    subjects <- subject_batches[[i]]
    
    ps_batch <- ps_subset %>% ps_filter(host_subject_id %in% subjects)
    
    barplot <- ps_batch %>%
      ps_mutate(host_sex = as.factor(host_sex)) %>%
      ps_mutate(group = as.factor(group)) %>%
      ps_arrange(group, week) %>%
      comp_barplot(group_by = "group", tax_level = "Genus",
                   tax_transform_for_plot = "compositional", n_taxa = 11,
                   bar_width = 0.8, merge_other = FALSE, label = "week",
                   sample_order = "default")
    
    wrapped_plot <- wrap_plots(barplot) +
      plot_layout(guides = "collect") &
      facet_wrap("host_subject_id", scales = "free_x") &
      theme(axis.text.x = element_text(angle = 60, hjust = 1, size = 10),
            axis.ticks.y = element_blank(), legend.title = element_blank(),
            legend.position = "bottom", strip.text = element_text(size = 12),
            legend.key.size = unit(0.4, "cm"),
            legend.text = element_text(size = 10))
    
    plots[[i]] <- wrapped_plot
    
    ggsave(filename = paste0(output_prefix, "_1_genus_batch_", i, ".png"), plot = wrapped_plot, 
           path = "../Results/", width = 10, height = 8, dpi = 300)
  }
  
  return(plots)
}

plots_genus_ac_ac_1 <- create_group_barplots(ps_filt_fixed_pruned_1, "active-active", max_per_plot = 6, output_prefix = "ac_ac_barplot")
plots_genus_ac_rem_1 <- create_group_barplots(ps_filt_fixed_pruned_1, "active-remission", max_per_plot = 6, output_prefix = "ac_rem_barplot")

plots_genus_ac_ac_1

plots_genus_ac_rem_1
```


## Alpha diversity
```{r}
meta_737472 <- as.data.frame(as.matrix(sample_data(ps_filt_fixed_pruned_1)))

alpha_div_1 <- estimate_richness(ps_filt_fixed_pruned_1, measures = c("Observed", "Shannon", "InvSimpson", "Chao1"))
PD_1 <- calculatePD(ps_filt_fixed_pruned_1, justDF = TRUE, include_root = FALSE) %>% pull(PD)
alpha_div_1 <- alpha_div_1 %>% mutate(PD = PD_1)
rownames(alpha_div_1) <- gsub("^X", "", rownames(alpha_div_1))
head(alpha_div_1, n=20)

metadata_samples_alpha_1 <- bind_cols(meta_737472, alpha_div_1)
metadata_samples_long_1 <- metadata_samples_alpha_1 %>% pivot_longer(cols= Observed:PD, names_to = "alpha_div", values_to = "value")
kruskal_test_alpha_1 <- metadata_samples_long_1 %>% group_by(alpha_div) %>%
    rstatix::kruskal_test(value ~ group) %>% mutate(p_adj_BH = p.adjust(p, method = "BH"))

kruskal_test_alpha_1

write.csv(kruskal_test_alpha_1, "../Results/kruskal_test_alpha_1.csv", row.names = FALSE)


```

```{r}
dunn_test_alpha_1 <- metadata_samples_long_1 %>% group_by(alpha_div) %>%
    rstatix::dunn_test(value ~ group, p.adjust.method = "BH") %>%
  add_xy_position(x = "group", fun = "mean_sd", step.increase = 0.05, scales = "free")
dunn_test_alpha_1 %>% kbl() %>%
  kableExtra::kable_paper(font_size = 12)

dunn_test_alpha_table_1 <- apply(dunn_test_alpha_1,2,as.character)
write.csv(dunn_test_alpha_table_1, "../Results/dunn_test_alpha_table_1.csv", row.names = FALSE)


```

```{r}

dunn_test_alpha_plot_1 <- ggplot(metadata_samples_long_1, aes(x = group, y = value)) +
  geom_boxplot() +
  geom_jitter(shape = 21, color = "black", size = 2, aes(fill = group)) +
  facet_wrap(~alpha_div, scales = "free", ncol = 3) +
  scale_fill_nejm() +
  stat_pvalue_manual(data = dunn_test_alpha_1, 
                     label = "p.adj.signif", tip.length = 0.01, bracket.shorten = 0.05,
                     hide.ns = TRUE, size = 4) +
  scale_x_discrete(guide = guide_axis(n.dodge = 2)) +
  theme_bw() +
  theme(axis.text.x = element_text(size = 10), legend.position = "none")

ggsave("../Results/dunn_test_alpha_plot_1.png", dunn_test_alpha_plot_1, width = 13, height = 8, dpi = 300)

dunn_test_alpha_plot_1

```



```{r}

metadata_samples_alpha_1

kruskal_chao1_1 <- kruskal.test(Chao1 ~ group, data = metadata_samples_alpha_1)
kruskal_shannon_1 <- kruskal.test(Shannon ~ group, data = metadata_samples_alpha_1)
kruskal_invsimps <- kruskal.test(InvSimpson ~ group, data = metadata_samples_alpha_1)
kruskal_obsv_1 <- kruskal.test(Observed ~ group, data = metadata_samples_alpha_1)
kruskal_pd_1 <- kruskal.test(PD ~ group, data = metadata_samples_alpha_1)


H_chao1_1 <- kruskal_chao1_1$statistic
p_chao1_1 <- kruskal_chao1_1$p.value

H_shannon_1 <- kruskal_shannon_1$statistic
p_shannon_1 <- kruskal_shannon_1$p.value

H_invsimps_1 <- kruskal_invsimps$statistic
p_invsimps_1 <- kruskal_invsimps$p.value

H_obsv_1 <- kruskal_obsv_1$statistic
p_obsv_1 <- kruskal_obsv_1$p.value

H_pd_1 <- kruskal_pd_1$statistic
p_pd_1 <- kruskal_pd_1$p.value

cat("Kruskal-Wallis test for Chao1 index: H =", round(H_chao1_1, 3), "p =", round(p_chao1_1, 3), "\n")
cat("Kruskal-Wallis test for Shannon index: H =", round(H_shannon_1, 3), "p =", round(p_shannon_1, 3), "\n")
cat("Kruskal-Wallis test for Inverse Simpson index: H =", round(H_invsimps_1, 3), "p =", round(p_invsimps_1, 3), "\n")
cat("Kruskal-Wallis test for Observed Number of Species: H =", round(H_obsv_1, 3), "p =", round(p_obsv_1, 3), "\n")
cat("Kruskal-Wallis test for Faith's Phylogenetic Diversity: H =", round(H_pd_1, 3), "p =", round(p_pd_1, 3), "\n")


dunn_chao1_1 <- dunn_test_alpha_1 %>% filter(alpha_div == "Chao1")
dunn_shannon_1 <- dunn_test_alpha_1 %>% filter(alpha_div == "Shannon")
dunn_invsimps_1 <- dunn_test_alpha_1 %>% filter(alpha_div == "InvSimpson")
dunn_obsv_1 <- dunn_test_alpha_1 %>% filter(alpha_div == "Observed")
dunn_pd_1 <- dunn_test_alpha_1 %>% filter(alpha_div == "PD")

significant_chao1_1 <- dunn_chao1_1 %>% filter(p.adj < 0.05)
significant_shannon_1 <- dunn_shannon_1 %>% filter(p.adj < 0.05)
significant_invsimps_1 <- dunn_invsimps_1 %>% filter(p.adj < 0.05)
significant_obsv_1 <- dunn_obsv_1 %>% filter(p.adj < 0.05)
significant_pd_1 <- dunn_pd_1 %>% filter(p.adj < 0.05)

print(significant_chao1_1)
print(significant_shannon_1)
print(significant_invsimps_1)
print(significant_obsv_1)
print(significant_pd_1)

alpha_summary_1 <- metadata_samples_alpha_1 %>%
  group_by(group) %>%
  summarise(
    Chao1_mean = mean(Chao1, na.rm = TRUE),
    Chao1_sd = sd(Chao1, na.rm = TRUE),
    Shannon_mean = mean(Shannon, na.rm = TRUE),
    Shannon_sd = sd(Shannon, na.rm = TRUE),
    InvSimps_mean = mean(InvSimpson, na.rm = TRUE),
    InvSimps_sd = sd(InvSimpson, na.rm = TRUE), 
    Obsv_mean = mean(Observed, na.rm = TRUE),
    Obsv_sd = sd(Observed, na.rm = TRUE),
    PD_mean = mean(PD, na.rm = TRUE),
    PD_sd = sd(PD, na.rm = TRUE)
  )

print(alpha_summary_1)

write.csv(alpha_summary_1, "../Results/alpha_summary_1_table.csv", row.names = FALSE)

```


## Rarefied

```{r}
library(ranacapa); packageVersion("ranacapa")

ps_filt_fixed_pruned_1 %>% 
  ggrare(step = 50, color = "group", se = FALSE)

```


```{r}
set.seed(19950622)

ps_ff_rare_1 <- rarefy_even_depth(ps_filt_fixed_pruned_1, sample.size = 24190, replace = FALSE)

ps_ff_rare_1 %>% 
  ggrare(step = 50, color = "group", se = FALSE)

alpha_div_rare_1 <- estimate_richness(ps_ff_rare_1, measures = c("Observed", "Shannon", "InvSimpson", "Chao1"))
PD_rare_1 <- calculatePD(ps_ff_rare_1, justDF = TRUE, include_root = FALSE) %>% pull(PD)
alpha_div_rare_1 <- alpha_div_rare_1 %>% mutate(PD = PD_rare_1)
rownames(alpha_div_rare_1) <- gsub("^X", "", rownames(alpha_div_rare_1))
head(alpha_div_rare_1, n=20)

meta_rare_1 <- as.data.frame(as.matrix(sample_data(ps_ff_rare_1)))

metadata_samples_alpha_1_rare <- bind_cols(meta_rare_1, alpha_div_rare_1)
metadata_samples_long_rare_1 <- metadata_samples_alpha_1_rare %>% pivot_longer(cols= Observed:PD, names_to = "alpha_div", values_to = "value")
kruskal_test_alpha_rare_1 <- metadata_samples_long_rare_1 %>% group_by(alpha_div) %>%
  rstatix::kruskal_test(value ~ group) %>% mutate(p_adj_BH = p.adjust(p, method = "BH"))

kruskal_test_alpha_rare_1

write.csv(kruskal_test_alpha_rare_1, "../Results/kruskal_test_alpha_rare_1.csv", row.names = FALSE)

dunn_test_alpha_rare_1 <- metadata_samples_long_rare_1 %>% group_by(alpha_div) %>%
  rstatix::dunn_test(value ~ group, p.adjust.method = "BH") %>%
  add_xy_position(x = "group", fun = "mean_sd", step.increase = 0.05, scales = "free")
dunn_test_alpha_rare_1 %>% kbl() %>%
  kableExtra::kable_paper(font_size = 12)

dunn_test_alpha_rare_table_1 <- apply(dunn_test_alpha_rare_1,2,as.character)
write.csv(dunn_test_alpha_rare_table_1, "../Results/dunn_test_alpha_rare_table_1.csv", row.names = FALSE)

dunn_test_alpha_rare_plot_1 <- ggplot(metadata_samples_long_rare_1, aes(x = group, y = value)) +
  geom_boxplot() +
  geom_jitter(shape = 21, color = "black", size = 2, aes(fill = group)) +
  facet_wrap(~alpha_div, scales = "free", ncol = 3) +
  scale_fill_nejm() +
  stat_pvalue_manual(data = dunn_test_alpha_rare_1, 
                     label = "p.adj.signif", tip.length = 0.01, bracket.shorten = 0.05,
                     hide.ns = TRUE, size = 4) +
  scale_x_discrete(guide = guide_axis(n.dodge = 2)) +
  theme_bw() +
  theme(axis.text.x = element_text(size = 10), legend.position = "none")

ggsave("../Results/dunn_test_alpha_rare_plot_1.png", dunn_test_alpha_rare_plot_1, width = 13, height = 8, dpi = 300)

dunn_test_alpha_rare_plot_1

```


```{r}

metadata_samples_alpha_1_rare

kruskal_chao1_1_rare <- kruskal.test(Chao1 ~ group, data = metadata_samples_alpha_1_rare)
kruskal_shannon_1_rare <- kruskal.test(Shannon ~ group, data = metadata_samples_alpha_1_rare)
kruskal_invsimps_rare <- kruskal.test(InvSimpson ~ group, data = metadata_samples_alpha_1_rare)
kruskal_obsv_1_rare <- kruskal.test(Observed ~ group, data = metadata_samples_alpha_1_rare)
kruskal_pd_1_rare <- kruskal.test(PD ~ group, data = metadata_samples_alpha_1_rare)


H_chao1_1_rare <- kruskal_chao1_1_rare$statistic
p_chao1_1_rare <- kruskal_chao1_1_rare$p.value

H_shannon_1_rare <- kruskal_shannon_1_rare$statistic
p_shannon_1_rare <- kruskal_shannon_1_rare$p.value

H_invsimps_1_rare <- kruskal_invsimps_rare$statistic
p_invsimps_1_rare <- kruskal_invsimps_rare$p.value

H_obsv_1_rare <- kruskal_obsv_1_rare$statistic
p_obsv_1_rare <- kruskal_obsv_1_rare$p.value

H_pd_1_rare <- kruskal_pd_1_rare$statistic
p_pd_1_rare <- kruskal_pd_1_rare$p.value

cat("Kruskal-Wallis test for Chao1 index: H =", round(H_chao1_1_rare, 3), "p =", round(p_chao1_1_rare, 3), "\n")
cat("Kruskal-Wallis test for Shannon index: H =", round(H_shannon_1_rare, 3), "p =", round(p_shannon_1_rare, 3), "\n")
cat("Kruskal-Wallis test for Inverse Simpson index: H =", round(H_invsimps_1_rare, 3), "p =", round(p_invsimps_1_rare, 3), "\n")
cat("Kruskal-Wallis test for Observed Number of Species: H =", round(H_obsv_1_rare, 3), "p =", round(p_obsv_1_rare, 3), "\n")
cat("Kruskal-Wallis test for Faith's Phylogenetic Diversity: H =", round(H_pd_1_rare, 3), "p =", round(p_pd_1_rare, 3), "\n")


dunn_chao1_1_rare <- dunn_test_alpha_rare_1 %>% filter(alpha_div == "Chao1")
dunn_shannon_1_rare <- dunn_test_alpha_rare_1 %>% filter(alpha_div == "Shannon")
dunn_invsimps_1_rare <- dunn_test_alpha_rare_1 %>% filter(alpha_div == "InvSimpson")
dunn_obsv_1_rare <- dunn_test_alpha_rare_1 %>% filter(alpha_div == "Observed")
dunn_pd_1_rare <- dunn_test_alpha_rare_1 %>% filter(alpha_div == "PD")

significant_chao1_1_rare <- dunn_chao1_1_rare %>% filter(p.adj < 0.05)
significant_shannon_1_rare <- dunn_shannon_1_rare %>% filter(p.adj < 0.05)
significant_invsimps_1_rare <- dunn_invsimps_1_rare %>% filter(p.adj < 0.05)
significant_obsv_1_rare <- dunn_obsv_1_rare %>% filter(p.adj < 0.05)
significant_pd_1_rare <- dunn_pd_1_rare %>% filter(p.adj < 0.05)

print(significant_chao1_1_rare)
print(significant_shannon_1_rare)
print(significant_invsimps_1_rare)
print(significant_obsv_1_rare)
print(significant_pd_1_rare)

alpha_rare_summary_1 <- metadata_samples_alpha_1_rare %>%
  group_by(group) %>%
  summarise(
    Chao1_mean = mean(Chao1, na.rm = TRUE),
    Chao1_sd = sd(Chao1, na.rm = TRUE),
    Shannon_mean = mean(Shannon, na.rm = TRUE),
    Shannon_sd = sd(Shannon, na.rm = TRUE),
    InvSimps_mean = mean(InvSimpson, na.rm = TRUE),
    InvSimps_sd = sd(InvSimpson, na.rm = TRUE), 
    Obsv_mean = mean(Observed, na.rm = TRUE),
    Obsv_sd = sd(Observed, na.rm = TRUE),
    PD_mean = mean(PD, na.rm = TRUE),
    PD_sd = sd(PD, na.rm = TRUE)
  )

print(alpha_rare_summary_1)
write.csv(alpha_rare_summary_1, "../Results/alpha_rare_summary_table_1.csv", row.names = FALSE)

```

## Alpha div baseline and end
```{r}
## Baseline BP1 a-div analysis
# meta_737472 <- as.data.frame(as.matrix(sample_data(ps_filt_fixed_pruned_1)))

ps_filt_fixed_pruned_base_1 <- ps_filt_fixed_pruned_1 %>% ps_filter(!is.na(baseline_status))
meta_base_737472 <- meta_737472 %>% filter(!is.na(baseline_status))

alpha_div_base_1 <- estimate_richness(ps_filt_fixed_pruned_base_1, measures = c("Observed", "Shannon", "InvSimpson", "Chao1"))
PD_base_1 <- calculatePD(ps_filt_fixed_pruned_base_1, justDF = TRUE, include_root = FALSE) %>% pull(PD)
alpha_div_base_1 <- alpha_div_base_1 %>% mutate(PD = PD_base_1)
rownames(alpha_div_base_1) <- gsub("^X", "", rownames(alpha_div_base_1))
head(alpha_div_base_1, n=20)

metadata_samples_alpha_base_1 <- bind_cols(meta_base_737472, alpha_div_base_1)
metadata_samples_long_base_1 <- metadata_samples_alpha_base_1 %>% pivot_longer(cols= Observed:PD, names_to = "alpha_div", values_to = "value")
kruskal_test_alpha_base_1 <- metadata_samples_long_base_1 %>% group_by(alpha_div) %>%
  rstatix::kruskal_test(value ~ baseline_status) %>% mutate(p_adj_BH = p.adjust(p, method = "BH"))

kruskal_test_alpha_base_1

write.csv(kruskal_test_alpha_base_1, "../Results/kruskal_test_alpha_base_1.csv", row.names = FALSE)

dunn_test_alpha_base_1 <- metadata_samples_long_base_1 %>% group_by(alpha_div) %>%
  rstatix::dunn_test(value ~ baseline_status, p.adjust.method = "BH") %>%
  add_xy_position(x = "baseline_status", fun = "mean_sd", step.increase = 0.05, scales = "free")
dunn_test_alpha_base_1 %>% kbl() %>%
  kableExtra::kable_paper(font_size = 12)

dunn_test_alpha_table_base_1 <- apply(dunn_test_alpha_base_1,2,as.character)
write.csv(dunn_test_alpha_table_base_1, "../Results/dunn_test_alpha_table_base_1.csv", row.names = FALSE)


dunn_test_alpha_plot_base_1 <- ggplot(metadata_samples_long_base_1, aes(x = baseline_status, y = value)) +
  geom_boxplot() +
  geom_jitter(shape = 21, color = "black", size = 2, aes(fill = baseline_status)) +
  facet_wrap(~alpha_div, scales = "free", ncol = 3) +
  scale_fill_nejm() +
  stat_pvalue_manual(data = dunn_test_alpha_base_1, 
                     label = "p.adj.signif", tip.length = 0.01, bracket.shorten = 0.05,
                     hide.ns = TRUE, size = 4) +
  scale_x_discrete(guide = guide_axis(n.dodge = 2)) +
  theme_bw() +
  theme(axis.text.x = element_text(size = 10), legend.position = "none")

ggsave("../Results/dunn_test_alpha_plot_base_1.png", dunn_test_alpha_plot_base_1, width = 13, height = 8, dpi = 300)

dunn_test_alpha_plot_base_1

metadata_samples_alpha_base_1

kruskal_chao1_base_1 <- kruskal.test(Chao1 ~ baseline_status, data = metadata_samples_alpha_base_1)
kruskal_shannon_base_1 <- kruskal.test(Shannon ~ baseline_status, data = metadata_samples_alpha_base_1)
kruskal_invsimps <- kruskal.test(InvSimpson ~ baseline_status, data = metadata_samples_alpha_base_1)
kruskal_obsv_base_1 <- kruskal.test(Observed ~ baseline_status, data = metadata_samples_alpha_base_1)
kruskal_pd_base_1 <- kruskal.test(PD ~ baseline_status, data = metadata_samples_alpha_base_1)


H_chao1_base_1 <- kruskal_chao1_base_1$statistic
p_chao1_base_1 <- kruskal_chao1_base_1$p.value

H_shannon_base_1 <- kruskal_shannon_base_1$statistic
p_shannon_base_1 <- kruskal_shannon_base_1$p.value

H_invsimps_base_1 <- kruskal_invsimps$statistic
p_invsimps_base_1 <- kruskal_invsimps$p.value

H_obsv_base_1 <- kruskal_obsv_base_1$statistic
p_obsv_base_1 <- kruskal_obsv_base_1$p.value

H_pd_base_1 <- kruskal_pd_base_1$statistic
p_pd_base_1 <- kruskal_pd_base_1$p.value

cat("Kruskal-Wallis test for Chao1 index: H =", round(H_chao1_base_1, 3), "p =", round(p_chao1_base_1, 3), "\n")
cat("Kruskal-Wallis test for Shannon index: H =", round(H_shannon_base_1, 3), "p =", round(p_shannon_base_1, 3), "\n")
cat("Kruskal-Wallis test for Inverse Simpson index: H =", round(H_invsimps_base_1, 3), "p =", round(p_invsimps_base_1, 3), "\n")
cat("Kruskal-Wallis test for Observed Number of Species: H =", round(H_obsv_base_1, 3), "p =", round(p_obsv_base_1, 3), "\n")
cat("Kruskal-Wallis test for Faith's Phylogenetic Diversity: H =", round(H_pd_base_1, 3), "p =", round(p_pd_base_1, 3), "\n")


dunn_chao1_base_1 <- dunn_test_alpha_base_1 %>% filter(alpha_div == "Chao1")
dunn_shannon_base_1 <- dunn_test_alpha_base_1 %>% filter(alpha_div == "Shannon")
dunn_invsimps_base_1 <- dunn_test_alpha_base_1 %>% filter(alpha_div == "InvSimpson")
dunn_obsv_base_1 <- dunn_test_alpha_base_1 %>% filter(alpha_div == "Observed")
dunn_pd_base_1 <- dunn_test_alpha_base_1 %>% filter(alpha_div == "PD")

significant_chao1_base_1 <- dunn_chao1_base_1 %>% filter(p.adj < 0.05)
significant_shannon_base_1 <- dunn_shannon_base_1 %>% filter(p.adj < 0.05)
significant_invsimps_base_1 <- dunn_invsimps_base_1 %>% filter(p.adj < 0.05)
significant_obsv_base_1 <- dunn_obsv_base_1 %>% filter(p.adj < 0.05)
significant_pd_base_1 <- dunn_pd_base_1 %>% filter(p.adj < 0.05)

print(significant_chao1_base_1)
print(significant_shannon_base_1)
print(significant_invsimps_base_1)
print(significant_obsv_base_1)
print(significant_pd_base_1)

alpha_summary_base_1 <- metadata_samples_alpha_base_1 %>%
  group_by(baseline_status) %>%
  summarise(
    Chao1_mean = mean(Chao1, na.rm = TRUE),
    Chao1_sd = sd(Chao1, na.rm = TRUE),
    Shannon_mean = mean(Shannon, na.rm = TRUE),
    Shannon_sd = sd(Shannon, na.rm = TRUE),
    InvSimps_mean = mean(InvSimpson, na.rm = TRUE),
    InvSimps_sd = sd(InvSimpson, na.rm = TRUE), 
    Obsv_mean = mean(Observed, na.rm = TRUE),
    Obsv_sd = sd(Observed, na.rm = TRUE),
    PD_mean = mean(PD, na.rm = TRUE),
    PD_sd = sd(PD, na.rm = TRUE)
  )

print(alpha_summary_base_1)

write.csv(alpha_summary_base_1, "../Results/alpha_summary_base_1_table.csv", row.names = FALSE)

```

```{r}
## End BP1 a-div analysis

# meta_737472 <- as.data.frame(as.matrix(sample_data(ps_filt_fixed_pruned_1)))

ps_filt_fixed_pruned_end_1 <- ps_filt_fixed_pruned_1 %>% ps_filter(!is.na(disease_status_end))

meta_end_737472 <- meta_737472 %>% filter(!is.na(disease_status_end))

alpha_div_end_1 <- estimate_richness(ps_filt_fixed_pruned_end_1, measures = c("Observed", "Shannon", "InvSimpson", "Chao1"))
PD_end_1 <- calculatePD(ps_filt_fixed_pruned_end_1, justDF = TRUE, include_root = FALSE) %>% pull(PD)
alpha_div_end_1 <- alpha_div_end_1 %>% mutate(PD = PD_end_1)
rownames(alpha_div_end_1) <- gsub("^X", "", rownames(alpha_div_end_1))
head(alpha_div_end_1, n=20)

metadata_samples_alpha_end_1 <- bind_cols(meta_end_737472, alpha_div_end_1)
metadata_samples_long_end_1 <- metadata_samples_alpha_end_1 %>% pivot_longer(cols= Observed:PD, names_to = "alpha_div", values_to = "value")
kruskal_test_alpha_end_1 <- metadata_samples_long_end_1 %>% group_by(alpha_div) %>%
  rstatix::kruskal_test(value ~ disease_status_end) %>% mutate(p_adj_BH = p.adjust(p, method = "BH"))

kruskal_test_alpha_end_1

write.csv(kruskal_test_alpha_end_1, "../Results/kruskal_test_alpha_end_1.csv", row.names = FALSE)

dunn_test_alpha_end_1 <- metadata_samples_long_end_1 %>% group_by(alpha_div) %>%
  rstatix::dunn_test(value ~ disease_status_end, p.adjust.method = "BH") %>%
  add_xy_position(x = "disease_status_end", fun = "mean_sd", step.increase = 0.05, scales = "free")
dunn_test_alpha_end_1 %>% kbl() %>%
  kableExtra::kable_paper(font_size = 12)

dunn_test_alpha_table_end_1 <- apply(dunn_test_alpha_end_1,2,as.character)
write.csv(dunn_test_alpha_table_end_1, "../Results/dunn_test_alpha_table_end_1.csv", row.names = FALSE)


dunn_test_alpha_plot_end_1 <- ggplot(metadata_samples_long_end_1, aes(x = disease_status_end, y = value)) +
  geom_boxplot() +
  geom_jitter(shape = 21, color = "black", size = 2, aes(fill = disease_status_end)) +
  facet_wrap(~alpha_div, scales = "free", ncol = 3) +
  scale_fill_nejm() +
  stat_pvalue_manual(data = dunn_test_alpha_end_1, 
                     label = "p.adj.signif", tip.length = 0.01, bracket.shorten = 0.05,
                     hide.ns = TRUE, size = 4) +
  scale_x_discrete(guide = guide_axis(n.dodge = 2)) +
  theme_bw() +
  theme(axis.text.x = element_text(size = 10), legend.position = "none")

ggsave("../Results/dunn_test_alpha_plot_end_1.png", dunn_test_alpha_plot_end_1, width = 13, height = 8, dpi = 300)

dunn_test_alpha_plot_end_1

metadata_samples_alpha_end_1

kruskal_chao1_end_1 <- kruskal.test(Chao1 ~ disease_status_end, data = metadata_samples_alpha_end_1)
kruskal_shannon_end_1 <- kruskal.test(Shannon ~ disease_status_end, data = metadata_samples_alpha_end_1)
kruskal_invsimps <- kruskal.test(InvSimpson ~ disease_status_end, data = metadata_samples_alpha_end_1)
kruskal_obsv_end_1 <- kruskal.test(Observed ~ disease_status_end, data = metadata_samples_alpha_end_1)
kruskal_pd_end_1 <- kruskal.test(PD ~ disease_status_end, data = metadata_samples_alpha_end_1)


H_chao1_end_1 <- kruskal_chao1_end_1$statistic
p_chao1_end_1 <- kruskal_chao1_end_1$p.value

H_shannon_end_1 <- kruskal_shannon_end_1$statistic
p_shannon_end_1 <- kruskal_shannon_end_1$p.value

H_invsimps_end_1 <- kruskal_invsimps$statistic
p_invsimps_end_1 <- kruskal_invsimps$p.value

H_obsv_end_1 <- kruskal_obsv_end_1$statistic
p_obsv_end_1 <- kruskal_obsv_end_1$p.value

H_pd_end_1 <- kruskal_pd_end_1$statistic
p_pd_end_1 <- kruskal_pd_end_1$p.value

cat("Kruskal-Wallis test for Chao1 index: H =", round(H_chao1_end_1, 3), "p =", round(p_chao1_end_1, 3), "\n")
cat("Kruskal-Wallis test for Shannon index: H =", round(H_shannon_end_1, 3), "p =", round(p_shannon_end_1, 3), "\n")
cat("Kruskal-Wallis test for Inverse Simpson index: H =", round(H_invsimps_end_1, 3), "p =", round(p_invsimps_end_1, 3), "\n")
cat("Kruskal-Wallis test for Observed Number of Species: H =", round(H_obsv_end_1, 3), "p =", round(p_obsv_end_1, 3), "\n")
cat("Kruskal-Wallis test for Faith's Phylogenetic Diversity: H =", round(H_pd_end_1, 3), "p =", round(p_pd_end_1, 3), "\n")


dunn_chao1_end_1 <- dunn_test_alpha_end_1 %>% filter(alpha_div == "Chao1")
dunn_shannon_end_1 <- dunn_test_alpha_end_1 %>% filter(alpha_div == "Shannon")
dunn_invsimps_end_1 <- dunn_test_alpha_end_1 %>% filter(alpha_div == "InvSimpson")
dunn_obsv_end_1 <- dunn_test_alpha_end_1 %>% filter(alpha_div == "Observed")
dunn_pd_end_1 <- dunn_test_alpha_end_1 %>% filter(alpha_div == "PD")

significant_chao1_end_1 <- dunn_chao1_end_1 %>% filter(p.adj < 0.05)
significant_shannon_end_1 <- dunn_shannon_end_1 %>% filter(p.adj < 0.05)
significant_invsimps_end_1 <- dunn_invsimps_end_1 %>% filter(p.adj < 0.05)
significant_obsv_end_1 <- dunn_obsv_end_1 %>% filter(p.adj < 0.05)
significant_pd_end_1 <- dunn_pd_end_1 %>% filter(p.adj < 0.05)

print(significant_chao1_end_1)
print(significant_shannon_end_1)
print(significant_invsimps_end_1)
print(significant_obsv_end_1)
print(significant_pd_end_1)

alpha_summary_end_1 <- metadata_samples_alpha_end_1 %>%
  group_by(disease_status_end) %>%
  summarise(
    Chao1_mean = mean(Chao1, na.rm = TRUE),
    Chao1_sd = sd(Chao1, na.rm = TRUE),
    Shannon_mean = mean(Shannon, na.rm = TRUE),
    Shannon_sd = sd(Shannon, na.rm = TRUE),
    InvSimps_mean = mean(InvSimpson, na.rm = TRUE),
    InvSimps_sd = sd(InvSimpson, na.rm = TRUE), 
    Obsv_mean = mean(Observed, na.rm = TRUE),
    Obsv_sd = sd(Observed, na.rm = TRUE),
    PD_mean = mean(PD, na.rm = TRUE),
    PD_sd = sd(PD, na.rm = TRUE)
  )

print(alpha_summary_end_1)

write.csv(alpha_summary_end_1, "../Results/alpha_summary_end_1_table.csv", row.names = FALSE)

```


## Beta diversity
```{r}

PCoA_ASV_bray_1 <- ordinate(ps_filt_fixed_rel_1, distance = "bray", method = "PCoA")
beta_ASV_bray_1 <- phyloseq::distance(ps_filt_fixed_rel_1, "bray")
ado_bray_group_1 <- adonis2(beta_ASV_bray_1 ~ group, meta_737472)
print(ado_bray_group_1)

group.bd_1 <- betadisper(d = beta_ASV_bray_1, group = meta_737472$group, type="centroid")
bd_group_boxplot_1 <- boxplot(group.bd_1)

bd_group_boxplot_1

png("../Results/beta_dispersion_bc_boxplot_1.png", width = 3500, height = 2000, res = 300)  # High resolution
boxplot(group.bd_1, main = "Beta Dispersion per Group (BC) BP1", cex.axis = 1.2, cex.lab = 1.5)
dev.off()

permutest(group.bd_1)


BC_pcoa_plot_1 <- plot_ordination(ps_filt_fixed_rel_1, PCoA_ASV_bray_1, axes = c(1,2))+
  stat_conf_ellipse(geom = "polygon", alpha = 0.1, aes(color = group,  fill = group))+
  geom_point(size = 3, shape = 21, aes(fill = group))+
  stat_mean(aes(fill = group), shape = 23, size = 4) + 
  annotate("text", x =- 0.3, y = 0.5, color = "black", label = paste0("p=", round(ado_bray_group_1$`Pr(>F)`[1], digits = 3)))+
  annotate("text", x = -0.3, y = 0.45, color = "black", label = paste0("R\u00b2=", round((ado_bray_group_1$R2[1])*100, digits = 2), "%"))+
  scale_fill_nejm(name = "Group")+
  scale_color_nejm(name = "Group")+
  theme_bw()+
  theme(legend.position = "right")

ggsave("../Results/BC_pcoa_plot_1.png", BC_pcoa_plot_1, width = 10, height = 8, dpi = 300)

BC_pcoa_plot_1

ps_filt_fixed_rel_1@sam_data$host_sex <- as.factor(ps_filt_fixed_rel_1@sam_data$host_sex)

ado_bray_sex_1 <- adonis2(beta_ASV_bray_1 ~ host_sex, meta_737472)
print(ado_bray_sex_1)

BC_pcoa_sex_plot_1 <- plot_ordination(ps_filt_fixed_rel_1, PCoA_ASV_bray_1, axes = c(1,2))+
  stat_conf_ellipse(geom = "polygon", alpha = 0.1, aes(color = host_sex,  fill = host_sex))+
  geom_point(size = 3, shape = 21, aes(fill = host_sex))+
  stat_mean(aes(fill = host_sex), shape = 23, size = 4) + 
  annotate("text", x =- 0.3, y = 0.5, color = "black", label = paste0("p=", round(ado_bray_sex_1$`Pr(>F)`[1], digits = 3)))+
  annotate("text", x = -0.3, y = 0.45, color = "black", label = paste0("R\u00b2=", round((ado_bray_sex_1$R2[1])*100, digits = 2), "%"))+
  scale_fill_nejm(name = "Host sex")+
  scale_color_nejm(name = "Host Sex")+
  theme_bw()+
  theme(legend.position = "right")

ggsave("../Results/BC_pcoa_sex_plot_1.png", BC_pcoa_sex_plot_1, width = 10, height = 8, dpi = 300)

BC_pcoa_sex_plot_1

bd_df_1 <- data.frame(samples = rownames(group.bd_1$vectors),
                    distances = group.bd_1$distances,
                    group = meta_737472$group)

kruskal_bc_test_1 <- bd_df_1 %>% rstatix::kruskal_test(distances ~ group)
print(kruskal_bc_test_1)

dunn_bc_test_1 <- bd_df_1 %>% rstatix::dunn_test(distances ~ group, p.adjust.method = "BH")
print(dunn_bc_test_1)

dunn_bc_test_table_1 <- apply(dunn_bc_test_1,2,as.character)
write.csv(dunn_bc_test_table_1, "../Results/dunn_bc_test_table_1.csv", row.names = FALSE)

dunn_bc_pvalues_1 <- dunn_bc_test_1 %>% select(group1, group2, p.adj) %>%
  pivot_wider(names_from = group2, values_from = p.adj) %>%
  column_to_rownames(var = "group1")

dunn_bc_matrix_1 <- as.matrix(dunn_bc_pvalues_1)
dunn_bc_matrix_1[lower.tri(dunn_bc_matrix_1)] <- t(dunn_bc_matrix_1)[lower.tri(dunn_bc_matrix_1)]


dunn_bc_labels_1 <- multcompLetters(dunn_bc_matrix_1)$Letters
print(dunn_bc_labels_1)

```

```{r}

## Set root for WU so it runs with the same root as previous wu analysis

# ASV5155

# Define the root sequence (use the exact sequence you mentioned before)
root_seq <- "ASV5155"

# Extract tree from phyloseq object
tree_1 <- phy_tree(ps_filt_fixed_rel_1)

# Set the root
tree_rooted_1 <- root(tree_1, outgroup = root_seq, resolve.root = TRUE)

# Replace the tree in the phyloseq object
phy_tree(ps_filt_fixed_rel_1) <- tree_rooted_1
# ggtree(ps_filt_fixed_rel@phy_tree, options(ignore.negative.edge=TRUE))


PCoA_ASV_wunifrac_1 <- ordinate(ps_filt_fixed_rel_1, distance = "wunifrac", method = "PCoA")
beta_ASV_wunifrac_1 <- phyloseq::distance(ps_filt_fixed_rel_1, "wunifrac")
ado_wunifrac_group_1 <- adonis2(beta_ASV_wunifrac_1 ~ group, meta_737472)
print(ado_wunifrac_group_1)


group_wunifrac.bd_1 <- betadisper(d = beta_ASV_wunifrac_1, group = meta_737472$group, type="centroid")
boxplot(group_wunifrac.bd_1)

png("../Results/beta_dispersion_wu_boxplot_1.png", width = 3500, height = 2000, res = 300)  # High resolution
boxplot(group_wunifrac.bd_1, main = "Beta Dispersion per Group (WU) BP1", cex.axis = 1.2, cex.lab = 1.5)
dev.off()

permutest(group_wunifrac.bd_1)


```


```{r}

WU_pcoa_plot_1 <- plot_ordination(ps_filt_fixed_rel_1, PCoA_ASV_wunifrac_1, axes = c(1,2))+
  stat_conf_ellipse(geom = "polygon", alpha = 0.1, aes(color = group,  fill = group))+
  geom_point(size = 3, shape = 21, aes(fill = group))+
  stat_mean(aes(fill = group), shape = 23, size = 4) + 
  annotate("text", x = 0.1, y = -0.15, color = "black", label = paste0("p=", round(ado_wunifrac_group_1$`Pr(>F)`[1], digits = 3)))+
  annotate("text", x = 0.1, y = -0.13, color = "black", label = paste0("R\u00b2=", round((ado_wunifrac_group_1$R2[1])*100, digits = 2), "%"))+
  scale_fill_nejm(name = "Group")+
  scale_color_nejm(name = "Group")+
  theme_bw()+
  theme(legend.position = "right")

ggsave("../Results/WU_pcoa_plot_1.png", WU_pcoa_plot_1, width = 10, height = 8, dpi = 300)


WU_pcoa_plot_1 

ado_wunifrac_sex_1 <- adonis2(beta_ASV_wunifrac_1 ~ host_sex, meta_737472)
print(ado_wunifrac_sex_1)

WU_pcoa_sex_plot_1 <- plot_ordination(ps_filt_fixed_rel_1, PCoA_ASV_wunifrac_1, axes = c(1,2))+
  stat_conf_ellipse(geom = "polygon", alpha = 0.1, aes(color = host_sex,  fill = host_sex))+
  geom_point(size = 3, shape = 21, aes(fill = host_sex))+
  stat_mean(aes(fill = host_sex), shape = 23, size = 4) + 
  annotate("text", x = 0.1, y = -0.15, color = "black", label = paste0("p=", round(ado_wunifrac_sex_1$`Pr(>F)`[1], digits = 3)))+
  annotate("text", x = 0.1, y = -0.13, color = "black", label = paste0("R\u00b2=", round((ado_wunifrac_sex_1$R2[1])*100, digits = 2), "%"))+
  scale_fill_nejm(name = "Host sex")+
  scale_color_nejm(name = "Host sex")+
  theme_bw()+
  theme(legend.position = "right")

ggsave("../Results/WU_pcoa_sex_plot_1.png", WU_pcoa_sex_plot_1, width = 10, height = 8, dpi = 300)

WU_pcoa_sex_plot_1


ado_wunifrac_ttm_1 <- adonis2(beta_ASV_wunifrac_1 ~ ttm_arm, meta_737472)
print(ado_wunifrac_ttm_1)

WU_pcoa_ttm_plot_1 <- plot_ordination(ps_filt_fixed_rel_1, PCoA_ASV_wunifrac_1, axes = c(1,2))+
  stat_conf_ellipse(geom = "polygon", alpha = 0.1, aes(color = ttm_arm,  fill = ttm_arm))+
  geom_point(size = 3, shape = 21, aes(fill = ttm_arm))+
  stat_mean(aes(fill = ttm_arm), shape = 23, size = 4) + 
  annotate("text", x = 0.1, y = -0.15, color = "black", label = paste0("p=", round(ado_wunifrac_ttm_1$`Pr(>F)`[1], digits = 3)))+
  annotate("text", x = 0.1, y = -0.13, color = "black", label = paste0("R\u00b2=", round((ado_wunifrac_ttm_1$R2[1])*100, digits = 2), "%"))+
  scale_fill_nejm(name = "Treatment arm")+
  scale_color_nejm(name = "Treatment arm")+
  theme_bw()+
  theme(legend.position = "right")

ggsave("../Results/WU_pcoa_ttm_plot_1.png", WU_pcoa_ttm_plot_1, width = 10, height = 8, dpi = 300)

WU_pcoa_ttm_plot_1

wu_bd_df_1 <- data.frame(samples = rownames(group_wunifrac.bd_1$vectors),
                    distances = group_wunifrac.bd_1$distances,
                    group = meta_737472$group)

kruskal_wu_test_1 <- wu_bd_df_1 %>% rstatix::kruskal_test(distances ~ group)
print(kruskal_wu_test_1)

dunn_wu_test_1 <- wu_bd_df_1 %>% rstatix::dunn_test(distances ~ group, p.adjust.method = "BH", detailed = TRUE)
print(dunn_wu_test_1)

dunn_wu_test_table_1 <- apply(dunn_wu_test_1,2,as.character)
write.csv(dunn_wu_test_table_1, "../Results/dunn_wu_test_table_1.csv", row.names = FALSE)

dunn_wu_pvalues_1 <- dunn_wu_test_1 %>% select(group1, group2, p.adj) %>%
  pivot_wider(names_from = group2, values_from = p.adj) %>%
  column_to_rownames(var = "group1")

dunn_wu_matrix_1 <- as.matrix(dunn_wu_pvalues_1)
dunn_wu_matrix_1[lower.tri(dunn_wu_matrix_1)] <- t(dunn_wu_matrix_1)[lower.tri(dunn_wu_matrix_1)]


dunn_wu_labels_1 <- multcompLetters(dunn_wu_matrix_1)$Letters
print(dunn_wu_labels_1)

```

## Same with baseline and end samples only
```{r}

## For baseline samples by group
ps_ff_rel_base_1 <- ps_filt_fixed_rel_1 %>% ps_filter(!is.na(baseline_status))

meta_beta_base_1 <- as.data.frame(as.matrix(sample_data(ps_ff_rel_base_1)))

PCoA_ASV_bray_base_1 <- ordinate(ps_ff_rel_base_1, distance = "bray", method = "PCoA")
PCoA_ASV_bray_base_1


beta_ASV_bray_base_1 <- phyloseq::distance(ps_ff_rel_base_1, "bray")
ado_bray_group_base_1 <- adonis2(beta_ASV_bray_base_1 ~ baseline_status, meta_beta_base_1)
print(ado_bray_group_base_1)

baseline.bd_1 <- betadisper(d = beta_ASV_bray_base_1, group = meta_beta_base_1$baseline_status, type="centroid")
bd_baseline_boxplot_1 <- boxplot(baseline.bd_1)

bd_baseline_boxplot_1

png("../Results/betadisp_baseline_bc_boxplot_1.png", width = 3500, height = 2000, res = 300)  
boxplot(baseline.bd_1, main = "Beta Dispersion per Baseline Status (BC) BP1", cex.axis = 1.2, cex.lab = 1.5)
dev.off()

permutest(baseline.bd_1)

BC_baseline_pcoa_plot_1 <- plot_ordination(ps_ff_rel_base_1, PCoA_ASV_bray_base_1, axes = c(1,2))+
  stat_conf_ellipse(geom = "polygon", alpha = 0.1, aes(color = baseline_status,  fill = baseline_status))+
  geom_point(size = 3, shape = 21, aes(fill = baseline_status))+
  stat_mean(aes(fill = baseline_status), shape = 23, size = 4) + 
  annotate("text", x =- 0.3, y = 0.5, color = "black", label = paste0("p=", round(ado_bray_group_base_1$`Pr(>F)`[1], digits = 3)))+
  annotate("text", x = -0.3, y = 0.45, color = "black", label = paste0("R\u00b2=", round((ado_bray_group_base_1$R2[1])*100, digits = 2), "%"))+
  scale_fill_nejm(name = "Baseline Status")+
  scale_color_nejm(name = "Baseline Status")+
  theme_bw()+
  theme(legend.position = "right")

ggsave("../Results/BC_baseline_pcoa_plot_1.png", BC_baseline_pcoa_plot_1, width = 10, height = 8, dpi = 300)

BC_baseline_pcoa_plot_1


bd_base_df_1 <- data.frame(samples = rownames(baseline.bd_1$vectors),
                         distances = baseline.bd_1$distances,
                         group = meta_beta_base_1$baseline_status)

kruskal_bc_test_base_1 <- bd_base_df_1 %>% rstatix::kruskal_test(distances ~ group)
print(kruskal_bc_test_base_1)

dunn_bc_test_base_1 <- bd_base_df_1 %>% rstatix::dunn_test(distances ~ group, p.adjust.method = "BH")
print(dunn_bc_test_base_1)

dunn_bc_test_base_1_table <- apply(dunn_bc_test_base_1,2,as.character)
write.csv(dunn_bc_test_base_1_table, "../Results/dunn_bc_test_base_1_table.csv", row.names = FALSE)

dunn_bc_pvalues_base_1 <- dunn_bc_test_base_1 %>% select(group1, group2, p.adj) %>%
  pivot_wider(names_from = group2, values_from = p.adj) %>%
  column_to_rownames(var = "group1")

dunn_bc_matrix_base_1 <- as.matrix(dunn_bc_pvalues_base_1)
dunn_bc_matrix_base_1[lower.tri(dunn_bc_matrix_base_1)] <- t(dunn_bc_matrix_base_1)[lower.tri(dunn_bc_matrix_base_1)]


dunn_bc_labels_base_1 <- multcompLetters(dunn_bc_matrix_base_1)$Letters
print(dunn_bc_labels_base_1)

```


```{r}

## For end samples by group
ps_ff_rel_end_1 <- ps_filt_fixed_rel_1 %>% ps_filter(!is.na(disease_status_end))

meta_beta_end_1 <- as.data.frame(as.matrix(sample_data(ps_ff_rel_end_1)))

PCoA_ASV_bray_end_1 <- ordinate(ps_ff_rel_end_1, distance = "bray", method = "PCoA")
PCoA_ASV_bray_end_1


beta_ASV_bray_end_1 <- phyloseq::distance(ps_ff_rel_end_1, "bray")
ado_bray_group_end_1 <- adonis2(beta_ASV_bray_end_1 ~ disease_status_end, meta_beta_end_1)
print(ado_bray_group_end_1)

end_bc.bd_1 <- betadisper(d = beta_ASV_bray_end_1, group = meta_beta_end_1$disease_status_end, type="centroid")
bd_end_boxplot_1 <- boxplot(end_bc.bd_1)

bd_end_boxplot_1

png("../Results/betadisp_end_bc_boxplot_1.png", width = 3500, height = 2000, res = 300)  
boxplot(end_bc.bd_1, main = "Beta Dispersion per End Status (BC) BP1", cex.axis = 1.2, cex.lab = 1.5)
dev.off()

permutest(end_bc.bd_1)

BC_end_pcoa_plot_1 <- plot_ordination(ps_ff_rel_end_1, PCoA_ASV_bray_end_1, axes = c(1,2))+
  stat_conf_ellipse(geom = "polygon", alpha = 0.1, aes(color = disease_status_end,  fill = disease_status_end))+
  geom_point(size = 3, shape = 21, aes(fill = disease_status_end))+
  stat_mean(aes(fill = disease_status_end), shape = 23, size = 4) + 
  annotate("text", x =- 0.3, y = 0.5, color = "black", label = paste0("p=", round(ado_bray_group_end_1$`Pr(>F)`[1], digits = 3)))+
  annotate("text", x = -0.3, y = 0.45, color = "black", label = paste0("R\u00b2=", round((ado_bray_group_end_1$R2[1])*100, digits = 2), "%"))+
  scale_fill_nejm(name = "End Status")+
  scale_color_nejm(name = "End Status")+
  theme_bw()+
  theme(legend.position = "right")

ggsave("../Results/BC_end_pcoa_plot_1.png", BC_end_pcoa_plot_1, width = 10, height = 8, dpi = 300)

BC_end_pcoa_plot_1

bd_end_df_1 <- data.frame(samples = rownames(end_bc.bd_1$vectors),
                        distances = end_bc.bd_1$distances,
                        group = meta_beta_end_1$disease_status_end)

kruskal_bc_test_end_1 <- bd_end_df_1 %>% rstatix::kruskal_test(distances ~ group)
print(kruskal_bc_test_end_1)

dunn_bc_test_end_1 <- bd_end_df_1 %>% rstatix::dunn_test(distances ~ group, p.adjust.method = "BH")
print(dunn_bc_test_end_1)

dunn_bc_test_end_table_1 <- apply(dunn_bc_test_end_1,2,as.character)
write.csv(dunn_bc_test_end_table_1, "../Results/dunn_bc_test_end_table_1.csv", row.names = FALSE)

dunn_bc_pvalues_end_1 <- dunn_bc_test_end_1 %>% select(group1, group2, p.adj) %>%
  pivot_wider(names_from = group2, values_from = p.adj) %>%
  column_to_rownames(var = "group1")

dunn_bc_matrix_end_1 <- as.matrix(dunn_bc_pvalues_end_1)
dunn_bc_matrix_end_1[lower.tri(dunn_bc_matrix_end_1)] <- t(dunn_bc_matrix_end_1)[lower.tri(dunn_bc_matrix_end_1)]


dunn_bc_labels_end_1 <- multcompLetters(dunn_bc_matrix_end_1)$Letters
print(dunn_bc_labels_end_1)

```

## beta analysis for baseline and end by state
```{r}

## For baseline samples by baseline_status (bray)
ps_ff_rel_base_1 <- ps_filt_fixed_rel_1 %>% ps_filter(!is.na(baseline_status))

meta_beta_base_1 <- as.data.frame(as.matrix(sample_data(ps_ff_rel_base_1)))

PCoA_ASV_bray_base_1 <- ordinate(ps_ff_rel_base_1, distance = "bray", method = "PCoA")
PCoA_ASV_bray_base_1


beta_ASV_bray_base_1 <- phyloseq::distance(ps_ff_rel_base_1, "bray")
ado_bray_baseline_status_base_1 <- adonis2(beta_ASV_bray_base_1 ~ baseline_status, meta_beta_base_1)
print(ado_bray_baseline_status_base_1)

baseline_status.bd_1 <- betadisper(d = beta_ASV_bray_base_1, group = meta_beta_base_1$baseline_status, type="centroid")
bd_baseline_status_boxplot_1 <- boxplot(baseline_status.bd_1)

bd_baseline_status_boxplot_1

png("../Results/bd_basestat_bc_boxplot_1.png", width = 3500, height = 2000, res = 300)
boxplot(baseline_status.bd_1, main = "Beta Dispersion per Baseline Status (BC) BP1", cex.axis = 1.2, cex.lab = 1.5)
dev.off()

permutest(baseline_status.bd_1)

BC_baseline_status_pcoa_plot_1 <- plot_ordination(ps_ff_rel_base_1, PCoA_ASV_bray_base_1, axes = c(1,2))+
  stat_conf_ellipse(geom = "polygon", alpha = 0.1, aes(color = baseline_status,  fill = baseline_status))+
  geom_point(size = 3, shape = 21, aes(fill = baseline_status))+
  stat_mean(aes(fill = baseline_status), shape = 23, size = 4) + 
  annotate("text", x =- 0.3, y = 0.5, color = "black", label = paste0("p=", round(ado_bray_baseline_status_base_1$`Pr(>F)`[1], digits = 3)))+
  annotate("text", x = -0.3, y = 0.45, color = "black", label = paste0("R\u00b2=", round((ado_bray_baseline_status_base_1$R2[1])*100, digits = 2), "%"))+
  scale_fill_nejm(name = "Baseline Status")+
  scale_color_nejm(name = "Baseline Status")+
  theme_bw()+
  theme(legend.position = "right")

ggsave("../Results/BC_baseline_status_pcoa_plot_1.png", BC_baseline_status_pcoa_plot_1, width = 10, height = 8, dpi = 300)

BC_baseline_status_pcoa_plot_1


bd_base_status_df_1 <- data.frame(samples = rownames(baseline_status.bd_1$vectors),
                           distances = baseline_status.bd_1$distances,
                           group = meta_beta_base_1$baseline_status)

kruskal_bc_test_basestatus_1 <- bd_base_status_df_1 %>% rstatix::kruskal_test(distances ~ group)
print(kruskal_bc_test_basestatus_1)

dunn_bc_test_basestatus_1 <- bd_base_status_df_1 %>% rstatix::dunn_test(distances ~ group, p.adjust.method = "BH")
print(dunn_bc_test_basestatus_1)

dunn_bc_test_basestatus_1_table <- apply(dunn_bc_test_basestatus_1,2,as.character)
write.csv(dunn_bc_test_basestatus_1_table, "../Results/dunn_bc_test_basestatus_1_table.csv", row.names = FALSE)

dunn_bc_pvalues_basestatus_1 <- dunn_bc_test_basestatus_1 %>% select(group1, group2, p.adj) %>%
  pivot_wider(names_from = group2, values_from = p.adj) %>%
  column_to_rownames(var = "group1")

dunn_bc_matrix_basestatus_1 <- as.matrix(dunn_bc_pvalues_basestatus_1)
dunn_bc_matrix_basestatus_1[lower.tri(dunn_bc_matrix_basestatus_1)] <- t(dunn_bc_matrix_basestatus_1)[lower.tri(dunn_bc_matrix_basestatus_1)]


dunn_bc_labels_basestatus_1 <- multcompLetters(dunn_bc_matrix_basestatus_1)$Letters
print(dunn_bc_labels_basestatus_1)

```



```{r}

## For end samples
ps_ff_rel_endstatus_1 <- ps_filt_fixed_rel_1 %>% ps_filter(!is.na(disease_status_end))

meta_beta_endstatus_1 <- as.data.frame(as.matrix(sample_data(ps_ff_rel_endstatus_1)))

PCoA_ASV_bray_endstatus_1 <- ordinate(ps_ff_rel_endstatus_1, distance = "bray", method = "PCoA")
PCoA_ASV_bray_endstatus_1


beta_ASV_bray_endstatus_1 <- phyloseq::distance(ps_ff_rel_endstatus_1, "bray")
ado_bray_endstatus_1 <- adonis2(beta_ASV_bray_endstatus_1 ~ disease_status_end, meta_beta_endstatus_1)
print(ado_bray_endstatus_1)

end_bc.bd_1 <- betadisper(d = beta_ASV_bray_endstatus_1, group = meta_beta_endstatus_1$disease_status_end, type="centroid")
bd_endstatus_bc_boxplot_1 <- boxplot(end_bc.bd_1)

bd_endstatus_bc_boxplot_1

png("../Results/bd_endstat_bc_boxplot_1.png", width = 3500, height = 2000, res = 300)  
boxplot(end_bc.bd_1, main = "Beta Dispersion per End Status (BC) BP1", cex.axis = 1.2, cex.lab = 1.5)
dev.off()

permutest(end_bc.bd_1)

BC_endstatus_pcoa_plot_1 <- plot_ordination(ps_ff_rel_endstatus_1, PCoA_ASV_bray_endstatus_1, axes = c(1,2))+
  stat_conf_ellipse(geom = "polygon", alpha = 0.1, aes(color = disease_status_end,  fill = disease_status_end))+
  geom_point(size = 3, shape = 21, aes(fill = disease_status_end))+
  stat_mean(aes(fill = disease_status_end), shape = 23, size = 4) + 
  annotate("text", x =- 0.3, y = 0.5, color = "black", label = paste0("p=", round(ado_bray_endstatus_1$`Pr(>F)`[1], digits = 3)))+
  annotate("text", x = -0.3, y = 0.45, color = "black", label = paste0("R\u00b2=", round((ado_bray_endstatus_1$R2[1])*100, digits = 2), "%"))+
  scale_fill_nejm(name = "End Status")+
  scale_color_nejm(name = "End Status")+
  theme_bw()+
  theme(legend.position = "right")

ggsave("../Results/BC_endstatus_pcoa_plot_1.png", BC_endstatus_pcoa_plot_1, width = 10, height = 8, dpi = 300)

BC_endstatus_pcoa_plot_1

bd_bc_endstatus_df_1 <- data.frame(samples = rownames(end_bc.bd_1$vectors),
                               distances = end_bc.bd_1$distances,
                               group = meta_beta_endstatus_1$disease_status_end)

kruskal_bc_test_endstatus_1 <- bd_bc_endstatus_df_1 %>% rstatix::kruskal_test(distances ~ group)
print(kruskal_bc_test_endstatus_1)

dunn_bc_test_endstatus_1 <- bd_bc_endstatus_df_1 %>% rstatix::dunn_test(distances ~ group, p.adjust.method = "BH")
print(dunn_bc_test_endstatus_1)

dunn_bc_test_endstatus_table_1 <- apply(dunn_bc_test_endstatus_1,2,as.character)
write.csv(dunn_bc_test_endstatus_table_1, "../Results/dunn_bc_test_endstatus_table_1.csv", row.names = FALSE)

dunn_bc_pvalues_endstatus_1 <- dunn_bc_test_endstatus_1 %>% select(group1, group2, p.adj) %>%
  pivot_wider(names_from = group2, values_from = p.adj) %>%
  column_to_rownames(var = "group1")

dunn_bc_matrix_endstatus_1 <- as.matrix(dunn_bc_pvalues_endstatus_1)
dunn_bc_matrix_endstatus_1[lower.tri(dunn_bc_matrix_endstatus_1)] <- t(dunn_bc_matrix_endstatus_1)[lower.tri(dunn_bc_matrix_endstatus_1)]


dunn_bc_labels_endstatus_1 <- multcompLetters(dunn_bc_matrix_endstatus_1)$Letters
print(dunn_bc_labels_endstatus_1)

```


```{r}
## For baseline samples by baseline_status
ps_ff_rel_base_1 <- ps_filt_fixed_rel_1 %>% ps_filter(!is.na(baseline_status))

meta_beta_base_1 <- as.data.frame(as.matrix(sample_data(ps_ff_rel_base_1)))

PCoA_ASV_wu_base_1 <- ordinate(ps_ff_rel_base_1, distance = "wunifrac", method = "PCoA")
PCoA_ASV_wu_base_1


beta_ASV_wu_base_1 <- phyloseq::distance(ps_ff_rel_base_1, "wunifrac")
ado_wu_baseline_status_1 <- adonis2(beta_ASV_wu_base_1 ~ baseline_status, meta_beta_base_1)
print(ado_wu_baseline_status_1)

baseline_status_wu.bd_1 <- betadisper(d = beta_ASV_wu_base_1, group = meta_beta_base_1$baseline_status, type="centroid")
bd_baseline_status_wu_boxplot_1 <- boxplot(baseline_status_wu.bd_1)

bd_baseline_status_wu_boxplot_1

png("../Results/bd_basestat_wu_boxplot_1.png", width = 3500, height = 2000, res = 300)  
boxplot(baseline_status_wu.bd_1, main = "Beta Dispersion per Baseline Status (WU) BP1", cex.axis = 1.2, cex.lab = 1.5)
dev.off()

permutest(baseline_status_wu.bd_1)

WU_baseline_status_pcoa_plot_1 <- plot_ordination(ps_ff_rel_base_1, PCoA_ASV_wu_base_1, axes = c(1,2))+
  stat_conf_ellipse(geom = "polygon", alpha = 0.1, aes(color = baseline_status,  fill = baseline_status))+
  geom_point(size = 3, shape = 21, aes(fill = baseline_status))+
  stat_mean(aes(fill = baseline_status), shape = 23, size = 4) + 
  annotate("text", x =- 0.3, y = 0.5, color = "black", label = paste0("p=", round(ado_wu_baseline_status_1$`Pr(>F)`[1], digits = 3)))+
  annotate("text", x = -0.3, y = 0.45, color = "black", label = paste0("R\u00b2=", round((ado_wu_baseline_status_1$R2[1])*100, digits = 2), "%"))+
  scale_fill_nejm(name = "Baseline Status")+
  scale_color_nejm(name = "Baseline Status")+
  theme_bw()+
  theme(legend.position = "right")

ggsave("../Results/WU_baseline_status_pcoa_plot_1.png", WU_baseline_status_pcoa_plot_1, width = 10, height = 8, dpi = 300)

WU_baseline_status_pcoa_plot_1

ado_wu_base_sex_1 <- adonis2(beta_ASV_wu_base_1 ~ host_sex, meta_beta_base_1)

WU_base_sex_pcoa_plot_1 <- plot_ordination(ps_ff_rel_base_1, PCoA_ASV_wu_base_1, axes = c(1,2))+
  stat_conf_ellipse(geom = "polygon", alpha = 0.1, aes(color = host_sex,  fill = host_sex))+
  geom_point(size = 3, shape = 21, aes(fill = host_sex))+
  stat_mean(aes(fill = host_sex), shape = 23, size = 4) + 
  annotate("text", x =- 0.3, y = 0.5, color = "black", label = paste0("p=", round(ado_wu_base_sex_1$`Pr(>F)`[1], digits = 3)))+
  annotate("text", x = -0.3, y = 0.45, color = "black", label = paste0("R\u00b2=", round((ado_wu_base_sex_1$R2[1])*100, digits = 2), "%"))+
  scale_fill_nejm(name = "Host Sex")+
  scale_color_nejm(name = "Host Sex")+
  theme_bw()+
  theme(legend.position = "right")

ggsave("../Results/WU_base_sex_pcoa_plot_1.png", WU_base_sex_pcoa_plot_1, width = 10, height = 8, dpi = 300)

WU_base_sex_pcoa_plot_1

bd_base_status_df_1 <- data.frame(samples = rownames(baseline_status_wu.bd_1$vectors),
                                  distances = baseline_status_wu.bd_1$distances,
                                  group = meta_beta_base_1$baseline_status)

kruskal_wu_test_basestatus_1 <- bd_base_status_df_1 %>% rstatix::kruskal_test(distances ~ group)
print(kruskal_wu_test_basestatus_1)

dunn_wu_test_basestatus_1 <- bd_base_status_df_1 %>% rstatix::dunn_test(distances ~ group, p.adjust.method = "BH")
print(dunn_wu_test_basestatus_1)

dunn_wu_test_basestatus_1_table <- apply(dunn_wu_test_basestatus_1,2,as.character)
write.csv(dunn_wu_test_basestatus_1_table, "../Results/dunn_wu_test_basestatus_1_table.csv", row.names = FALSE)

dunn_wu_pvalues_basestatus_1 <- dunn_wu_test_basestatus_1 %>% select(group1, group2, p.adj) %>%
  pivot_wider(names_from = group2, values_from = p.adj) %>%
  column_to_rownames(var = "group1")

dunn_wu_matrix_basestatus_1 <- as.matrix(dunn_wu_pvalues_basestatus_1)
dunn_wu_matrix_basestatus_1[lower.tri(dunn_wu_matrix_basestatus_1)] <- t(dunn_wu_matrix_basestatus_1)[lower.tri(dunn_wu_matrix_basestatus_1)]


dunn_wu_labels_basestatus_1 <- multcompLetters(dunn_wu_matrix_basestatus_1)$Letters
print(dunn_wu_labels_basestatus_1)

```



```{r}

## For end samples
ps_ff_rel_endstatus_1 <- ps_filt_fixed_rel_1 %>% ps_filter(!is.na(disease_status_end))

meta_beta_endstatus_1 <- as.data.frame(as.matrix(sample_data(ps_ff_rel_endstatus_1)))

PCoA_ASV_wu_endstatus_1 <- ordinate(ps_ff_rel_endstatus_1, distance = "wunifrac", method = "PCoA")
PCoA_ASV_wu_endstatus_1


beta_ASV_wu_endstatus_1 <- phyloseq::distance(ps_ff_rel_endstatus_1, "wunifrac")
ado_wu_endstatus_1 <- adonis2(beta_ASV_wu_endstatus_1 ~ disease_status_end, meta_beta_endstatus_1)
print(ado_wu_endstatus_1)

end_wu.bd_1 <- betadisper(d = beta_ASV_wu_endstatus_1, group = meta_beta_endstatus_1$disease_status_end, type="centroid")
bd_endstatus_wu_boxplot_1 <- boxplot(end_wu.bd_1)

bd_endstatus_wu_boxplot_1

png("../Results/bd_endstat_wu_boxplot_1.png", width = 3500, height = 2000, res = 300)  
boxplot(end_wu.bd_1, main = "Beta Dispersion per End Status (WU) BP1", cex.axis = 1.2, cex.lab = 1.5)
dev.off()

permutest(end_wu.bd_1)

WU_endstatus_pcoa_plot_1 <- plot_ordination(ps_ff_rel_endstatus_1, PCoA_ASV_wu_endstatus_1, axes = c(1,2))+
  stat_conf_ellipse(geom = "polygon", alpha = 0.1, aes(color = disease_status_end,  fill = disease_status_end))+
  geom_point(size = 3, shape = 21, aes(fill = disease_status_end))+
  stat_mean(aes(fill = disease_status_end), shape = 23, size = 4) + 
  annotate("text", x =- 0.3, y = 0.5, color = "black", label = paste0("p=", round(ado_wu_endstatus_1$`Pr(>F)`[1], digits = 3)))+
  annotate("text", x = -0.3, y = 0.45, color = "black", label = paste0("R\u00b2=", round((ado_wu_endstatus_1$R2[1])*100, digits = 2), "%"))+
  scale_fill_nejm(name = "End Status")+
  scale_color_nejm(name = "End Status")+
  theme_bw()+
  theme(legend.position = "right")

ggsave("../Results/WU_endstatus_pcoa_plot_1.png", WU_endstatus_pcoa_plot_1, width = 10, height = 8, dpi = 300)

WU_endstatus_pcoa_plot_1

ado_wu_end_sex_1 <- adonis2(beta_ASV_wu_endstatus_1 ~ host_sex, meta_beta_endstatus_1)

WU_end_sex_pcoa_plot_1 <- plot_ordination(ps_ff_rel_endstatus_1, PCoA_ASV_wu_endstatus_1, axes = c(1,2))+
  stat_conf_ellipse(geom = "polygon", alpha = 0.1, aes(color = host_sex,  fill = host_sex))+
  geom_point(size = 3, shape = 21, aes(fill = host_sex))+
  stat_mean(aes(fill = host_sex), shape = 23, size = 4) + 
  annotate("text", x =- 0.3, y = 0.5, color = "black", label = paste0("p=", round(ado_wu_end_sex_1$`Pr(>F)`[1], digits = 3)))+
  annotate("text", x = -0.3, y = 0.45, color = "black", label = paste0("R\u00b2=", round((ado_wu_end_sex_1$R2[1])*100, digits = 2), "%"))+
  scale_fill_nejm(name = "Host Sex")+
  scale_color_nejm(name = "Host Sex")+
  theme_bw()+
  theme(legend.position = "right")

ggsave("../Results/WU_end_sex_pcoa_plot_1.png", WU_end_sex_pcoa_plot_1, width = 10, height = 8, dpi = 300)

WU_end_sex_pcoa_plot_1

ado_wu_end_ttm_1 <- adonis2(beta_ASV_wu_endstatus_1 ~ ttm_arm, meta_beta_endstatus_1)

WU_end_ttm_pcoa_plot_1 <- plot_ordination(ps_ff_rel_endstatus_1, PCoA_ASV_wu_endstatus_1, axes = c(1,2))+
  stat_conf_ellipse(geom = "polygon", alpha = 0.1, aes(color = ttm_arm,  fill = ttm_arm))+
  geom_point(size = 3, shape = 21, aes(fill = ttm_arm))+
  stat_mean(aes(fill = ttm_arm), shape = 23, size = 4) + 
  annotate("text", x =- 0.3, y = 0.5, color = "black", label = paste0("p=", round(ado_wu_end_ttm_1$`Pr(>F)`[1], digits = 3)))+
  annotate("text", x = -0.3, y = 0.45, color = "black", label = paste0("R\u00b2=", round((ado_wu_end_ttm_1$R2[1])*100, digits = 2), "%"))+
  scale_fill_nejm(name = "Treatment arm")+
  scale_color_nejm(name = "Treatment arm")+
  theme_bw()+
  theme(legend.position = "right")

ggsave("../Results/WU_end_ttm_pcoa_plot_1.png", WU_end_ttm_pcoa_plot_1, width = 10, height = 8, dpi = 300)

WU_end_ttm_pcoa_plot_1

bd_endstatus_df_1 <- data.frame(samples = rownames(end_wu.bd_1$vectors),
                               distances = end_wu.bd_1$distances,
                               group = meta_beta_endstatus_1$disease_status_end)

kruskal_wu_test_endstatus_1 <- bd_endstatus_df_1 %>% rstatix::kruskal_test(distances ~ group)
print(kruskal_wu_test_endstatus_1)

dunn_wu_test_endstatus_1 <- bd_endstatus_df_1 %>% rstatix::dunn_test(distances ~ group, p.adjust.method = "BH")
print(dunn_wu_test_endstatus_1)

dunn_wu_test_endstatus_table_1 <- apply(dunn_wu_test_endstatus_1,2,as.character)
write.csv(dunn_wu_test_endstatus_table_1, "../Results/dunn_wu_test_endstatus_table_1.csv", row.names = FALSE)

dunn_wu_pvalues_endstatus_1 <- dunn_wu_test_endstatus_1 %>% select(group1, group2, p.adj) %>%
  pivot_wider(names_from = group2, values_from = p.adj) %>%
  column_to_rownames(var = "group1")

dunn_wu_matrix_endstatus_1 <- as.matrix(dunn_wu_pvalues_endstatus_1)
dunn_wu_matrix_endstatus_1[lower.tri(dunn_wu_matrix_endstatus_1)] <- t(dunn_wu_matrix_endstatus_1)[lower.tri(dunn_wu_matrix_endstatus_1)]


dunn_wu_labels_endstatus_1 <- multcompLetters(dunn_wu_matrix_endstatus_1)$Letters
print(dunn_wu_labels_endstatus_1)


## Same KW and Dunn's testing by ttm_arm
end_ttm_wu.bd_1 <- betadisper(d = beta_ASV_wu_endstatus_1, group = meta_beta_endstatus_1$ttm_arm, type="centroid")
bd_end_ttm_wu_boxplot_1 <- boxplot(end_ttm_wu.bd_1)

print(ado_wu_end_ttm_1)

bd_end_ttm_wu_boxplot_1

png("../Results/bd_end_ttm_wu_boxplot_1.png", width = 3500, height = 2000, res = 300)  
boxplot(end_ttm_wu.bd_1, main = "Beta Dispersion at End per Treatment Arm (WU) BP1", cex.axis = 1.2, cex.lab = 1.5)
dev.off()

permutest(end_ttm_wu.bd_1)

bd_ttm_df_1 <- data.frame(samples = rownames(end_ttm_wu.bd_1$vectors),
                                distances = end_ttm_wu.bd_1$distances,
                                group = meta_beta_endstatus_1$ttm_arm)

kruskal_wu_test_ttm_1 <- bd_ttm_df_1 %>% rstatix::kruskal_test(distances ~ group)
print(kruskal_wu_test_ttm_1)

dunn_wu_test_ttm_1 <- bd_ttm_df_1 %>% rstatix::dunn_test(distances ~ group, p.adjust.method = "BH")
print(dunn_wu_test_ttm_1)

dunn_wu_test_ttm_table_1 <- apply(dunn_wu_test_ttm_1,2,as.character)
write.csv(dunn_wu_test_ttm_table_1, "../Results/dunn_wu_test_ttm_table_1.csv", row.names = FALSE)

dunn_wu_pvalues_ttm_1 <- dunn_wu_test_ttm_1 %>% select(group1, group2, p.adj) %>%
  pivot_wider(names_from = group2, values_from = p.adj) %>%
  column_to_rownames(var = "group1")

dunn_wu_matrix_ttm_1 <- as.matrix(dunn_wu_pvalues_ttm_1)
dunn_wu_matrix_ttm_1[lower.tri(dunn_wu_matrix_ttm_1)] <- t(dunn_wu_matrix_ttm_1)[lower.tri(dunn_wu_matrix_ttm_1)]


dunn_wu_labels_ttm_1 <- multcompLetters(dunn_wu_matrix_ttm_1)$Letters
print(dunn_wu_labels_ttm_1)


```


## Check prevalence of taxa
```{r}

print(ps_filt_fixed_pruned_1)

ps_filt_fixed_prev_1 <- tax_filter(ps_filt_fixed_pruned_1,
  min_prevalence = 0.01)
  
print(ps_filt_fixed_prev_1)


prev_thresholds_1 <- c(0.05, 0.01, 0.001)

prev_filt_results_1 <- data.frame(min_prevalence = prev_thresholds_1, num_taxa_retained = NA, perc_taxa_removed = NA)

total_taxa_1 <- nrow(tax_table(ps_filt_fixed_pruned_1))

for (i in seq_along(prev_thresholds_1)) {
  ps_filt_fixed_samples_prev_all_1 <- ps_filt_fixed_pruned_1 %>%
    tax_filter(min_prevalence = prev_thresholds_1[i])
  
  num_taxa_retained <- nrow(tax_table(ps_filt_fixed_samples_prev_all_1))
  perc_taxa_removed <- 100 * (total_taxa_1 - num_taxa_retained) / total_taxa_1
  
  prev_filt_results_1$num_taxa_retained[i] <- num_taxa_retained
  prev_filt_results_1$perc_taxa_removed[i] <- perc_taxa_removed
}

print(prev_filt_results_1)

```

```{r}
ps_filt_fixed_genus_pat_1 <- ps_filt_fixed_genus_1 %>% ps_filter(donor == "no")

ps_filt_fixed_genus_pat_rel_1 <- tax_transform(ps_filt_fixed_genus_pat_1, trans = "compositional")
input_sample_metadata_pat_1 <- ps_filt_fixed_genus_pat_rel_1 %>% meta
input_sample_metadata_pat_1 <- input_sample_metadata_pat_1 %>% mutate(host_sex = as.factor(host_sex))

input_data_genus_pat_Maaslin_1 <- as.data.frame(otu_table(ps_filt_fixed_genus_pat_rel_1))

input_data_genus_pat_Maaslin_1 <- input_data_genus_Maaslin_1[sapply(input_data_genus_Maaslin_1, is.numeric)]

output_directory_genus_1 <- "../Results/Maaslin2_output_genus_pat_1"

fit_data_genus_pat_1 <- Maaslin2::Maaslin2(
    input_data = input_data_genus_pat_Maaslin_1,
    input_metadata = input_sample_metadata_pat_1,
    output = output_directory_genus_1,
    analysis_method = "LM",
    fixed_effects = c("group", "host_sex", "host_age"),
    random_effects = "host_subject_id",
    normalization = "NONE", 
    correction = "BH",
    transform = "LOG",
    min_abundance = 0,
    min_prevalence = 0.01,
    reference = "group,active-active")

```

```{r}

head(fit_data_genus_1$results, n=20) %>% kbl() %>%
  kableExtra::kable_paper(font_size = 12) %>%
  scroll_box(width = "1000px", height = "500px")

```


```{r}
taxa_abund_kw_1 <- as.data.frame(otu_table(ps_filt_fixed_genus_1))

# Convert row names to a column (sample IDs)
taxa_abund_kw_1 <- taxa_abund_kw_1 %>% 
  tibble::rownames_to_column(var = "samples")

# Convert sample_data from phyloseq object to a proper data frame
metadata_samples_kw_1 <- as(sample_data(ps_filt_fixed_genus_1), "data.frame") %>% 
  tibble::rownames_to_column(var = "samples") # Ensure sample IDs are a column

metadata_samples_kw_1 <- metadata_samples_kw_1 %>% 
  select(c(samples, group))


# Merge metadata with abundance data
taxa_data_kw_1 <- left_join(metadata_samples_kw_1, taxa_abund_kw_1, by = "samples") 

taxa_data_kw_tib_1 <- taxa_data_kw_1 %>%
  as_tibble()  

# Function to perform Kruskal-Wallis test on each taxon
kruskal_test_results_1 <- taxa_data_kw_tib_1 %>%
  pivot_longer(cols = -c(samples, group), names_to = "taxon", values_to = "abundance") %>%
  group_by(taxon) %>%
  rstatix::kruskal_test(abundance ~ group) %>%
  rstatix::adjust_pvalue(method = "BH") %>%  # Adjust for multiple testing (Benjamini-Hochberg)
  arrange(p.adj)

# View significant results (p.adj < 0.05)
significant_taxa_kw_1 <- kruskal_test_results_1 %>% filter(p.adj < 0.05)

# Display top results
significant_taxa_kw_1 %>% 
  arrange(p.adj) %>% 
  head(n=20)

top_10_sig_taxa_1 <- significant_taxa_kw_1 %>% arrange(p.adj) %>% head(n=10)

top_10_sig_taxa_1

```


```{r}

top10_sig_taxa_names_1 <- top_10_sig_taxa_1$taxon

ps_ff_genus_melt_1 <- ps_filt_fixed_genus_rel_1 %>% psmelt()


ps_traj_taxa_filt_1 <- ps_ff_genus_melt_1 %>% filter(OTU %in% top10_sig_taxa_names_1)
ps_traj_taxa_filt_1 <- ps_traj_taxa_filt_1 %>% filter(!host_assigned_donor_subject_id == "is donor")

plot_subject_trajectory <- function(df, group_name) {
  
  df_filt <- df %>% filter(group == group_name)
  
  ggplot(df_filt, aes(x = week, y = Abundance, group = OTU, color = OTU)) +
    geom_line(aes(linetype = OTU), size = 1) + # Line style per taxa
    geom_point(size = 3) + 
    scale_color_manual(values = distinct_palette(10, pal = "brewerPlus"))
    
    
}

plots_acac_1 <- plot_subject_trajectory(ps_traj_taxa_filt_1, "active-active")

plots_acrem_1 <- plot_subject_trajectory(ps_traj_taxa_filt_1, "active-remission")

wr_acac_lineplots_1 <- wrap_plots(plots_acac_1) + plot_layout(guides = "collect") &
  facet_wrap("host_subject_id", scales = "free_x") &
  theme(axis.text.x = element_text(angle = 60, hjust = 1, size = 10),
        axis.ticks.y = element_blank(), legend.title = element_blank(),
        legend.position = "bottom", strip.text = element_text(size = 12),
        legend.key.size = unit(0.4, "cm"),
        legend.text = element_text(size = 14))
wr_acrem_lineplots_1 <- wrap_plots(plots_acrem_1) + plot_layout(guides = "collect") &
  facet_wrap("host_subject_id", scales = "free_x") &
  theme(axis.text.x = element_text(angle = 60, hjust = 1, size = 10),
        axis.ticks.y = element_blank(), legend.title = element_blank(),
        legend.position = "bottom", strip.text = element_text(size = 12),
        legend.key.size = unit(0.4, "cm"),
        legend.text = element_text(size = 14))

ggsave(filename = "wr_acac_lineplots_1.png", plot = wr_acac_lineplots_1, path = "../Results/",
       width = 14, height = 11, dpi = 320)

ggsave(filename = "wr_acrem_lineplots_1.png", plot = wr_acrem_lineplots_1, path = "../Results/",
       width = 14, height = 11, dpi = 320)


wr_acac_lineplots_1
wr_acrem_lineplots_1

```



