---
title: "data_analysis"
author: "Leire A. Murua"
date: "2025-03-06"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(DECIPHER); packageVersion("DECIPHER")
library(phangorn); packageVersion("phangorn")
library(phyloseq); packageVersion("phyloseq")
library(Biostrings); packageVersion("Biostrings")
library(ggplot2); packageVersion("ggplot2")
library(tidyverse); packageVersion("tidyverse")
library(vegan); packageVersion("vegan")
library(microbiome); packageVersion("microbiome")
library(dplyr); packageVersion("dplyr")
library(tidyr); packageVersion("tidyr")
library(microViz); packageVersion("microViz")
library(dada2); packageVersion("dada2")
library(ggtree); packageVersion("ggtree")
library(biomeUtils); packageVersion("biomeUtils")
library(kableExtra); packageVersion("kableExtra")
library(patchwork); packageVersion("patchwork")
library(ggsci); packageVersion("ggsci")
library(RColorBrewer); packageVersion("RColorBrewer")
library(rstatix); packageVersion("rstatix")
library(ggpubr); packageVersion("ggpubr")
library(Maaslin2); packageVersion("Maaslin2")
library(multcompView); packageVersion("multcompView")

library(beepr)

```


```{r}

seqtab_nochim <- readRDS("../seqtab_nochim_combined.rds")

taxa_combined <- readRDS("../taxa_combined.rds")

meta_x_merged <- readRDS("meta_merged.rds")

```

## COMBINED DATA WORKFLOW
```{r}
sample_names <- rownames(meta_merged)
  
sample_names 

rownames(seqtab_nochim) <- sub("^\\d+\\.", "", rownames(seqtab_nochim))

seqtab_filtered <- seqtab_nochim[sample_names, ]

taxa_names <- colnames(seqtab_filtered)

taxa_filtered <- taxa_combined[taxa_names, ]


seqs <- getSequences(seqtab_filtered)
names(seqs) <- seqs

alignment <- AlignSeqs(DNAStringSet(seqs), anchor=NA)
saveRDS(alignment, "alignment.rds")
# alignment <- readRDS("alignment.rds")


phang_align <- phyDat(as(alignment, "matrix"), type="DNA")
saveRDS(phang_align, "phang_align.rds")
# phang_align <- readRDS("phang_align.rds")


## Check kind of distance
dm <- dist.ml(phang_align)

saveRDS(dm, "dm.rds")
# dm <- readRDS("dm.rds")


treeNJ <- NJ(dm)
saveRDS(treeNJ, "treeNJ.rds")
# treeNJ <- readRDS("treeNJ.rds")

```

```{r}

ps <- phyloseq(
  otu_table(seqtab_filtered, taxa_are_rows = FALSE),
  tax_table(as.matrix(taxa_filtered)),
  sample_data(meta_merged),
  phy_tree(treeNJ)
  )


dna <- Biostrings::DNAStringSet(taxa_names(ps))
names(dna) <- taxa_names(ps)
ps <- merge_phyloseq(ps, dna)
taxa_names(ps) <- paste0("ASV", seq(ntaxa(ps)))
ps

ps <- ps %>% 
  ps_mutate(bioproject = case_when(
    bioproject == "PRJNA737472" ~ 1,
    bioproject == "PRJNA951422" ~ 2
  ))

ps_sum <- summarize_phyloseq(ps)
summary_ps_table <- data.frame(do.call(rbind, ps_sum[c(1:10)]))
write.csv(summary_ps_table, "Results/summary_ps_table.csv", row.names = FALSE)

summary_ps_table

```

```{r}

head(ps@otu_table, n=10) %>% kbl() %>% 
  kableExtra::kable_paper(font_size = 12) %>% 
  scroll_box(width = "800px", height = "300px")

head(ps@tax_table, n=10) %>% kbl() %>%
  kableExtra::kable_paper(font_size = 12) %>%
  scroll_box(width = "800px", height = "300px")

metadata_x <- ps %>% meta()
head(metadata_x, n=10) %>% kbl() %>%
  kableExtra::kable_paper(font_size = 12) %>%
  scroll_box(width = "800px", height = "300px")

ggtree(ps@phy_tree, options(ignore.negative.edge=TRUE))

```


```{r}
Chloroplast <- ntaxa(subset_taxa(ps, Order == "Chloroplast"))

Mitochondria <- ntaxa(subset_taxa(ps, Family == "Mitochondria"))

chl_mt_df <- data.frame(Taxa = c("Chloroplast", "Mitochondria"),
                       nASVs = c(Chloroplast, Mitochondria)) %>%
  mutate(fraction = round(nASVs/ntaxa(ps), 3))
chl_mt_df 

ps_filt <- subset_taxa(ps, Order != "Chloroplast")
ps_filt <- subset_taxa(ps_filt, Family != "Mitochondria")

ps_filt@tax_table %>% 
  data.frame() %>% 
  group_by(Kingdom) %>% 
  dplyr::count() %>%
  mutate(fraction = round(n/ntaxa(ps_filt), 3))

ps_filt <- subset_taxa(ps_filt, Kingdom == "Bacteria")

## Filtered / cleaned ps object
phylum_filt_table <- ps_filt@tax_table %>% 
  data.frame() %>% 
  group_by(Phylum) %>% 
  dplyr::count() %>% 
  arrange(desc(n)) 

phylum_filt_table

write.csv(phylum_filt_table, "Results/phylum_filt_table.csv", row.names = FALSE)

phylum_filt_pie <- ggplot(phylum_filt_table, aes(x="", y=n, fill=Phylum)) +
  geom_bar(stat="identity", width=1) +
  coord_polar("y", start=0) +
  theme(legend.text = element_text(size=14))

ggsave("Results/phylum_filt_pie.png", phylum_filt_pie, device = png, width = 10, height = 7)

# ps_filt <- ps_filt %>% 
#   ps_mutate(bioproject = case_when(
#     bioproject == "PRJNA737472" ~ 1,
#     bioproject == "PRJNA951422" ~ 2
#   ))

## Unfilteed ps object
phylum_table <- ps@tax_table %>% 
  data.frame() %>% 
  group_by(Phylum) %>% 
  dplyr::count() %>% 
  arrange(desc(n)) 

phylum_table

write.csv(phylum_table, "Results/phylum_table.csv", row.names = FALSE)

phylum_pie <- ggplot(phylum_table, aes(x="", y=n, fill=Phylum)) +
  geom_bar(stat="identity", width=1) +
  coord_polar("y", start=0)

ggsave("Results/phylum_pie.png", phylum_pie, device = png, width = 10, height = 7)


ps_filt_sum <- summarize_phyloseq(ps_filt)
summary_ps_filt_table <- data.frame(do.call(rbind, ps_filt_sum[c(1:10)]))
write.csv(summary_ps_filt_table, "Results/summary_ps_filt_table.csv", row.names = FALSE)

summary_ps_filt_table

```

```{r}

ps_filt_fixed <- ps_filt %>%
 tax_fix(
  min_length = 4,
  sep="_", anon_unique=TRUE, 
  suffix_rank="classified")

ps_filt_fixed@tax_table

ggtree(ps_filt_fixed@phy_tree, options(ignore.negative.edge=TRUE))


```


```{r}

sum_of_reads <- as.data.frame(sample_sums(ps_filt_fixed@otu_table)) %>% rownames_to_column(var = "samples") %>%
  dplyr::rename(sum_of_reads = `sample_sums(ps_filt_fixed@otu_table)`)%>%
  left_join((metadata_x %>% rownames_to_column(var = "samples")), by = "samples")
head((sum_of_reads %>% arrange(sum_of_reads)), n=20)%>% kbl() %>%
  kableExtra::kable_paper(font_size = 12) %>%
  scroll_box(width = "800px", height = "300px")

range(sum_of_reads$sum_of_reads)

ps_filt_fixed_1 <- ps_filt_fixed %>% ps_filter(bioproject == "1")

sum_of_reads <- as.data.frame(sample_sums(ps_filt_fixed@otu_table)) %>% rownames_to_column(var = "samples") %>%
  dplyr::rename(sum_of_reads = `sample_sums(ps_filt_fixed@otu_table)`)%>%
  left_join((metadata_x %>% rownames_to_column(var = "samples")), by = "samples")


# View(sum_of_reads)
sum_of_reads %>% 
  group_by(bioproject) %>% 
  summarise(libsize = range(sum_of_reads))



```

```{r}
sum_of_reads_plot <- sum_of_reads %>% arrange(sum_of_reads) %>% 
  ggplot(aes(x = reorder(samples, sum_of_reads), y = sum_of_reads, fill = group)) +
  geom_point(shape = 21, size = 2, color = "black")+
  scale_fill_manual(values = c(
    "active-active" = "darkred",
    "active-remission" = "orange",
    "remission-remission" = "yellow",
    "remission-active" = "indianred",
    "healthy-healthy" = "darkgreen"
  ))+
  theme_bw()+
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank(),
        legend.text =  element_text(size = 14))

ggsave("Results/SumOfReads_plot.png", sum_of_reads_plot, device = png, width = 10, height = 8)

sum_of_reads_plot

```

## Normalization
```{r}
ps_filt_fixed_gen_rel <- ps_filt_fixed %>% 
  tax_transform("compositional", rank = "Genus")

ps_filt_fixed_gen_rel@sam_data$group <- as.factor(ps_filt_fixed_gen_rel@sam_data$group)

top_taxa <- tax_top(ps_filt_fixed_gen_rel, n=40)

melt_genus_rel_end <- ps_filt_fixed_gen_rel %>% psmelt() %>% 
  mutate(Genus_top = case_when(Genus %in% top_taxa ~ Genus, 
                               TRUE ~ "other")) %>% 
  mutate(Genus_top = as.factor(Genus_top)) %>% 
  mutate(Genus_top = fct_relevel(Genus_top, c(top_taxa, "other"))) %>% 
  filter(!is.na(disease_status_end))

main_genus_end <- melt_genus_rel_end %>%
  ggplot(aes(x = Sample, y = Abundance, fill = Genus_top)) +
  geom_bar(stat = "identity", color = "black") +
  scale_fill_manual(values = distinct_palette(40, pal = "brewerPlus", add = "grey70"))+
  theme_bw()+
  labs(title= "Genus Bar plot for end samples") +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank(), 
        legend.position = "right")


melt_genus_rel_base <- ps_filt_fixed_gen_rel %>% psmelt() %>% 
  mutate(Genus_top = case_when(Genus %in% top_taxa ~ Genus, 
                               TRUE ~ "other")) %>% 
  mutate(Genus_top = as.factor(Genus_top)) %>% 
  mutate(Genus_top = fct_relevel(Genus_top, c(top_taxa, "other"))) %>% 
  filter(!is.na(baseline_status))

main_genus_base <- melt_genus_rel_base %>%
  ggplot(aes(x = Sample, y = Abundance, fill = Genus_top)) +
  geom_bar(stat = "identity", color = "black")+
  scale_fill_manual(values = distinct_palette(40, pal = "brewerPlus", add = "grey70"))+
  theme_bw()+
  labs(title= "Genus Bar plot for baseline samples") +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank(),
        legend.position = "right")

melt_genus_rel_end_patient <- ps_filt_fixed_gen_rel %>% psmelt() %>% 
  mutate(Genus_top = case_when(Genus %in% top_taxa ~ Genus, 
                               TRUE ~ "other")) %>% 
  mutate(Genus_top = as.factor(Genus_top)) %>% 
  mutate(Genus_top = fct_relevel(Genus_top, c(top_taxa, "other"))) %>% 
  filter(!is.na(disease_status_end)) %>% 
  filter(donor == "no")

main_genus_end_pat <- melt_genus_rel_end_patient %>%
  ggplot(aes(x = Sample, y = Abundance, fill = Genus_top)) +
  geom_bar(stat = "identity", color = "black")+
  scale_fill_manual(values = distinct_palette(40, pal = "brewerPlus", add = "grey70"))+
  theme_bw()+
  labs(title= "Genus Bar plot for end Patient samples") +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank(),
        legend.position = "right")

melt_genus_rel_base_patient <- ps_filt_fixed_gen_rel %>% psmelt() %>% 
  mutate(Genus_top = case_when(Genus %in% top_taxa ~ Genus, 
                               TRUE ~ "other")) %>% 
  mutate(Genus_top = as.factor(Genus_top)) %>% 
  mutate(Genus_top = fct_relevel(Genus_top, c(top_taxa, "other"))) %>% 
  filter(!is.na(baseline_status)) %>% 
  filter(donor == "no")

main_genus_base_pat <- melt_genus_rel_base_patient %>%
  ggplot(aes(x = Sample, y = Abundance, fill = Genus_top)) +
  geom_bar(stat = "identity", color = "black")+
  scale_fill_manual(values = distinct_palette(40, pal = "brewerPlus", add = "grey70"))+
  theme_bw()+
  labs(title= "Genus Bar plot for baseline Patient samples") +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank(), 
        legend.position = "right")


ggsave("Results/main_genus_end_barplot.png", main_genus_end, device = png, width = 10, height = 8)
ggsave("Results/main_genus_base_barplot.png", main_genus_base, device = png, width = 10, height = 8)
ggsave("Results/main_genus_end_pat_barplot.png", main_genus_end_pat, device = png, width = 10, height = 8)
ggsave("Results/main_genus_base_pat_barplot.png", main_genus_base_pat, device = png, width = 10, height = 8)

main_genus_end
main_genus_base
main_genus_end_pat
main_genus_base_pat

ps_ff_gen_rel_end <- ps_filt_fixed_gen_rel %>% ps_filter(!is.na(disease_status_end))
top_taxa_end <- tax_top(ps_ff_gen_rel_end, n=40)

melt_genus_rel_end <- ps_filt_fixed_gen_rel %>% psmelt() %>% 
  mutate(Genus_top = case_when(Genus %in% top_taxa_end ~ Genus, 
                               TRUE ~ "other")) %>% 
  mutate(Genus_top = as.factor(Genus_top)) %>% 
  mutate(Genus_top = fct_relevel(Genus_top, c(top_taxa_end, "other"))) %>% 
  filter(!is.na(disease_status_end))

main_gr_genus_end <- melt_genus_rel_end %>%
  ggplot(aes(x = Sample, y = Abundance, fill = Genus_top)) +
  geom_bar(stat = "identity", color = "black") +
  facet_wrap("group", scales = "free_x") +
  scale_fill_manual(values = distinct_palette(40, pal = "brewerPlus", add = "grey70"))+
  theme_bw()+
  labs(title= "Genus Bar plot for end samples") +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank(), 
        legend.position = "right",
        legend.text = element_text(size=10))

ps_ff_gen_rel_base <- ps_filt_fixed_gen_rel %>% ps_filter(!is.na(baseline_status))
top_taxa_base <- tax_top(ps_ff_gen_rel_base, n=40)

melt_genus_rel_base <- ps_filt_fixed_gen_rel %>% psmelt() %>% 
  mutate(Genus_top = case_when(Genus %in% top_taxa_base ~ Genus, 
                               TRUE ~ "other")) %>% 
  mutate(Genus_top = as.factor(Genus_top)) %>% 
  mutate(Genus_top = fct_relevel(Genus_top, c(top_taxa_base, "other"))) %>% 
  filter(!is.na(baseline_status))

main_gr_genus_base <- melt_genus_rel_base %>%
  ggplot(aes(x = Sample, y = Abundance, fill = Genus_top)) +
  geom_bar(stat = "identity", color = "black")+
  facet_wrap("group", scales = "free_x") +
  scale_fill_manual(values = distinct_palette(40, pal = "brewerPlus", add = "grey70"))+
  theme_bw()+
  labs(title= "Genus Bar plot for baseline samples") +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank(),
        legend.position = "right", 
        legend.text = element_text(size=10))

ps_ff_gen_rel_end_pat <- ps_filt_fixed_gen_rel %>% ps_filter(!is.na(disease_status_end) & donor == "no")
top_taxa_end_pat <- tax_top(ps_ff_gen_rel_end_pat, n=40)

melt_genus_rel_end_patient <- ps_filt_fixed_gen_rel %>% psmelt() %>% 
  mutate(Genus_top = case_when(Genus %in% top_taxa_end_pat ~ Genus, 
                               TRUE ~ "other")) %>% 
  mutate(Genus_top = as.factor(Genus_top)) %>% 
  mutate(Genus_top = fct_relevel(Genus_top, c(top_taxa_end_pat, "other"))) %>% 
  filter(!is.na(disease_status_end)) %>% 
  filter(donor == "no")

main_gr_genus_end_pat <- melt_genus_rel_end_patient %>%
  ggplot(aes(x = Sample, y = Abundance, fill = Genus_top)) +
  geom_bar(stat = "identity", color = "black")+
  facet_wrap("group", scales = "free_x") +
  scale_fill_manual(values = distinct_palette(40, pal = "brewerPlus", add = "grey70"))+
  theme_bw()+
  labs(title= "Genus Bar plot for end Patient samples") +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank(),
        legend.position = "right", 
        legend.text = element_text(size=10))

ps_ff_gen_rel_base_pat <- ps_filt_fixed_gen_rel %>% ps_filter(!is.na(baseline_status) & donor == "no")
top_taxa_base_pat <- tax_top(ps_ff_gen_rel_base, n=40)

melt_genus_rel_base_patient <- ps_filt_fixed_gen_rel %>% psmelt() %>% 
  mutate(Genus_top = case_when(Genus %in% top_taxa_base_pat ~ Genus, 
                               TRUE ~ "other")) %>% 
  mutate(Genus_top = as.factor(Genus_top)) %>% 
  mutate(Genus_top = fct_relevel(Genus_top, c(top_taxa_base_pat, "other"))) %>% 
  filter(!is.na(baseline_status)) %>% 
  filter(donor == "no")

main_gr_genus_base_pat <- melt_genus_rel_base_patient %>%
  ggplot(aes(x = Sample, y = Abundance, fill = Genus_top)) +
  geom_bar(stat = "identity", color = "black")+
  facet_wrap("group", scales = "free_x") +
  scale_fill_manual(values = distinct_palette(40, pal = "brewerPlus", add = "grey70"))+
  theme_bw()+
  labs(title= "Genus Bar plot for baseline Patient samples") +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank(), 
        legend.position = "right", 
        legend.text = element_text(size=10))


ggsave("Results/main_gr_genus_end_barplot.png", main_gr_genus_end, device = png, width = 13, height = 8)
ggsave("Results/main_gr_genus_base_barplot.png", main_gr_genus_base, device = png, width = 13, height = 8)
ggsave("Results/main_gr_genus_end_pat_barplot.png", main_gr_genus_end_pat, device = png, width = 13, height = 8)
ggsave("Results/main_gr_genus_base_pat_barplot.png", main_gr_genus_base_pat, device = png, width = 13, height = 8)

main_gr_genus_end
main_gr_genus_base
main_gr_genus_end_pat
main_gr_genus_base_pat
```



## Check for singletons and doubletons
```{r}
length(which(taxa_sums(ps_filt_fixed) <= 1))

length(which(taxa_sums(ps_filt_fixed) <= 2))

```


```{r}

taxa_sums(ps_filt_fixed) %>%
  data.frame() %>%
  dplyr::rename(sum = ".") %>%
  arrange(sum)%>%
  tail(n=20) %>% kbl() %>%
  kableExtra::kable_paper(font_size = 12) %>%
  scroll_box(width = "200px", height = "300px")

```

## Normalization and check distribution
```{r}

ps_filt_fixed_rel <- tax_transform(ps_filt_fixed, trans = "compositional")
head(ps_filt_fixed_rel@otu_table, n=10) 

ggplot(data = data.frame("total_reads" =  sample_sums(ps_filt_fixed),
                         "observed" = estimate_richness(ps_filt_fixed, measures = "Observed")[, 1]),
       aes(x = total_reads, y = observed)) +
  geom_point() +
  geom_smooth(method="lm", se = FALSE) +
  labs(x = "\nTotal Reads", y = "Observed Richness\n")

```

## Continue here ***

## Transformation
```{r}

ps_filt_fixed_rel_clr <- tax_transform(ps_filt_fixed_rel, trans = "clr", chain = TRUE)
head(ps_filt_fixed_rel_clr@otu_table, n=10)


```

## Aggregate data at specific taxonomic rank: Phylum and Genus
```{r}

ps_filt_fixed_phy_rel <- ps_filt_fixed %>% 
  tax_transform("compositional", rank = "Phylum")
ps_filt_fixed_class_rel <- ps_filt_fixed %>% 
  tax_transform("compositional", rank = "Class")
ps_filt_fixed_ord_rel <- ps_filt_fixed %>% 
  tax_transform("compositional", rank = "Order")
ps_filt_fixed_fam_rel <- ps_filt_fixed %>% 
  tax_transform("compositional", rank = "Family")
# ps_filt_fixed_gen_rel <- ps_filt_fixed %>% 
#   tax_transform("compositional", rank = "Genus")

print(ps_filt_fixed_phy_rel)
print(ps_filt_fixed_gen_rel)


summary_agg <- data.frame(Taxa = c(colnames(ps_filt_fixed@tax_table)[c(2:6)], "ASVs"),
                 number_of_taxa=c(dim(ps_filt_fixed_phy_rel@otu_table)[2], dim(ps_filt_fixed_class_rel@otu_table)[2], dim(ps_filt_fixed_ord_rel@otu_table)[2], dim(ps_filt_fixed_fam_rel@otu_table)[2], dim(ps_filt_fixed_gen_rel@otu_table)[2], dim(ps_filt_fixed@otu_table)[2]))
                          
print(summary_agg)


```


## Data exploration
```{r}
barplots_group <- ps_filt_fixed %>%
  ps_mutate(host_sex = as.factor(host_sex)) %>%
  ps_mutate(group = as.factor(group)) %>%
  ps_arrange(group, week) %>% 
  comp_barplot(group_by = "group", tax_level = "Phylum", 
               tax_transform_for_plot = "compositional", n_taxa = 11, 
               bar_width = 0.8, merge_other = FALSE, label = "week", 
               sample_order = "default")

barplots_group_wrapped <- wrap_plots(barplots_group) +
  plot_layout(guides = "collect", nrow=2)

groups_barplot <- barplots_group_wrapped&
  facet_wrap("host_subject_id", scales = "free_x")& 
  theme(axis.text.x = element_text(angle = 60, hjust = 1, size = 10), 
        axis.ticks.y = element_blank(), leg.end.title = element_blank(), 
        legend.position = "bottom", strip.text = element_text(size=12),
        legend.key.size = unit(0.4, "cm"),
        legend.text = element_text(size=10))

groups_barplot

ggsave("groups_barplot.png", groups_barplot, width = 10, height = 8, dpi = 300)
```


```{r}

barplots_genus_group <- ps_filt_fixed %>%
  ps_mutate(host_sex = as.factor(host_sex)) %>%
  ps_mutate(group = as.factor(group)) %>%
  ps_arrange(group, week) %>% 
  comp_barplot(group_by = "group", tax_level = "Genus", 
               tax_transform_for_plot = "compositional", n_taxa = 11, 
               bar_width = 0.8, merge_other = FALSE, label = "week", 
               sample_order = "default")

barplots_genus_group_wrapped <- wrap_plots(barplots_genus_group) +
  plot_layout(guides = "collect", nrow=2)

groups_genus_barplot <- barplots_genus_group_wrapped&
  facet_wrap("host_subject_id", scales = "free_x")& 
  theme(axis.text.x = element_text(angle = 60, hjust = 1, size = 10), 
        axis.ticks.y = element_blank(), leg.end.title = element_blank(), 
        legend.position = "bottom", strip.text = element_text(size=12),
        legend.key.size = unit(0.4, "cm"),
        legend.text = element_text(size=10))

groups_genus_barplot

ggsave("groups_genus_barplot.png", groups_genus_barplot, width = 10, height = 8, dpi = 300)

```


```{r}
# Define a function to create bar plots for each group separately
create_group_barplot <- function(ps_obj, group_name) {
  ps_subset <- ps_obj %>% ps_filter(group == group_name)
  
  barplot <- ps_subset %>%
    ps_mutate(host_sex = as.factor(host_sex)) %>%
    ps_mutate(group = as.factor(group)) %>%
    ps_arrange(group, week) %>%
    comp_barplot(group_by = "group", tax_level = "Phylum",
                 tax_transform_for_plot = "compositional", n_taxa = 11,
                 bar_width = 0.8, merge_other = FALSE, label = "week",
                 sample_order = "default")
  
  wrapped_plot <- wrap_plots(barplot) +
    plot_layout(guides = "collect") &
    facet_wrap("host_subject_id", scales = "free_x") &
    theme(axis.text.x = element_text(angle = 60, hjust = 1, size = 10),
          axis.ticks.y = element_blank(), legend.title = element_blank(),
          legend.position = "bottom", strip.text = element_text(size = 12),
          legend.key.size = unit(0.4, "cm"),
          legend.text = element_text(size = 10))
  
  return(wrapped_plot)
}

# Create plots for each group
plot_1 <- create_group_barplot(ps_filt_fixed, "active-active")
plot_2 <- create_group_barplot(ps_filt_fixed, "active-remission")
plot_3 <- create_group_barplot(ps_filt_fixed, "remission-remission")
plot_4 <- create_group_barplot(ps_filt_fixed, "remission-active")
plot_5 <- create_group_barplot(ps_filt_fixed, "healthy-healthy")

# Save plots as separate images
ggsave("ac-ac_barplot.png", plot_1, width = 10, height = 8, dpi = 300)
ggsave("ac-rem_barplot.png", plot_2, width = 10, height = 8, dpi = 300)
ggsave("rem-rem_barplot.png", plot_3, width = 10, height = 8, dpi = 300)
ggsave("rem-ac_barplot.png", plot_4, width = 10, height = 8, dpi = 300)
ggsave("healthy_barplot.png", plot_5, width = 10, height = 8, dpi = 300)

# If you want to visualize them:
plot_1
plot_2
plot_3
plot_4
plot_5


```
## Same analysis but with relative abundance
```{r}
ps_ff_rel_abund <- phyloseq::transform_sample_counts(ps_filt_fixed, function(x){x/sum(x)})


# Define a function to create bar plots for each group separately
create_group_barplot <- function(ps_obj, group_name) {
  ps_subset <- ps_obj %>% ps_filter(group == group_name)
  
  barplot <- ps_subset %>%
    ps_mutate(host_sex = as.factor(host_sex)) %>%
    ps_mutate(group = as.factor(group)) %>%
    ps_arrange(group, week) %>%
    comp_barplot(group_by = "group", tax_level = "Phylum",
                 tax_transform_for_plot = "compositional", n_taxa = 11,
                 bar_width = 0.8, merge_other = FALSE, label = "week",
                 sample_order = "default")
  
  wrapped_plot <- wrap_plots(barplot) +
    plot_layout(guides = "collect") &
    facet_wrap("host_subject_id", scales = "free_x") &
    theme(axis.text.x = element_text(angle = 60, hjust = 1, size = 10),
          axis.ticks.y = element_blank(), legend.title = element_blank(),
          legend.position = "bottom", strip.text = element_text(size = 12),
          legend.key.size = unit(0.4, "cm"),
          legend.text = element_text(size = 10))
  
  return(wrapped_plot)
}

# Create plots for each group
plot_1_relabund <- create_group_barplot(ps_ff_rel_abund, "active-active")
plot_2_relabund <- create_group_barplot(ps_ff_rel_abund, "active-remission")
plot_3_relabund <- create_group_barplot(ps_ff_rel_abund, "remission-remission")
plot_4_relabund <- create_group_barplot(ps_ff_rel_abund, "remission-active")
plot_5_relabund <- create_group_barplot(ps_ff_rel_abund, "healthy-healthy")

# Save plots as separate images
ggsave("ac-ac_relabund_barplot.png", plot_1_relabund, width = 10, height = 8, dpi = 300)
ggsave("ac-rem_relabund_barplot.png", plot_2_relabund, width = 10, height = 8, dpi = 300)
ggsave("rem-rem_relabund_barplot.png", plot_3_relabund, width = 10, height = 8, dpi = 300)
ggsave("rem-ac_relabund_barplot.png", plot_4_relabund, width = 10, height = 8, dpi = 300)
ggsave("healthy_relabund_barplot.png", plot_5_relabund, width = 10, height = 8, dpi = 300)

# If you want to visualize them:
plot_1_relabund
plot_2_relabund
plot_3_relabund
plot_4_relabund
plot_5_relabund


```
## Boxplot of phylum_agg data by groups
```{r}
ps_ff_phylum_agg <- tax_glom(ps_filt_fixed, "Phylum")
taxa_names(ps_ff_phylum_agg) <- tax_table(ps_ff_phylum_agg)[, "Phylum"]

phylum_agg_boxplot <- psmelt(ps_ff_phylum_agg) %>% 
  ggplot(data = ., aes(x = group, y = Abundance)) +
  geom_boxplot(outlier.shape = NA) +
  geom_jitter(aes(color = OTU), height = 0, width = .2) +
  labs(x = "", y = "Abundance\n") +
  facet_wrap(~ OTU, scales = "free") +
  scale_x_discrete(guide = guide_axis(n.dodge = 2))

ggsave("Results/phylum_agg_boxplot.png", phylum_agg_boxplot, width = 13, height = 8, dpi = 300)

```
# Do the same by end status


```{r}
# Define a function to create bar plots for each group separately
create_genus_group_barplot <- function(ps_obj, group_name) {
  ps_subset <- ps_obj %>% ps_filter(group == group_name)
  
  barplot <- ps_subset %>%
    ps_mutate(host_sex = as.factor(host_sex)) %>%
    ps_mutate(group = as.factor(group)) %>%
    ps_arrange(group, week) %>%
    comp_barplot(group_by = "group", tax_level = "Genus",
                 tax_transform_for_plot = "compositional", n_taxa = 11,
                 bar_width = 0.8, merge_other = FALSE, label = "week",
                 sample_order = "default")
  
  wrapped_plot <- wrap_plots(barplot) +
    plot_layout(guides = "collect") &
    facet_wrap("host_subject_id", scales = "free_x") &
    theme(axis.text.x = element_text(angle = 60, hjust = 1, size = 10),
          axis.ticks.y = element_blank(), legend.title = element_blank(),
          legend.position = "bottom", strip.text = element_text(size = 12),
          legend.key.size = unit(0.4, "cm"),
          legend.text = element_text(size = 10))
  
  return(wrapped_plot)
}

# Create plots for each group
plot_genus_1 <- create_genus_group_barplot(ps_filt_fixed, "active-active")
plot_genus_2 <- create_genus_group_barplot(ps_filt_fixed, "active-remission")
plot_genus_3 <- create_genus_group_barplot(ps_filt_fixed, "remission-remission")
plot_genus_4 <- create_genus_group_barplot(ps_filt_fixed, "remission-active")
plot_genus_5 <- create_genus_group_barplot(ps_filt_fixed, "healthy-healthy")

# Save plots as separate images
ggsave("ac-ac_genus_barplot.png", plot_genus_1, width = 10, height = 8, dpi = 300)
ggsave("ac-rem_genus_barplot.png", plot_genus_2, width = 10, height = 8, dpi = 300)
ggsave("rem-rem_genus_barplot.png", plot_genus_3, width = 10, height = 8, dpi = 300)
ggsave("rem-ac_genus_barplot.png", plot_genus_4, width = 10, height = 8, dpi = 300)
ggsave("healthy_genus_barplot.png", plot_genus_5, width = 10, height = 8, dpi = 300)

# If you want to visualize them:
plot_genus_1
plot_genus_2
plot_genus_3
plot_genus_4
plot_genus_5


```

```{r}
create_group_barplots <- function(ps_obj, group_name, max_per_plot = 6, output_prefix = "plot") {
  ps_subset <- ps_obj %>% ps_filter(group == group_name)
  
  # Get unique subjects
  subject_ids <- unique(sample_data(ps_subset)$host_subject_id)
  
  # Split into chunks of 'max_per_plot'
  subject_batches <- split(subject_ids, ceiling(seq_along(subject_ids) / max_per_plot))
  
  # Generate plots for each batch
  plots <- list()
  
  for (i in seq_along(subject_batches)) {
    subjects <- subject_batches[[i]]
    
    ps_batch <- ps_subset %>% ps_filter(host_subject_id %in% subjects)
    
    barplot <- ps_batch %>%
      ps_mutate(host_sex = as.factor(host_sex)) %>%
      ps_mutate(group = as.factor(group)) %>%
      ps_arrange(group, week) %>%
      comp_barplot(group_by = "group", tax_level = "Phylum",
                   tax_transform_for_plot = "compositional", n_taxa = 11,
                   bar_width = 0.8, merge_other = FALSE, label = "week",
                   sample_order = "default")
    
    wrapped_plot <- wrap_plots(barplot) +
      plot_layout(guides = "collect") &
      facet_wrap("host_subject_id", scales = "free_x") &
      theme(axis.text.x = element_text(angle = 60, hjust = 1, size = 10),
            axis.ticks.y = element_blank(), legend.title = element_blank(),
            legend.position = "bottom", strip.text = element_text(size = 12),
            legend.key.size = unit(0.4, "cm"),
            legend.text = element_text(size = 10))
    
    plots[[i]] <- wrapped_plot
    
    # Save each plot separately
    ggsave(filename = paste0(output_prefix, "_batch_", i, ".png"), plot = wrapped_plot, 
           width = 10, height = 8, dpi = 300)
  }
  
  return(plots)
}

# Generate and save plots
plots_rem_rem <- create_group_barplots(ps_filt_fixed, "remission-remission", max_per_plot = 6, output_prefix = "rem_rem_barplot")
plots_rem_ac <- create_group_barplots(ps_filt_fixed, "remission-active", max_per_plot = 6, output_prefix = "rem_active_barplot")
plots_ac_ac <- create_group_barplots(ps_filt_fixed, "active-active", max_per_plot = 6, output_prefix = "ac_ac_barplot")
plots_ac_rem <- create_group_barplots(ps_filt_fixed, "active-remission", max_per_plot = 6, output_prefix = "ac_rem_barplot")
plots_healthy <- create_group_barplots(ps_filt_fixed, "healthy-healthy", max_per_plot = 6, output_prefix = "healthy_barplot")

plots_ac_ac
plots_ac_rem
plots_rem_rem
plots_rem_ac
plots_healthy

```


```{r}
create_group_barplots <- function(ps_obj, group_name, max_per_plot = 6, output_prefix = "plot") {
  ps_subset <- ps_obj %>% ps_filter(group == group_name)
  
  # Get unique subjects
  subject_ids <- unique(sample_data(ps_subset)$host_subject_id)
  
  # Split into chunks of 'max_per_plot'
  subject_batches <- split(subject_ids, ceiling(seq_along(subject_ids) / max_per_plot))
  
  # Generate plots for each batch
  plots <- list()
  
  for (i in seq_along(subject_batches)) {
    subjects <- subject_batches[[i]]
    
    ps_batch <- ps_subset %>% ps_filter(host_subject_id %in% subjects)
    
    barplot <- ps_batch %>%
      ps_mutate(host_sex = as.factor(host_sex)) %>%
      ps_mutate(group = as.factor(group)) %>%
      ps_arrange(group, week) %>%
      comp_barplot(group_by = "group", tax_level = "Genus",
                   tax_transform_for_plot = "compositional", n_taxa = 11,
                   bar_width = 0.8, merge_other = FALSE, label = "week",
                   sample_order = "default")
    
    wrapped_plot <- wrap_plots(barplot) +
      plot_layout(guides = "collect") &
      facet_wrap("host_subject_id", scales = "free_x") &
      theme(axis.text.x = element_text(angle = 60, hjust = 1, size = 10),
            axis.ticks.y = element_blank(), legend.title = element_blank(),
            legend.position = "bottom", strip.text = element_text(size = 12),
            legend.key.size = unit(0.4, "cm"),
            legend.text = element_text(size = 10))
    
    plots[[i]] <- wrapped_plot
    
    # Save each plot separately
    ggsave(filename = paste0(output_prefix, "_genus_batch_", i, ".png"), plot = wrapped_plot, 
           width = 10, height = 8, dpi = 300)
  }
  
  return(plots)
}

# Generate and save plots
plots_genus_rem_rem <- create_group_barplots(ps_filt_fixed, "remission-remission", max_per_plot = 6, output_prefix = "rem_rem_barplot")
plots_genus_rem_ac <- create_group_barplots(ps_filt_fixed, "remission-active", max_per_plot = 6, output_prefix = "rem_active_barplot")
plots_genus_ac_ac <- create_group_barplots(ps_filt_fixed, "active-active", max_per_plot = 6, output_prefix = "ac_ac_barplot")
plots_genus_ac_rem <- create_group_barplots(ps_filt_fixed, "active-remission", max_per_plot = 6, output_prefix = "ac_rem_barplot")
plots_genus_healthy <- create_group_barplots(ps_filt_fixed, "healthy-healthy", max_per_plot = 6, output_prefix = "healthy_barplot")

plots_genus_ac_ac
plots_genus_ac_rem
plots_genus_rem_rem
plots_genus_rem_ac
plots_genus_healthy

```



## Alpha diversity
```{r}
## Not rarefied
alpha_div <- estimate_richness(ps_filt_fixed, measures = c("Observed", "Shannon", "InvSimpson", "Chao1"))
PD <- calculatePD(ps_filt_fixed, justDF = TRUE, include_root = FALSE) %>% pull(PD)
alpha_div <- alpha_div %>% mutate(PD = PD)
rownames(alpha_div) <- gsub("^X", "", rownames(alpha_div))
head(alpha_div, n=20)

metadata_samples_alpha <- bind_cols(meta_merged, alpha_div)
metadata_samples_long <- metadata_samples_alpha %>% pivot_longer(cols= Observed:PD, names_to = "alpha_div", values_to = "value")
kruskal_test_alpha <- metadata_samples_long %>% group_by(alpha_div) %>%
    rstatix::kruskal_test(value ~ group) %>% mutate(p_adj_BH = p.adjust(p, method = "BH"))

kruskal_test_alpha
write.csv(kruskal_test_alpha, "Results/kruskal_test_alpha.csv", row.names = FALSE)

```

```{r}
dunn_test_alpha <- metadata_samples_long %>% group_by(alpha_div) %>%
    rstatix::dunn_test(value ~ group, p.adjust.method = "BH") %>%
  add_xy_position(x = "group", fun = "mean_sd", step.increase = 0.05, scales = "free")
dunn_test_alpha %>% kbl() %>%
  kableExtra::kable_paper(font_size = 12)

dunn_test_alpha_table <- apply(dunn_test_alpha,2,as.character)
write.csv(dunn_test_alpha_table, "Results/dunn_test_alpha_table.csv", row.names = FALSE)


```

```{r}

dunn_test_alpha_plot <- ggplot(metadata_samples_long, aes(x = group, y = value)) +
  geom_boxplot() +
  geom_jitter(shape = 21, color = "black", size = 2, aes(fill = group)) +
  facet_wrap(~alpha_div, scales = "free", ncol = 3) +
  scale_fill_nejm() +
  stat_pvalue_manual(data = dunn_test_alpha, 
                     label = "p.adj.signif", tip.length = 0.01, bracket.shorten = 0.05,
                     hide.ns = TRUE, size = 4) +
  scale_x_discrete(guide = guide_axis(n.dodge = 2)) +
  theme_bw() +
  theme(axis.text.x = element_text(size = 10), legend.position = "none")

ggsave("Results/dunn_test_alpha_plot.png", dunn_test_alpha_plot, width = 13, height = 8, dpi = 300)

dunn_test_alpha_plot

```
```{r}

metadata_samples_alpha

kruskal_chao1 <- kruskal.test(Chao1 ~ group, data = metadata_samples_alpha)
kruskal_shannon <- kruskal.test(Shannon ~ group, data = metadata_samples_alpha)
kruskal_invsimps <- kruskal.test(InvSimpson ~ group, data = metadata_samples_alpha)
kruskal_obsv <- kruskal.test(Observed ~ group, data = metadata_samples_alpha)
kruskal_pd <- kruskal.test(PD ~ group, data = metadata_samples_alpha)


# Extract values
H_chao1 <- kruskal_chao1$statistic
p_chao1 <- kruskal_chao1$p.value

H_shannon <- kruskal_shannon$statistic
p_shannon <- kruskal_shannon$p.value

H_invsimps <- kruskal_invsimps$statistic
p_invsimps <- kruskal_invsimps$p.value

H_obsv <- kruskal_obsv$statistic
p_obsv <- kruskal_obsv$p.value

H_pd <- kruskal_pd$statistic
p_pd <- kruskal_pd$p.value

# Print results
cat("Kruskal-Wallis test for Chao1 index: H =", round(H_chao1, 3), "p =", round(p_chao1, 3), "\n")
cat("Kruskal-Wallis test for Shannon index: H =", round(H_shannon, 3), "p =", round(p_shannon, 3), "\n")
cat("Kruskal-Wallis test for Inverse Simpson index: H =", round(H_invsimps, 3), "p =", round(p_invsimps, 3), "\n")
cat("Kruskal-Wallis test for Observed Number of Species: H =", round(H_obsv, 3), "p =", round(p_obsv, 3), "\n")
cat("Kruskal-Wallis test for Faith's Phylogenetic Diversity: H =", round(H_pd, 3), "p =", round(p_pd, 3), "\n")


# Filter for Chao1 and Shannon indices
dunn_chao1 <- dunn_test_alpha %>% filter(alpha_div == "Chao1")
dunn_shannon <- dunn_test_alpha %>% filter(alpha_div == "Shannon")
dunn_invsimps <- dunn_test_alpha %>% filter(alpha_div == "InvSimpson")
dunn_obsv <- dunn_test_alpha %>% filter(alpha_div == "Observed")
dunn_pd <- dunn_test_alpha %>% filter(alpha_div == "PD")

# Show significant results (p < 0.05)
significant_chao1 <- dunn_chao1 %>% filter(p.adj < 0.05)
significant_shannon <- dunn_shannon %>% filter(p.adj < 0.05)
significant_invsimps <- dunn_invsimps %>% filter(p.adj < 0.05)
significant_obsv <- dunn_obsv %>% filter(p.adj < 0.05)
significant_pd <- dunn_pd %>% filter(p.adj < 0.05)

# Print significant comparisons
print(significant_chao1)
print(significant_shannon)
print(significant_invsimps)
print(significant_obsv)
print(significant_pd)

sig_shannon_pairwise <- as.matrix(significant_shannon)
sig_invsimps_pairwise <- as.matrix(significant_invsimps)
sig_obsv_pairwise <- as.matrix(significant_obsv)


write.csv(sig_shannon_pairwise, "Results/sig_shannon_pairwise_table.csv", row.names = FALSE)
write.csv(sig_invsimps_pairwise, "Results/sig_invsimps_pairwise_table.csv", row.names = FALSE)


alpha_summary <- metadata_samples_alpha %>%
  group_by(group) %>%
  summarise(
    Chao1_mean = mean(Chao1, na.rm = TRUE),
    Chao1_sd = sd(Chao1, na.rm = TRUE),
    Shannon_mean = mean(Shannon, na.rm = TRUE),
    Shannon_sd = sd(Shannon, na.rm = TRUE),
    InvSimps_mean = mean(InvSimpson, na.rm = TRUE),
    InvSimps_sd = sd(InvSimpson, na.rm = TRUE), 
    Obsv_mean = mean(Observed, na.rm = TRUE),
    Obsv_sd = sd(Observed, na.rm = TRUE),
    PD_mean = mean(PD, na.rm = TRUE),
    PD_sd = sd(PD, na.rm = TRUE)
  )

print(alpha_summary)

write.csv(alpha_summary, "Results/alpha_summary_table.csv", row.names = FALSE)

```


## Rarefied

```{r}
library(ranacapa); packageVersion("ranacapa")

ps_filt_fixed %>% 
  ggrare(step = 50, color = "group", se = FALSE)

```


```{r}
set.seed(19950622)

ps_ff_rare <- rarefy_even_depth(ps_filt_fixed, sample.size = 10558, replace = FALSE)

ps_ff_rare %>% 
  ggrare(step = 50, color = "group", se = FALSE)

alpha_div_rare <- estimate_richness(ps_ff_rare, measures = c("Observed", "Shannon", "InvSimpson", "Chao1"))
PD_rare <- calculatePD(ps_ff_rare, justDF = TRUE, include_root = FALSE) %>% pull(PD)
alpha_div_rare <- alpha_div_rare %>% mutate(PD = PD_rare)
rownames(alpha_div_rare) <- gsub("^X", "", rownames(alpha_div_rare))
head(alpha_div_rare, n=20)

meta_rare <- as.data.frame(as.matrix(sample_data(ps_ff_rare)))

metadata_samples_alpha_rare <- bind_cols(meta_rare, alpha_div_rare)
metadata_samples_long_rare <- metadata_samples_alpha_rare %>% pivot_longer(cols= Observed:PD, names_to = "alpha_div", values_to = "value")
kruskal_test_alpha_rare <- metadata_samples_long_rare %>% group_by(alpha_div) %>%
    rstatix::kruskal_test(value ~ group) %>% mutate(p_adj_BH = p.adjust(p, method = "BH"))

kruskal_test_alpha_rare

write.csv(kruskal_test_alpha_rare, "Results/kruskal_test_alpha_rare.csv", row.names = FALSE)

dunn_test_alpha_rare <- metadata_samples_long_rare %>% group_by(alpha_div) %>%
    rstatix::dunn_test(value ~ group, p.adjust.method = "BH") %>%
    add_xy_position(x = "group", fun = "mean_sd", step.increase = 0.05, scales = "free")
dunn_test_alpha_rare %>% kbl() %>%
  kableExtra::kable_paper(font_size = 12)

dunn_test_alpha_rare_table <- apply(dunn_test_alpha_rare,2,as.character)
write.csv(dunn_test_alpha_rare_table, "Results/dunn_test_alpha_rare_table.csv", row.names = FALSE)

dunn_test_alpha_rare_plot <- ggplot(metadata_samples_long_rare, aes(x = group, y = value)) +
  geom_boxplot() +
  geom_jitter(shape = 21, color = "black", size = 2, aes(fill = group)) +
  facet_wrap(~alpha_div, scales = "free", ncol = 3) +
  scale_fill_nejm() +
  stat_pvalue_manual(data = dunn_test_alpha_rare, 
                     label = "p.adj.signif", tip.length = 0.01, bracket.shorten = 0.05,
                     hide.ns = TRUE, size = 4) +
  scale_x_discrete(guide = guide_axis(n.dodge = 2)) +
  theme_bw() +
  theme(axis.text.x = element_text(size = 10), legend.position = "none")

ggsave("Results/dunn_test_alpha_rare_plot.png", dunn_test_alpha_rare_plot, width = 13, height = 8, dpi = 300)

dunn_test_alpha_rare_plot

```
```{r}

metadata_samples_alpha_rare

kruskal_chao1_rare <- kruskal.test(Chao1 ~ group, data = metadata_samples_alpha_rare)
kruskal_shannon_rare <- kruskal.test(Shannon ~ group, data = metadata_samples_alpha_rare)
kruskal_invsimps_rare <- kruskal.test(InvSimpson ~ group, data = metadata_samples_alpha_rare)
kruskal_obsv_rare <- kruskal.test(Observed ~ group, data = metadata_samples_alpha_rare)
kruskal_pd_rare <- kruskal.test(PD ~ group, data = metadata_samples_alpha_rare)


# Extract values
H_chao1_rare <- kruskal_chao1_rare$statistic
p_chao1_rare <- kruskal_chao1_rare$p.value

H_shannon_rare <- kruskal_shannon_rare$statistic
p_shannon_rare <- kruskal_shannon_rare$p.value

H_invsimps_rare <- kruskal_invsimps_rare$statistic
p_invsimps_rare <- kruskal_invsimps_rare$p.value

H_obsv_rare <- kruskal_obsv_rare$statistic
p_obsv_rare <- kruskal_obsv_rare$p.value

H_pd_rare <- kruskal_pd_rare$statistic
p_pd_rare <- kruskal_pd_rare$p.value

# Print results
cat("Kruskal-Wallis test for Chao1 index: H =", round(H_chao1_rare, 3), "p =", round(p_chao1_rare, 3), "\n")
cat("Kruskal-Wallis test for Shannon index: H =", round(H_shannon_rare, 3), "p =", round(p_shannon_rare, 3), "\n")
cat("Kruskal-Wallis test for Inverse Simpson index: H =", round(H_invsimps_rare, 3), "p =", round(p_invsimps_rare, 3), "\n")
cat("Kruskal-Wallis test for Observed Number of Species: H =", round(H_obsv_rare, 3), "p =", round(p_obsv_rare, 3), "\n")
cat("Kruskal-Wallis test for Faith's Phylogenetic Diversity: H =", round(H_pd_rare, 3), "p =", round(p_pd_rare, 3), "\n")


# Filter for Chao1 and Shannon indices
dunn_chao1_rare <- dunn_test_alpha_rare %>% filter(alpha_div == "Chao1")
dunn_shannon_rare <- dunn_test_alpha_rare %>% filter(alpha_div == "Shannon")
dunn_invsimps_rare <- dunn_test_alpha_rare %>% filter(alpha_div == "InvSimpson")
dunn_obsv_rare <- dunn_test_alpha_rare %>% filter(alpha_div == "Observed")
dunn_pd_rare <- dunn_test_alpha_rare %>% filter(alpha_div == "PD")

# Show significant results (p < 0.05)
significant_chao1_rare <- dunn_chao1_rare %>% filter(p.adj < 0.05)
significant_shannon_rare <- dunn_shannon_rare %>% filter(p.adj < 0.05)
significant_invsimps_rare <- dunn_invsimps_rare %>% filter(p.adj < 0.05)
significant_obsv_rare <- dunn_obsv_rare %>% filter(p.adj < 0.05)
significant_pd_rare <- dunn_pd_rare %>% filter(p.adj < 0.05)

# Print significant comparisons
print(significant_chao1_rare)
print(significant_shannon_rare)
print(significant_invsimps_rare)
print(significant_obsv_rare)
print(significant_pd_rare)

alpha_rare_summary <- metadata_samples_alpha_rare %>%
  group_by(group) %>%
  summarise(
    Chao1_mean = mean(Chao1, na.rm = TRUE),
    Chao1_sd = sd(Chao1, na.rm = TRUE),
    Shannon_mean = mean(Shannon, na.rm = TRUE),
    Shannon_sd = sd(Shannon, na.rm = TRUE),
    InvSimps_mean = mean(InvSimpson, na.rm = TRUE),
    InvSimps_sd = sd(InvSimpson, na.rm = TRUE), 
    Obsv_mean = mean(Observed, na.rm = TRUE),
    Obsv_sd = sd(Observed, na.rm = TRUE),
    PD_mean = mean(PD, na.rm = TRUE),
    PD_sd = sd(PD, na.rm = TRUE)
  )

print(alpha_rare_summary)
write.csv(alpha_rare_summary, "Results/alpha_rare_summary_table.csv", row.names = FALSE)

```

## alpha diversity metrics by initial or final timepoint
```{r}
# ps_filt_fixed_end

ps_filt_fixed_end <- ps_filt_fixed_end %>% ps_mutate(disease_status_end = as.factor(disease_status_end))

meta_end <- as.matrix(sample_data(ps_filt_fixed_end))

alpha_div_end <- estimate_richness(ps_filt_fixed_end, measures = c("Observed", "Shannon", "InvSimpson", "Chao1"))
PD_end <- calculatePD(ps_filt_fixed_end, justDF = TRUE, include_root = FALSE) %>% pull(PD)
alpha_div_end <- alpha_div_end %>% mutate(PD_end = PD_end)
rownames(alpha_div_end) <- gsub("^X", "", rownames(alpha_div_end))
head(alpha_div_end, n=20)

metadata_samples_alpha_end <- bind_cols(meta_end, alpha_div_end)
metadata_samples_long_end <- metadata_samples_alpha_end %>% pivot_longer(cols= Observed:PD_end, names_to = "alpha_div_end", values_to = "value")
kruskal_test_alpha_end <- metadata_samples_long_end %>% group_by(alpha_div_end) %>%
  rstatix::kruskal_test(value ~ disease_status_end) %>% mutate(p_adj_BH = p.adjust(p, method = "BH"))

kruskal_test_alpha_end
write.csv(kruskal_test_alpha_end, "Results/kruskal_test_alpha_end.csv", row.names = FALSE)

```

```{r}
dunn_test_alpha_end <- metadata_samples_long_end %>% group_by(alpha_div_end) %>%
  rstatix::dunn_test(value ~ disease_status_end, p.adjust.method = "BH") %>%
  add_xy_position(x = "disease_status_end", fun = "mean_sd", step.increase = 0.05, scales = "free")
dunn_test_alpha_end %>% kbl() %>%
  kableExtra::kable_paper(font_size = 12)

dunn_test_alpha_end_table <- apply(dunn_test_alpha_end,2,as.character)
write.csv(dunn_test_alpha_end_table, "Results/dunn_test_alpha_end_table.csv", row.names = FALSE)


```

```{r}

dunn_test_alpha_end_plot <- ggplot(metadata_samples_long_end, aes(x = disease_status_end, y = value)) +
  geom_boxplot() +
  geom_jitter(shape = 21, color = "black", size = 2, aes(fill = disease_status_end)) +
  facet_wrap(~alpha_div_end, scales = "free", ncol = 3) +
  scale_fill_nejm() +
  stat_pvalue_manual(data = dunn_test_alpha_end, 
                     label = "p.adj.signif", tip.length = 0.01, bracket.shorten = 0.05,
                     hide.ns = TRUE, size = 4) +
  scale_x_discrete(guide = guide_axis(n.dodge = 2)) +
  theme_bw() +
  theme(axis.text.x = element_text(size = 10), legend.position = "none")

ggsave("Results/dunn_test_alpha_end_plot.png", dunn_test_alpha_end_plot, width = 13, height = 8, dpi = 300)

dunn_test_alpha_end_plot

```

```{r}

metadata_samples_alpha_end

kruskal_chao1_end <- kruskal.test(Chao1 ~ disease_status_end, data = metadata_samples_alpha_end)
kruskal_shannon_end <- kruskal.test(Shannon ~ disease_status_end, data = metadata_samples_alpha_end)
kruskal_invsimps_end <- kruskal.test(InvSimpson ~ disease_status_end, data = metadata_samples_alpha_end)
kruskal_obsv_end <- kruskal.test(Observed ~ disease_status_end, data = metadata_samples_alpha_end)
kruskal_pd_end <- kruskal.test(PD_end ~ disease_status_end, data = metadata_samples_alpha_end)


# Extract values
H_chao1_end <- kruskal_chao1_end$statistic
p_chao1_end <- kruskal_chao1_end$p.value

H_shannon_end <- kruskal_shannon_end$statistic
p_shannon_end <- kruskal_shannon_end$p.value

H_invsimps_end <- kruskal_invsimps_end$statistic
p_invsimps_end <- kruskal_invsimps_end$p.value

H_obsv_end <- kruskal_obsv_end$statistic
p_obsv_end <- kruskal_obsv_end$p.value

H_pd_end <- kruskal_pd_end$statistic
p_pd_end <- kruskal_pd_end$p.value

# Print results
cat("Kruskal-Wallis test for Chao1 index: H =", round(H_chao1_end, 3), "p =", round(p_chao1_end, 3), "\n")
cat("Kruskal-Wallis test for Shannon index: H =", round(H_shannon_end, 3), "p =", round(p_shannon_end, 3), "\n")
cat("Kruskal-Wallis test for Inverse Simpson index: H =", round(H_invsimps_end, 3), "p =", round(p_invsimps_end, 3), "\n")
cat("Kruskal-Wallis test for Observed Number of Species: H =", round(H_obsv_end, 3), "p =", round(p_obsv_end, 3), "\n")
cat("Kruskal-Wallis test for Faith's Phylogenetic Diversity: H =", round(H_pd_end, 3), "p =", round(p_pd_end, 3), "\n")


# Filter for Chao1 and Shannon indices
dunn_chao1_end <- dunn_test_alpha_end %>% filter(alpha_div_end == "Chao1")
dunn_shannon_end <- dunn_test_alpha_end %>% filter(alpha_div_end == "Shannon")
dunn_invsimps_end <- dunn_test_alpha_end %>% filter(alpha_div_end == "InvSimpson")
dunn_obsv_end <- dunn_test_alpha_end %>% filter(alpha_div_end == "Observed")
dunn_pd_end <- dunn_test_alpha_end %>% filter(alpha_div_end == "PD_end")

# Show significant results (p < 0.05)
significant_chao1_end <- dunn_chao1_end %>% filter(p.adj < 0.05)
significant_shannon_end <- dunn_shannon_end %>% filter(p.adj < 0.05)
significant_invsimps_end <- dunn_invsimps_end %>% filter(p.adj < 0.05)
significant_obsv_end <- dunn_obsv_end %>% filter(p.adj < 0.05)
significant_pd_end <- dunn_pd_end %>% filter(p.adj < 0.05)

# Print significant comparisons
print(significant_chao1_end)
print(significant_shannon_end)
print(significant_invsimps_end)
print(significant_obsv_end)
print(significant_pd_end)

sig_shannon_pairwise_end <- as.matrix(significant_shannon_end)
sig_invsimps_pairwise_end <- as.matrix(significant_invsimps_end)


write.csv(sig_shannon_pairwise_end, "Results/sig_shannon_pairwise_end_table.csv", row.names = FALSE)
write.csv(sig_invsimps_pairwise_end, "Results/sig_invsimps_pairwise_end_table.csv", row.names = FALSE)


alpha_summary_end <- metadata_samples_alpha_end %>%
  group_by(disease_status_end) %>%
  summarise(
    Chao1_mean = mean(Chao1, na.rm = TRUE),
    Chao1_sd = sd(Chao1, na.rm = TRUE),
    Shannon_mean = mean(Shannon, na.rm = TRUE),
    Shannon_sd = sd(Shannon, na.rm = TRUE),
    InvSimps_mean = mean(InvSimpson, na.rm = TRUE),
    InvSimps_sd = sd(InvSimpson, na.rm = TRUE), 
    Obsv_mean = mean(Observed, na.rm = TRUE),
    Obsv_sd = sd(Observed, na.rm = TRUE),
    PD_mean = mean(PD_end, na.rm = TRUE),
    PD_sd = sd(PD_end, na.rm = TRUE)
  )

print(alpha_summary_end)

write.csv(alpha_summary_end, "Results/alpha_summary_end_table.csv", row.names = FALSE)

```
## Same for baseline samples
```{r} 
ps_filt_fixed_base <- ps_filt_fixed %>% ps_filter(!is.na(baseline_status))

ps_filt_fixed_base <- ps_filt_fixed_base %>% ps_mutate(baseline_status = as.factor(baseline_status))

meta_base <- as.matrix(sample_data(ps_filt_fixed_base))

alpha_div_base <- estimate_richness(ps_filt_fixed_base, measures = c("Observed", "Shannon", "InvSimpson", "Chao1"))
PD_base <- calculatePD(ps_filt_fixed_base, justDF = TRUE, include_root = FALSE) %>% pull(PD)
alpha_div_base <- alpha_div_base %>% mutate(PD_base = PD_base)
rownames(alpha_div_base) <- gsub("^X", "", rownames(alpha_div_base))
head(alpha_div_base, n=20)

metadata_samples_alpha_base <- bind_cols(meta_base, alpha_div_base)
metadata_samples_long_base <- metadata_samples_alpha_base %>% pivot_longer(cols= Observed:PD_base, names_to = "alpha_div_base", values_to = "value")
kruskal_test_alpha_base <- metadata_samples_long_base %>% group_by(alpha_div_base) %>%
  rstatix::kruskal_test(value ~ baseline_status) %>% mutate(p_adj_BH = p.adjust(p, method = "BH"))

kruskal_test_alpha_base
write.csv(kruskal_test_alpha_base, "Results/kruskal_test_alpha_base.csv", row.names = FALSE)

```

```{r}
dunn_test_alpha_base <- metadata_samples_long_base %>% group_by(alpha_div_base) %>%
  rstatix::dunn_test(value ~ baseline_status, p.adjust.method = "BH") %>%
  add_xy_position(x = "baseline_status", fun = "mean_sd", step.increase = 0.05, scales = "free")
dunn_test_alpha_base %>% kbl() %>%
  kableExtra::kable_paper(font_size = 12)

dunn_test_alpha_base_table <- apply(dunn_test_alpha_base,2,as.character)
write.csv(dunn_test_alpha_base_table, "Results/dunn_test_alpha_base_table.csv", row.names = FALSE)


```

```{r}

dunn_test_alpha_base_plot <- ggplot(metadata_samples_long_base, aes(x = baseline_status, y = value)) +
  geom_boxplot() +
  geom_jitter(shape = 21, color = "black", size = 2, aes(fill = baseline_status)) +
  facet_wrap(~alpha_div_base, scales = "free", ncol = 3) +
  scale_fill_nejm() +
  stat_pvalue_manual(data = dunn_test_alpha_base, 
                     label = "p.adj.signif", tip.length = 0.01, bracket.shorten = 0.05,
                     hide.ns = TRUE, size = 4) +
  scale_x_discrete(guide = guide_axis(n.dodge = 2)) +
  theme_bw() +
  theme(axis.text.x = element_text(size = 10), legend.position = "none")

ggsave("Results/dunn_test_alpha_base_plot.png", dunn_test_alpha_base_plot, width = 13, height = 8, dpi = 300)

dunn_test_alpha_base_plot

```


```{r}

metadata_samples_alpha_base

kruskal_chao1_base <- kruskal.test(Chao1 ~ baseline_status, data = metadata_samples_alpha_base)
kruskal_shannon_base <- kruskal.test(Shannon ~ baseline_status, data = metadata_samples_alpha_base)
kruskal_invsimps_base <- kruskal.test(InvSimpson ~ baseline_status, data = metadata_samples_alpha_base)
kruskal_obsv_base <- kruskal.test(Observed ~ baseline_status, data = metadata_samples_alpha_base)
kruskal_pd_base <- kruskal.test(PD_base ~ baseline_status, data = metadata_samples_alpha_base)


# Extract values
H_chao1_base <- kruskal_chao1_base$statistic
p_chao1_base <- kruskal_chao1_base$p.value

H_shannon_base <- kruskal_shannon_base$statistic
p_shannon_base <- kruskal_shannon_base$p.value

H_invsimps_base <- kruskal_invsimps_base$statistic
p_invsimps_base <- kruskal_invsimps_base$p.value

H_obsv_base <- kruskal_obsv_base$statistic
p_obsv_base <- kruskal_obsv_base$p.value

H_pd_base <- kruskal_pd_base$statistic
p_pd_base <- kruskal_pd_base$p.value

# Print results
cat("Kruskal-Wallis test for Chao1 index: H =", round(H_chao1_base, 3), "p =", round(p_chao1_base, 3), "\n")
cat("Kruskal-Wallis test for Shannon index: H =", round(H_shannon_base, 3), "p =", round(p_shannon_base, 3), "\n")
cat("Kruskal-Wallis test for Inverse Simpson index: H =", round(H_invsimps_base, 3), "p =", round(p_invsimps_base, 3), "\n")
cat("Kruskal-Wallis test for Observed Number of Species: H =", round(H_obsv_base, 3), "p =", round(p_obsv_base, 3), "\n")
cat("Kruskal-Wallis test for Faith's Phylogenetic Diversity: H =", round(H_pd_base, 3), "p =", round(p_pd_base, 3), "\n")


# Filter for Chao1 and Shannon indices
dunn_chao1_base <- dunn_test_alpha_base %>% filter(alpha_div_base == "Chao1")
dunn_shannon_base <- dunn_test_alpha_base %>% filter(alpha_div_base == "Shannon")
dunn_invsimps_base <- dunn_test_alpha_base %>% filter(alpha_div_base == "InvSimpson")
dunn_obsv_base <- dunn_test_alpha_base %>% filter(alpha_div_base == "Observed")
dunn_pd_base <- dunn_test_alpha_base %>% filter(alpha_div_base == "PD_base")

# Show significant results (p < 0.05)
significant_chao1_base <- dunn_chao1_base %>% filter(p.adj < 0.05)
significant_shannon_base <- dunn_shannon_base %>% filter(p.adj < 0.05)
significant_invsimps_base <- dunn_invsimps_base %>% filter(p.adj < 0.05)
significant_obsv_base <- dunn_obsv_base %>% filter(p.adj < 0.05)
significant_pd_base <- dunn_pd_base %>% filter(p.adj < 0.05)

# Print significant comparisons
print(significant_chao1_base)
print(significant_shannon_base)
print(significant_invsimps_base)
print(significant_obsv_base)
print(significant_pd_base)

sig_shannon_pairwise_base <- as.matrix(significant_shannon_base)
sig_invsimps_pairwise_base <- as.matrix(significant_invsimps_base)


write.csv(sig_shannon_pairwise_base, "Results/sig_shannon_pairwise_base_table.csv", row.names = FALSE)
write.csv(sig_invsimps_pairwise_base, "Results/sig_invsimps_pairwise_base_table.csv", row.names = FALSE)


alpha_summary_base <- metadata_samples_alpha_base %>%
  group_by(baseline_status) %>%
  summarise(
    Chao1_mean = mean(Chao1, na.rm = TRUE),
    Chao1_sd = sd(Chao1, na.rm = TRUE),
    Shannon_mean = mean(Shannon, na.rm = TRUE),
    Shannon_sd = sd(Shannon, na.rm = TRUE),
    InvSimps_mean = mean(InvSimpson, na.rm = TRUE),
    InvSimps_sd = sd(InvSimpson, na.rm = TRUE), 
    Obsv_mean = mean(Observed, na.rm = TRUE),
    Obsv_sd = sd(Observed, na.rm = TRUE),
    PD_mean = mean(PD_base, na.rm = TRUE),
    PD_sd = sd(PD_base, na.rm = TRUE)
  )

print(alpha_summary_base)

write.csv(alpha_summary_base, "Results/alpha_summary_base_table.csv", row.names = FALSE)

```

## Beta diversity
```{r}
PCoA_ASV_bray <- ordinate(ps_filt_fixed_rel, distance = "bray", method = "PCoA")
PCoA_ASV_bray


beta_ASV_bray <- phyloseq::distance(ps_filt_fixed_rel, "bray")
ado_bray_group <- adonis2(beta_ASV_bray ~ group, meta_merged)
print(ado_bray_group)

group.bd <- betadisper(d = beta_ASV_bray, group = meta_merged$group, type="centroid")
bd_group_boxplot <- boxplot(group.bd)

bd_group_boxplot

png("Results/beta_dispersion_bc_boxplot.png", width = 3500, height = 2000, res = 300)  # High resolution
boxplot(group.bd, main = "Beta Dispersion per Group (BC)", cex.axis = 1.2, cex.lab = 1.5)
dev.off()

permutest(group.bd)

```

```{r}

BC_pcoa_plot <- plot_ordination(ps_filt_fixed_rel, PCoA_ASV_bray, axes = c(1,2))+
  stat_conf_ellipse(geom = "polygon", alpha = 0.1, aes(color = group,  fill = group))+
  geom_point(size = 3, shape = 21, aes(fill = group))+
  stat_mean(aes(fill = group), shape = 23, size = 4) + 
  annotate("text", x =- 0.3, y = 0.5, color = "black", label = paste0("p=", round(ado_bray_group$`Pr(>F)`[1], digits = 3)))+
  annotate("text", x = -0.3, y = 0.45, color = "black", label = paste0("R\u00b2=", round((ado_bray_group$R2[1])*100, digits = 2), "%"))+
  scale_fill_nejm(name = "Group")+
  scale_color_nejm(name = "Group")+
  theme_bw()+
  theme(legend.position = "right")

ggsave("Results/BC_pcoa_plot.png", BC_pcoa_plot, width = 10, height = 8, dpi = 300)

BC_pcoa_plot

```

## Color by different variables
```{r}
## By bioproject

ps_filt_fixed_rel@sam_data$bioproject <- as.factor(ps_filt_fixed_rel@sam_data$bioproject)

ado_bray_biop <- adonis2(beta_ASV_bray ~ bioproject, meta_merged)
print(ado_bray_biop)

BC_pcoa_biop_plot <- plot_ordination(ps_filt_fixed_rel, PCoA_ASV_bray, axes = c(1,2))+
  stat_conf_ellipse(geom = "polygon", alpha = 0.1, aes(color = bioproject,  fill = bioproject))+
  geom_point(size = 3, shape = 21, aes(fill = bioproject))+
  stat_mean(aes(fill = bioproject), shape = 23, size = 4) + 
  annotate("text", x =- 0.3, y = 0.5, color = "black", label = paste0("p=", round(ado_bray_biop$`Pr(>F)`[1], digits = 3)))+
  annotate("text", x = -0.3, y = 0.45, color = "black", label = paste0("R\u00b2=", round((ado_bray_biop$R2[1])*100, digits = 2), "%"))+
  scale_fill_nejm(name = "Bioproject")+
  scale_color_nejm(name = "Bioproject")+
  theme_bw()+
  theme(legend.position = "right")

ggsave("Results/BC_pcoa_biop_plot.png", BC_pcoa_biop_plot, width = 10, height = 8, dpi = 300)

BC_pcoa_biop_plot

```
```{r}

## By host sex

ps_filt_fixed_rel@sam_data$host_sex <- as.factor(ps_filt_fixed_rel@sam_data$host_sex)

ado_bray_sex <- adonis2(beta_ASV_bray ~ host_sex, meta_merged)
print(ado_bray_sex)

BC_pcoa_sex_plot <- plot_ordination(ps_filt_fixed_rel, PCoA_ASV_bray, axes = c(1,2))+
  stat_conf_ellipse(geom = "polygon", alpha = 0.1, aes(color = host_sex,  fill = host_sex))+
  geom_point(size = 3, shape = 21, aes(fill = host_sex))+
  stat_mean(aes(fill = host_sex), shape = 23, size = 4) + 
  annotate("text", x =- 0.3, y = 0.5, color = "black", label = paste0("p=", round(ado_bray_sex$`Pr(>F)`[1], digits = 3)))+
  annotate("text", x = -0.3, y = 0.45, color = "black", label = paste0("R\u00b2=", round((ado_bray_sex$R2[1])*100, digits = 2), "%"))+
  scale_fill_nejm(name = "Host sex")+
  scale_color_nejm(name = "Host Sex")+
  theme_bw()+
  theme(legend.position = "right")

ggsave("Results/BC_pcoa_sex_plot.png", BC_pcoa_sex_plot, width = 10, height = 8, dpi = 300)

BC_pcoa_sex_plot

```


```{r}
bd_df <- data.frame(samples = rownames(group.bd$vectors),
                    distances = group.bd$distances,
                    group = meta_merged$group)

kruskal_bc_test <- bd_df %>% rstatix::kruskal_test(distances ~ group)
print(kruskal_bc_test)

# Dunn’s test for pairwise comparisons with Benjamini-Hochberg correction
dunn_bc_test <- bd_df %>% rstatix::dunn_test(distances ~ group, p.adjust.method = "BH")
print(dunn_bc_test)

dunn_bc_test_table <- apply(dunn_bc_test,2,as.character)
write.csv(dunn_bc_test_table, "Results/dunn_bc_test_table.csv", row.names = FALSE)

# Extract p-values for compact letter display
dunn_bc_pvalues <- dunn_bc_test %>% select(group1, group2, p.adj) %>%
  pivot_wider(names_from = group2, values_from = p.adj) %>%
  column_to_rownames(var = "group1")

# Convert to a matrix format for multcompLetters
dunn_bc_matrix <- as.matrix(dunn_bc_pvalues)
# Fill the lower triangle with upper triangle values to make the matrix symmetric
dunn_bc_matrix[lower.tri(dunn_bc_matrix)] <- t(dunn_bc_matrix)[lower.tri(dunn_bc_matrix)]


# Generate compact letter display for groups
dunn_bc_labels <- multcompLetters(dunn_bc_matrix)$Letters
print(dunn_bc_labels)

```



```{r}
# root: ASV5155
root_seq <- "ASV5155"

# Extract tree from phyloseq object
tree <- phy_tree(ps_filt_fixed_rel)

# Set the root
tree_rooted <- root(tree, outgroup = root_seq, resolve.root = TRUE)

# Replace the tree in the phyloseq object
phy_tree(ps_filt_fixed_rel) <- tree_rooted
# ggtree(ps_filt_fixed_rel@phy_tree, options(ignore.negative.edge=TRUE))

PCoA_ASV_wunifrac <- ordinate(ps_filt_fixed_rel, distance = "wunifrac", method = "PCoA")
beta_ASV_wunifrac <- phyloseq::distance(ps_filt_fixed_rel, "wunifrac")
ado_wunifrac_group <- adonis2(beta_ASV_wunifrac ~ group, meta_merged)
print(ado_wunifrac_group)


group_wunifrac.bd <- betadisper(d = beta_ASV_wunifrac, group = meta_merged$group, type="centroid")
boxplot(group_wunifrac.bd)

png("Results/beta_dispersion_wu_boxplot.png", width = 3500, height = 2000, res = 300)  # High resolution
boxplot(group_wunifrac.bd, main = "Beta Dispersion per Group (WU)", cex.axis = 1.2, cex.lab = 1.5)
dev.off()

permutest(group_wunifrac.bd)


```

```{r}
WU_pcoa_plot <- plot_ordination(ps_filt_fixed_rel, PCoA_ASV_wunifrac, axes = c(1,2))+
  stat_conf_ellipse(geom = "polygon", alpha = 0.1, aes(color = group,  fill = group))+
  geom_point(size = 3, shape = 21, aes(fill = group))+
  stat_mean(aes(fill = group), shape = 23, size = 4) + 
  annotate("text", x = 0.1, y = -0.15, color = "black", label = paste0("p=", round(ado_wunifrac_group$`Pr(>F)`[1], digits = 3)))+
  annotate("text", x = 0.1, y = -0.13, color = "black", label = paste0("R\u00b2=", round((ado_wunifrac_group$R2[1])*100, digits = 2), "%"))+
  scale_fill_nejm(name = "Group")+
  scale_color_nejm(name = "Group")+
  theme_bw()+
  theme(legend.position = "right")

ggsave("Results/WU_pcoa_plot.png", WU_pcoa_plot, width = 10, height = 8, dpi = 300)


WU_pcoa_plot

```
```{r}
ado_wunifrac_biop <- adonis2(beta_ASV_wunifrac ~ bioproject, meta_merged)
print(ado_wunifrac_biop)

WU_pcoa_biop_plot <- plot_ordination(ps_filt_fixed_rel, PCoA_ASV_wunifrac, axes = c(1,2))+
  stat_conf_ellipse(geom = "polygon", alpha = 0.1, aes(color = bioproject,  fill = bioproject))+
  geom_point(size = 3, shape = 21, aes(fill = bioproject))+
  stat_mean(aes(fill = bioproject), shape = 23, size = 4) + 
  annotate("text", x = 0.1, y = -0.15, color = "black", label = paste0("p=", round(ado_wunifrac_biop$`Pr(>F)`[1], digits = 3)))+
  annotate("text", x = 0.1, y = -0.13, color = "black", label = paste0("R\u00b2=", round((ado_wunifrac_biop$R2[1])*100, digits = 2), "%"))+
  scale_fill_nejm(name = "Bioproject")+
  scale_color_nejm(name = "Bioproject")+
  theme_bw()+
  theme(legend.position = "right")

ggsave("Results/WU_pcoa_biop_plot.png", WU_pcoa_biop_plot, width = 10, height = 8, dpi = 300)


WU_pcoa_biop_plot

```
```{r}
ado_wunifrac_sex <- adonis2(beta_ASV_wunifrac ~ host_sex, meta_merged)
print(ado_wunifrac_sex)

WU_pcoa_sex_plot <- plot_ordination(ps_filt_fixed_rel, PCoA_ASV_wunifrac, axes = c(1,2))+
  stat_conf_ellipse(geom = "polygon", alpha = 0.1, aes(color = host_sex,  fill = host_sex))+
  geom_point(size = 3, shape = 21, aes(fill = host_sex))+
  stat_mean(aes(fill = host_sex), shape = 23, size = 4) + 
  annotate("text", x = 0.1, y = -0.15, color = "black", label = paste0("p=", round(ado_wunifrac_sex$`Pr(>F)`[1], digits = 3)))+
  annotate("text", x = 0.1, y = -0.13, color = "black", label = paste0("R\u00b2=", round((ado_wunifrac_sex$R2[1])*100, digits = 2), "%"))+
  scale_fill_nejm(name = "host_sex")+
  scale_color_nejm(name = "host_sex")+
  theme_bw()+
  theme(legend.position = "right")

ggsave("Results/WU_pcoa_sex_plot.png", WU_pcoa_sex_plot, width = 10, height = 8, dpi = 300)


WU_pcoa_biop_plot
```



```{r}
wu_bd_df <- data.frame(samples = rownames(group_wunifrac.bd$vectors),
                    distances = group_wunifrac.bd$distances,
                    group = meta_merged$group)

kruskal_wu_test <- wu_bd_df %>% rstatix::kruskal_test(distances ~ group)
print(kruskal_wu_test)

# Dunn’s test for pairwise comparisons with Benjamini-Hochberg correction
dunn_wu_test <- wu_bd_df %>% rstatix::dunn_test(distances ~ group, p.adjust.method = "BH", detailed = TRUE)
print(dunn_wu_test)

dunn_wu_test_table <- apply(dunn_wu_test,2,as.character)
write.csv(dunn_wu_test_table, "Results/dunn_wu_test_table.csv", row.names = FALSE)

# Extract p-values for compact letter display
dunn_wu_pvalues <- dunn_wu_test %>% select(group1, group2, p.adj) %>%
  pivot_wider(names_from = group2, values_from = p.adj) %>%
  column_to_rownames(var = "group1")

# Convert to a matrix format for multcompLetters
dunn_wu_matrix <- as.matrix(dunn_wu_pvalues)
# Fill the lower triangle with upper triangle values to make the matrix symmetric
dunn_wu_matrix[lower.tri(dunn_wu_matrix)] <- t(dunn_wu_matrix)[lower.tri(dunn_wu_matrix)]


# Generate compact letter display for groups
dunn_wu_labels <- multcompLetters(dunn_wu_matrix)$Letters
print(dunn_wu_labels)

```

## Beta diversity analysis on baseline and end samples
```{r}
## For baseline samples
ps_ff_rel_base <- ps_filt_fixed_rel %>% ps_filter(!is.na(baseline_status))

meta_beta_base <- as.data.frame(as.matrix(sample_data(ps_ff_rel_base)))

PCoA_ASV_bray_base <- ordinate(ps_ff_rel_base, distance = "bray", method = "PCoA")
PCoA_ASV_bray_base


beta_ASV_bray_base <- phyloseq::distance(ps_ff_rel_base, "bray")
ado_bray_group_base <- adonis2(beta_ASV_bray_base ~ baseline_status, meta_beta_base)
print(ado_bray_group_base)

baseline_bc.bd <- betadisper(d = beta_ASV_bray_base, group = meta_beta_base$baseline_status, type="centroid")
bd_baseline_bc_boxplot <- boxplot(baseline_bc.bd)

bd_baseline_bc_boxplot

png("Results/betadisp_baseline_bc_boxplot.png", width = 3500, height = 2000, res = 300)  # High resolution
boxplot(baseline_bc.bd, main = "Beta Dispersion per Baseline Status (BC)", cex.axis = 1.2, cex.lab = 1.5)
dev.off()

permutest(baseline_bc.bd)

BC_baseline_pcoa_plot <- plot_ordination(ps_ff_rel_base, PCoA_ASV_bray_base, axes = c(1,2))+
  stat_conf_ellipse(geom = "polygon", alpha = 0.1, aes(color = baseline_status,  fill = baseline_status))+
  geom_point(size = 3, shape = 21, aes(fill = baseline_status))+
  stat_mean(aes(fill = baseline_status), shape = 23, size = 4) + 
  annotate("text", x =- 0.3, y = 0.5, color = "black", label = paste0("p=", round(ado_bray_group_base$`Pr(>F)`[1], digits = 3)))+
  annotate("text", x = -0.3, y = 0.45, color = "black", label = paste0("R\u00b2=", round((ado_bray_group_base$R2[1])*100, digits = 2), "%"))+
  scale_fill_nejm(name = "Baseline Status")+
  scale_color_nejm(name = "Baseline Status")+
  theme_bw()+
  theme(legend.position = "right")

ggsave("Results/BC_baseline_pcoa_plot.png", BC_baseline_pcoa_plot, width = 10, height = 8, dpi = 300)

BC_baseline_pcoa_plot

ado_bray_biop_base <- adonis2(beta_ASV_bray_base ~ bioproject, meta_beta_base)
print(ado_bray_biop_base)

BC_baseline_pcoa_biop_plot <- plot_ordination(ps_ff_rel_base, PCoA_ASV_bray_base, axes = c(1,2))+
  stat_conf_ellipse(geom = "polygon", alpha = 0.1, aes(color = bioproject,  fill = bioproject))+
  geom_point(size = 3, shape = 21, aes(fill = bioproject))+
  stat_mean(aes(fill = bioproject), shape = 23, size = 4) + 
  annotate("text", x =- 0.3, y = 0.5, color = "black", label = paste0("p=", round(ado_bray_biop_base$`Pr(>F)`[1], digits = 3)))+
  annotate("text", x = -0.3, y = 0.45, color = "black", label = paste0("R\u00b2=", round((ado_bray_biop_base$R2[1])*100, digits = 2), "%"))+
  scale_fill_nejm(name = "Bioproject")+
  scale_color_nejm(name = "Bioproject")+
  theme_bw()+
  theme(legend.position = "right")

ggsave("Results/BC_baseline_pcoa_biop_plot.png", BC_baseline_pcoa_biop_plot, width = 10, height = 8, dpi = 300)

BC_baseline_pcoa_biop_plot

bd_base_df <- data.frame(samples = rownames(baseline_bc.bd$vectors),
                    distances = baseline_bc.bd$distances,
                    baseline_status = meta_beta_base$baseline_status)

kruskal_bc_test_base <- bd_base_df %>% rstatix::kruskal_test(distances ~ baseline_status)
print(kruskal_bc_test_base)

# Dunn’s test for pairwise comparisons with Benjamini-Hochberg correction
dunn_bc_test_base <- bd_base_df %>% rstatix::dunn_test(distances ~ baseline_status, p.adjust.method = "BH")
print(dunn_bc_test_base)

dunn_bc_test_base_table <- apply(dunn_bc_test_base,2,as.character)
write.csv(dunn_bc_test_base_table, "Results/dunn_bc_test_base_table.csv", row.names = FALSE)

# Extract p-values for compact letter display
dunn_bc_pvalues_base <- dunn_bc_test_base %>% select(group1, group2, p.adj) %>%
  pivot_wider(names_from = group2, values_from = p.adj) %>%
  column_to_rownames(var = "group1")

# Convert to a matrix format for multcompLetters
dunn_bc_matrix_base <- as.matrix(dunn_bc_pvalues_base)
# Fill the lower triangle with upper triangle values to make the matrix symmetric
dunn_bc_matrix_base[lower.tri(dunn_bc_matrix_base)] <- t(dunn_bc_matrix_base)[lower.tri(dunn_bc_matrix_base)]


# Generate compact letter display for groups
dunn_bc_labels_base <- multcompLetters(dunn_bc_matrix_base)$Letters
print(dunn_bc_labels_base)

```


```{r}
## For end samples
ps_ff_rel_end <- ps_filt_fixed_rel %>% ps_filter(!is.na(disease_status_end))

meta_beta_end <- as.data.frame(as.matrix(sample_data(ps_ff_rel_end)))

PCoA_ASV_bray_end <- ordinate(ps_ff_rel_end, distance = "bray", method = "PCoA")
PCoA_ASV_bray_end


beta_ASV_bray_end <- phyloseq::distance(ps_ff_rel_end, "bray")
ado_bray_group_end <- adonis2(beta_ASV_bray_end ~ disease_status_end, meta_beta_end)
print(ado_bray_group_end)

end_bc.bd <- betadisper(d = beta_ASV_bray_end, group = meta_beta_end$disease_status_end, type="centroid")
bd_end_bc_boxplot <- boxplot(end_bc.bd)

bd_end_bc_boxplot

png("Results/betadisp_end_bc_boxplot.png", width = 3500, height = 2000, res = 300)  # High resolution
boxplot(end_bc.bd, main = "Beta Dispersion per End Status (BC)", cex.axis = 1.2, cex.lab = 1.5)
dev.off()

permutest(end_bc.bd)

BC_end_pcoa_plot <- plot_ordination(ps_ff_rel_end, PCoA_ASV_bray_end, axes = c(1,2))+
  stat_conf_ellipse(geom = "polygon", alpha = 0.1, aes(color = disease_status_end,  fill = disease_status_end))+
  geom_point(size = 3, shape = 21, aes(fill = disease_status_end))+
  stat_mean(aes(fill = disease_status_end), shape = 23, size = 4) + 
  annotate("text", x =- 0.3, y = 0.5, color = "black", label = paste0("p=", round(ado_bray_group_end$`Pr(>F)`[1], digits = 3)))+
  annotate("text", x = -0.3, y = 0.45, color = "black", label = paste0("R\u00b2=", round((ado_bray_group_end$R2[1])*100, digits = 2), "%"))+
  scale_fill_nejm(name = "End Status")+
  scale_color_nejm(name = "End Status")+
  theme_bw()+
  theme(legend.position = "right")

ggsave("Results/BC_end_pcoa_plot.png", BC_end_pcoa_plot, width = 10, height = 8, dpi = 300)

BC_end_pcoa_plot

ado_bray_biop_end <- adonis2(beta_ASV_bray_end ~ bioproject, meta_beta_end)
print(ado_bray_biop_end)

BC_end_pcoa_biop_plot <- plot_ordination(ps_ff_rel_end, PCoA_ASV_bray_end, axes = c(1,2))+
  stat_conf_ellipse(geom = "polygon", alpha = 0.1, aes(color = bioproject,  fill = bioproject))+
  geom_point(size = 3, shape = 21, aes(fill = bioproject))+
  stat_mean(aes(fill = bioproject), shape = 23, size = 4) + 
  annotate("text", x =- 0.3, y = 0.5, color = "black", label = paste0("p=", round(ado_bray_biop_end$`Pr(>F)`[1], digits = 3)))+
  annotate("text", x = -0.3, y = 0.45, color = "black", label = paste0("R\u00b2=", round((ado_bray_biop_end$R2[1])*100, digits = 2), "%"))+
  scale_fill_nejm(name = "Bioproject")+
  scale_color_nejm(name = "Bioproject")+
  theme_bw()+
  theme(legend.position = "right")

ggsave("Results/BC_end_pcoa_biop_plot.png", BC_end_pcoa_biop_plot, width = 10, height = 8, dpi = 300)

BC_end_pcoa_biop_plot

bd_bc_end_df <- data.frame(samples = rownames(end_bc.bd$vectors),
                         distances = end_bc.bd$distances,
                         disease_status_end = meta_beta_end$disease_status_end)

kruskal_bc_test_end <- bd_bc_end_df %>% rstatix::kruskal_test(distances ~ disease_status_end)
print(kruskal_bc_test_end)

# Dunn’s test for pairwise comparisons with Benjamini-Hochberg correction
dunn_bc_test_end <- bd_bc_end_df %>% rstatix::dunn_test(distances ~ disease_status_end, p.adjust.method = "BH")
print(dunn_bc_test_end)

dunn_bc_test_end_table <- apply(dunn_bc_test_end,2,as.character)
write.csv(dunn_bc_test_end_table, "Results/dunn_bc_test_end_table.csv", row.names = FALSE)

# Extract p-values for compact letter display
dunn_bc_pvalues_end <- dunn_bc_test_end %>% select(group1, group2, p.adj) %>%
  pivot_wider(names_from = group2, values_from = p.adj) %>%
  column_to_rownames(var = "group1")

# Convert to a matrix format for multcompLetters
dunn_bc_matrix_end <- as.matrix(dunn_bc_pvalues_end)
# Fill the lower triangle with upper triangle values to make the matrix symmetric
dunn_bc_matrix_end[lower.tri(dunn_bc_matrix_end)] <- t(dunn_bc_matrix_end)[lower.tri(dunn_bc_matrix_end)]


# Generate compact letter display for groups
dunn_bc_labels_end <- multcompLetters(dunn_bc_matrix_end)$Letters
print(dunn_bc_labels_end)

```

## Using Weighted Unifrac instead of BC
```{r}
## Set root for WU so it runs with the same root as previous wu analysis

# ASV5155

# Define the root sequence (use the exact sequence you mentioned before)
root_seq <- "ASV5155"

# Extract tree from phyloseq object
tree <- phy_tree(ps_filt_fixed_rel)

# Set the root
tree_rooted <- root(tree, outgroup = root_seq, resolve.root = TRUE)

# Replace the tree in the phyloseq object
phy_tree(ps_filt_fixed_rel) <- tree_rooted
# ggtree(ps_filt_fixed_rel@phy_tree, options(ignore.negative.edge=TRUE))

## For base samples
ps_ff_rel_base <- ps_filt_fixed_rel %>% ps_filter(!is.na(baseline_status))

meta_beta_base <- as.data.frame(as.matrix(sample_data(ps_ff_rel_base)))

PCoA_ASV_wunifrac_base <- ordinate(ps_ff_rel_base, distance = "wunifrac", method = "PCoA")
PCoA_ASV_wunifrac_base


beta_ASV_wunifrac_base <- phyloseq::distance(ps_ff_rel_base, "wunifrac")
ado_wunifrac_group_base <- adonis2(beta_ASV_wunifrac_base ~ baseline_status, meta_beta_base)
print(ado_wunifrac_group_base)

baseline_wu.bd <- betadisper(d = beta_ASV_wunifrac_base, group = meta_beta_base$baseline_status, type="centroid")
bd_baseline_wu_boxplot <- boxplot(baseline_wu.bd)

bd_baseline_wu_boxplot

png("Results/betadisp_baseline_wu_boxplot.png", width = 3500, height = 2000, res = 300)  # High resolution
boxplot(baseline_wu.bd, main = "Beta Dispersion per Baseline Status (WU)", cex.axis = 1.2, cex.lab = 1.5)
dev.off()

permutest(baseline_wu.bd)

WU_baseline_pcoa_plot <- plot_ordination(ps_ff_rel_base, PCoA_ASV_wunifrac_base, axes = c(1,2))+
  stat_conf_ellipse(geom = "polygon", alpha = 0.1, aes(color = baseline_status,  fill = baseline_status))+
  geom_point(size = 3, shape = 21, aes(fill = baseline_status))+
  stat_mean(aes(fill = baseline_status), shape = 23, size = 4) + 
  annotate("text", x =- 0.3, y = 0.5, color = "black", label = paste0("p=", round(ado_wunifrac_group_base$`Pr(>F)`[1], digits = 3)))+
  annotate("text", x = -0.3, y = 0.45, color = "black", label = paste0("R\u00b2=", round((ado_wunifrac_group_base$R2[1])*100, digits = 2), "%"))+
  scale_fill_nejm(name = "base Status")+
  scale_color_nejm(name = "base Status")+
  theme_bw()+
  theme(legbase.position = "right")

ggsave("Results/WU_baseline_pcoa_plot.png", WU_baseline_pcoa_plot, width = 10, height = 8, dpi = 300)

WU_baseline_pcoa_plot

ado_wunifrac_biop_base <- adonis2(beta_ASV_wunifrac_base ~ bioproject, meta_beta_base)
print(ado_wunifrac_biop_base)

WU_baseline_pcoa_biop_plot <- plot_ordination(ps_ff_rel_base, PCoA_ASV_wunifrac_base, axes = c(1,2))+
  stat_conf_ellipse(geom = "polygon", alpha = 0.1, aes(color = bioproject,  fill = bioproject))+
  geom_point(size = 3, shape = 21, aes(fill = bioproject))+
  stat_mean(aes(fill = bioproject), shape = 23, size = 4) + 
  annotate("text", x =- 0.3, y = 0.5, color = "black", label = paste0("p=", round(ado_wunifrac_biop_base$`Pr(>F)`[1], digits = 3)))+
  annotate("text", x = -0.3, y = 0.45, color = "black", label = paste0("R\u00b2=", round((ado_wunifrac_biop_base$R2[1])*100, digits = 2), "%"))+
  scale_fill_nejm(name = "Bioproject")+
  scale_color_nejm(name = "Bioproject")+
  theme_bw()+
  theme(legbase.position = "right")

ggsave("Results/WU_baseline_pcoa_biop_plot.png", WU_baseline_pcoa_biop_plot, width = 10, height = 8, dpi = 300)

WU_baseline_pcoa_biop_plot

bd_wu_base_df <- data.frame(samples = rownames(baseline_wu.bd$vectors),
                         distances = baseline_wu.bd$distances,
                         baseline_status = meta_beta_base$baseline_status)

kruskal_wu_test_base <- bd_wu_base_df %>% rstatix::kruskal_test(distances ~ baseline_status)
print(kruskal_wu_test_base)

# Dunn’s test for pairwise comparisons with Benjamini-Hochberg correction
dunn_wu_test_base <- bd_base_df %>% rstatix::dunn_test(distances ~ baseline_status, p.adjust.method = "BH")
print(dunn_wu_test_base)

dunn_wu_test_base_table <- apply(dunn_wu_test_base,2,as.character)
write.csv(dunn_wu_test_base_table, "Results/dunn_wu_test_base_table.csv", row.names = FALSE)

# Extract p-values for compact letter display
dunn_wu_pvalues_base <- dunn_wu_test_base %>% select(group1, group2, p.adj) %>%
  pivot_wider(names_from = group2, values_from = p.adj) %>%
  column_to_rownames(var = "group1")

# Convert to a matrix format for multcompLetters
dunn_wu_matrix_base <- as.matrix(dunn_wu_pvalues_base)
# Fill the lower triangle with upper triangle values to make the matrix symmetric
dunn_wu_matrix_base[lower.tri(dunn_wu_matrix_base)] <- t(dunn_wu_matrix_base)[lower.tri(dunn_wu_matrix_base)]


# Generate compact letter display for groups
dunn_wu_labels_base <- multcompLetters(dunn_wu_matrix_base)$Letters
print(dunn_wu_labels_base)

```


```{r}
# root: ASV5155

## For end samples
ps_ff_rel_end <- ps_filt_fixed_rel %>% ps_filter(!is.na(disease_status_end))

meta_beta_end <- as.data.frame(as.matrix(sample_data(ps_ff_rel_end)))

PCoA_ASV_wunifrac_end <- ordinate(ps_ff_rel_end, distance = "wunifrac", method = "PCoA")
PCoA_ASV_wunifrac_end


beta_ASV_wunifrac_end <- phyloseq::distance(ps_ff_rel_end, "wunifrac")
ado_wunifrac_group_end <- adonis2(beta_ASV_wunifrac_end ~ disease_status_end, meta_beta_end)
print(ado_wunifrac_group_end)

end_wu.bd <- betadisper(d = beta_ASV_wunifrac_end, group = meta_beta_end$disease_status_end, type="centroid")
bd_end_wu_boxplot <- boxplot(end_wu.bd)

bd_end_wu_boxplot

png("Results/betadisp_end_wu_boxplot.png", width = 3500, height = 2000, res = 300)  # High resolution
boxplot(end_wu.bd, main = "Beta Dispersion per end Status (WU)", cex.axis = 1.2, cex.lab = 1.5)
dev.off()

permutest(end_wu.bd)

WU_end_pcoa_plot <- plot_ordination(ps_ff_rel_end, PCoA_ASV_wunifrac_end, axes = c(1,2))+
  stat_conf_ellipse(geom = "polygon", alpha = 0.1, aes(color = disease_status_end,  fill = disease_status_end))+
  geom_point(size = 3, shape = 21, aes(fill = disease_status_end))+
  stat_mean(aes(fill = disease_status_end), shape = 23, size = 4) + 
  annotate("text", x =- 0.3, y = 0.5, color = "black", label = paste0("p=", round(ado_wunifrac_group_end$`Pr(>F)`[1], digits = 3)))+
  annotate("text", x = -0.3, y = 0.45, color = "black", label = paste0("R\u00b2=", round((ado_wunifrac_group_end$R2[1])*100, digits = 2), "%"))+
  scale_fill_nejm(name = "End Status")+
  scale_color_nejm(name = "End Status")+
  theme_bw()+
  theme(legend.position = "right")

ggsave("Results/WU_end_pcoa_plot.png", WU_end_pcoa_plot, width = 10, height = 8, dpi = 300)

WU_end_pcoa_plot

ado_wunifrac_biop_end <- adonis2(beta_ASV_wunifrac_end ~ bioproject, meta_beta_end)
print(ado_wunifrac_biop_end)

WU_end_pcoa_biop_plot <- plot_ordination(ps_ff_rel_end, PCoA_ASV_wunifrac_end, axes = c(1,2))+
  stat_conf_ellipse(geom = "polygon", alpha = 0.1, aes(color = bioproject,  fill = bioproject))+
  geom_point(size = 3, shape = 21, aes(fill = bioproject))+
  stat_mean(aes(fill = bioproject), shape = 23, size = 4) + 
  annotate("text", x =- 0.3, y = 0.5, color = "black", label = paste0("p=", round(ado_wunifrac_biop_end$`Pr(>F)`[1], digits = 3)))+
  annotate("text", x = -0.3, y = 0.45, color = "black", label = paste0("R\u00b2=", round((ado_wunifrac_biop_end$R2[1])*100, digits = 2), "%"))+
  scale_fill_nejm(name = "Bioproject")+
  scale_color_nejm(name = "Bioproject")+
  theme_bw()+
  theme(legend.position = "right")

ggsave("Results/WU_end_pcoa_biop_plot.png", WU_end_pcoa_biop_plot, width = 10, height = 8, dpi = 300)

WU_end_pcoa_biop_plot

bd_wu_end_df <- data.frame(samples = rownames(end_wu.bd$vectors),
                         distances = end_wu.bd$distances,
                         disease_status_end = meta_beta_end$disease_status_end)

kruskal_wu_test_end <- bd_wu_end_df %>% rstatix::kruskal_test(distances ~ disease_status_end)
print(kruskal_wu_test_end)

# Dunn’s test for pairwise comparisons with Benjamini-Hochberg correction
dunn_wu_test_end <- bd_wu_end_df %>% rstatix::dunn_test(distances ~ disease_status_end, p.adjust.method = "BH")
print(dunn_wu_test_end)

dunn_wu_test_end_table <- apply(dunn_wu_test_end,2,as.character)
write.csv(dunn_wu_test_end_table, "Results/dunn_wu_test_end_table.csv", row.names = FALSE)

# Extract p-values for compact letter display
dunn_wu_pvalues_end <- dunn_wu_test_end %>% select(group1, group2, p.adj) %>%
  pivot_wider(names_from = group2, values_from = p.adj) %>%
  column_to_rownames(var = "group1")

# Convert to a matrix format for multcompLetters
dunn_wu_matrix_end <- as.matrix(dunn_wu_pvalues_end)
# Fill the lower triangle with upper triangle values to make the matrix symmetric
dunn_wu_matrix_end[lower.tri(dunn_wu_matrix_end)] <- t(dunn_wu_matrix_end)[lower.tri(dunn_wu_matrix_end)]


# Generate compact letter display for groups
dunn_wu_labels_end <- multcompLetters(dunn_wu_matrix_end)$Letters
print(dunn_wu_labels_end)

```

## Find significant taxa
```{r}
# Convert OTU table to a data frame
taxa_abund_kw <- as.data.frame(otu_table(ps_filt_fixed_gen_rel))

# Convert row names to a column (sample IDs)
taxa_abund_kw <- taxa_abund_kw %>% 
  tibble::rownames_to_column(var = "samples")

# Convert sample_data from phyloseq object to a proper data frame
metadata_samples_kw <- as(sample_data(ps_filt_fixed_gen_rel), "data.frame") %>% 
  tibble::rownames_to_column(var = "samples") %>% # Ensure sample IDs are a column
  select(c(samples, group))


# Merge metadata with abundance data
taxa_data_kw <- left_join(metadata_samples_kw, taxa_abund_kw, by = "samples") 

taxa_data_kw_tib <- taxa_data_kw %>%
  as_tibble()  # Convert to tibble for tidyverse functions

# Function to perform Kruskal-Wallis test on each taxon
kruskal_test_results <- taxa_data_kw_tib %>%
  pivot_longer(cols = -c(samples, group), names_to = "taxon", values_to = "abundance") %>%
  group_by(taxon) %>%
  rstatix::kruskal_test(abundance ~ group) %>%
  rstatix::adjust_pvalue(method = "BH") %>%  # Adjust for multiple testing (Benjamini-Hochberg)
  arrange(p.adj)

# View significant results (p.adj < 0.05)
significant_taxa_kw <- kruskal_test_results %>% filter(p.adj < 0.05)

significant_taxa_kw

# Display top results
significant_taxa_kw %>% 
  arrange(p.adj) %>% 
  head(n=20)

top_10_sig_taxa <- significant_taxa_kw %>% arrange(p.adj) %>% head(n=10)

top_10_sig_taxa

```


```{r}
long_taxa_kw <- taxa_data_kw %>%
  pivot_longer(cols = -c(samples, group), names_to = "taxon", values_to = "abundance") 

long_taxa_kw <- long_taxa_kw %>%
  group_by(taxon) %>%
  filter(any(abundance > 0)) %>%
  ungroup()

long_taxa_kw <- long_taxa_kw %>%
  group_by(taxon) %>%
  filter(n_distinct(group) > 1) %>%
  ungroup()

dunn_test_taxa <- long_taxa_kw %>%
  group_by(taxon) %>%
  rstatix::dunn_test(abundance ~ group, p.adjust.method = "BH")


# View significant pairwise comparisons
sig_pairs <- dunn_test_taxa %>% filter(p.adj < 0.05)

sig_pairs %>% kbl() %>% 
  kableExtra::kable_paper(font_size=12)

sig_pairs %>% arrange(p.adj) %>% head(n=10)

sig_pairs %>% arrange(p.adj) %>% head(n=20)

## Create new dataframe combining group pairs if necessary

# Display top results
dunn_test_taxa %>% kbl() %>% 
  kableExtra::kable_paper(font_size=12)



```


## Check prevalence of taxa
```{r}

print(ps_filt_fixed)

ps_filt_fixed_prev <- tax_filter(ps_filt_fixed,
  min_prevalence = 0.01)
  
print(ps_filt_fixed_prev)


prev_thresholds <- c(0.05, 0.01, 0.001)

prev_filt_results <- data.frame(min_prevalence = prev_thresholds, num_taxa_retained = NA, perc_taxa_removed = NA)

total_taxa <- nrow(tax_table(ps_filt_fixed))

for (i in seq_along(prev_thresholds)) {
  ps_filt_fixed_samples_prev_all <- ps_filt_fixed %>%
    tax_filter(min_prevalence = prev_thresholds[i])
  
  num_taxa_retained <- nrow(tax_table(ps_filt_fixed_samples_prev_all))
  perc_taxa_removed <- 100 * (total_taxa - num_taxa_retained) / total_taxa
  
  prev_filt_results$num_taxa_retained[i] <- num_taxa_retained
  prev_filt_results$perc_taxa_removed[i] <- perc_taxa_removed
}

print(prev_filt_results)

```

```{r}

# ps_filt_fixed_genus_rel <- tax_transform(ps_filt_fixed_genus, trans = "compositional")
input_sample_metadata <- ps_filt_fixed_gen_rel %>% meta
input_data_genus_Maaslin <- as.data.frame(otu_table(ps_filt_fixed_genus_rel))

input_data_genus_Maaslin <- input_data_genus_Maaslin[sapply(input_data_genus_Maaslin, is.numeric)]

output_directory_genus <- "Results/Maaslin2_output_genus"

fit_data_genus <- Maaslin2::Maaslin2(
    input_data = input_data_genus_Maaslin,
    input_metadata = input_sample_metadata,
    output = output_directory_genus,
    analysis_method = "LM",
    fixed_effects = "group",
    normalization = "NONE", 
    correction = "BH",
    transform = "LOG",
    min_abundance = 0,
    min_prevalence = 0,
    reference = "group,healthy-healthy")

```


```{r}

# ps_filt_fixed_genus_rel <- tax_transform(ps_filt_fixed_genus, trans = "compositional")
input_sample_metadata <- ps_filt_fixed_gen_rel %>% meta
input_data_genus_Maaslin <- as.data.frame(otu_table(ps_filt_fixed_genus_rel))

input_data_genus_Maaslin <- input_data_genus_Maaslin[sapply(input_data_genus_Maaslin, is.numeric)]

output_directory_gen <- "Results/Maaslin2_output_dir_genus"

fit_data_gen <- Maaslin2::Maaslin2(
    input_data = input_data_genus_Maaslin,
    input_metadata = input_sample_metadata,
    output = output_directory_gen,
    analysis_method = "LM",
    fixed_effects = "group",
    normalization = "NONE", 
    correction = "BH",
    min_abundance = 0,
    min_prevalence = 0,
    reference = "group,healthy-healthy")

```


```{r}
fit_data_genus$results %>% arrange(pval) %>% 
  select(feature, value, pval) %>% 
  head(n=10)

```

## DAA only for patient data
```{r}
ps_filt_fixed_genus_pat <- ps_filt_fixed_genus %>% ps_filter(host_assigned_donor_subject_id != "is donor")

ps_filt_fixed_genus_rel_pat <- tax_transform(ps_filt_fixed_genus_pat, trans = "compositional")
input_sample_metadata_pat <- ps_filt_fixed_genus_rel_pat %>% meta
input_data_genus_Maaslin_pat <- as.data.frame(otu_table(ps_filt_fixed_genus_rel_pat))

input_data_genus_Maaslin_pat <- input_data_genus_Maaslin_pat[sapply(input_data_genus_Maaslin_pat, is.numeric)]

output_directory_genus <- "Maaslin2_output_genus_pat"

fit_data_genus_pat <- Maaslin2::Maaslin2(
    input_data = input_data_genus_Maaslin_pat,
    input_metadata = input_sample_metadata_pat,
    output = output_directory_genus,
    analysis_method = "LM",
    fixed_effects = "group",
    normalization = "NONE", 
    correction = "BH",
    transform = "LOG",
    min_abundance = 0,
    min_prevalence = 0.01,
    reference = "group,active-remission")

```
```{r}
fit_data_genus_pat$results %>% arrange(pval) %>% 
  select(feature, value, pval) %>% 
  head(n=10)
```


## Do analysis for filtered baseline samples and end samples (!is.na). Compare.


```{r}

## End samples
ps_filt_fixed_end <- ps_filt_fixed %>% ps_filter(!is.na(disease_status_end))

ps_filt_fixed_end <- ps_filt_fixed_end %>% 
  # ps_mutate(bioproject = case_when(
  #   bioproject == "PRJNA737472" ~ 1,
  #   bioproject == "PRJNA951422" ~ 2
  # )) %>% 
  ps_mutate(disease_status_end = as.factor(disease_status_end))

ps_filt_fixed_end_pat <-  ps_filt_fixed %>% ps_filter(!is.na(disease_status_end) &
                                     donor == "no")

ps_filt_fixed_end_pat <- ps_filt_fixed_end_pat %>% 
  ps_mutate(bioproject = case_when(
    bioproject == "PRJNA737472" ~ 1,
    bioproject == "PRJNA951422" ~ 2
  ))

sum_of_reads_end <- as.data.frame(sample_sums(ps_filt_fixed_end@otu_table)) %>% rownames_to_column(var = "samples") %>%
  dplyr::rename(sum_of_reads = `sample_sums(ps_filt_fixed_end@otu_table)`)%>%
  left_join((metadata_x %>% rownames_to_column(var = "samples")), by = "samples")

sum_of_reads_end_pat <- as.data.frame(sample_sums(ps_filt_fixed_end_pat@otu_table)) %>% rownames_to_column(var = "samples") %>%
  dplyr::rename(sum_of_reads = `sample_sums(ps_filt_fixed_end_pat@otu_table)`)%>%
  left_join((metadata_x %>% rownames_to_column(var = "samples")), by = "samples")


sum_of_reads_end %>% arrange(sum_of_reads) %>% 
  ggplot(aes(x = reorder(samples, sum_of_reads), y = sum_of_reads, fill = group)) +
  geom_point(shape = 21, size = 2, color = "black")+
  scale_fill_manual(values = c(
    "active-active" = "darkred",
    "active-remission" = "orange",
    "remission-remission" = "yellow",
    "remission-active" = "indianred",
    "healthy-healthy" = "darkgreen"
  ))+
  labs(x = "samples")+
  theme_bw()+
  theme(axis.text.x=element_text(angle=90))

sum_of_reads_end_pat %>% arrange(sum_of_reads) %>% 
  ggplot(aes(x = reorder(samples, sum_of_reads), y = sum_of_reads, fill = group)) +
  geom_point(shape = 21, size = 2, color = "black")+
  scale_fill_manual(values = c(
    "active-active" = "darkred",
    "active-remission" = "orange",
    "remission-remission" = "yellow",
    "remission-active" = "indianred"
    ))+
  labs(x = "samples")+
  theme_bw()+
  theme(axis.text.x=element_text(angle=90))

## All end samples 
ps_filt_fixed_end_rel <- tax_transform(ps_filt_fixed_end, trans = "compositional")

ps_filt_fixed_end_rel_clr <- tax_transform(ps_filt_fixed_end_rel, trans = "clr", chain = TRUE)

ps_filt_fixed_end_phylum <- tax_agg(ps_filt_fixed_end, rank = "Phylum")
ps_filt_fixed_end_class <- tax_agg(ps_filt_fixed_end, rank = "Class")
ps_filt_fixed_end_order <- tax_agg(ps_filt_fixed_end, rank = "Order")
ps_filt_fixed_end_family <- tax_agg(ps_filt_fixed_end, rank = "Family")
ps_filt_fixed_end_genus <- tax_agg(ps_filt_fixed_end, rank = "Genus")
#ps_filt_fixed_species <- tax_agg(ps_filt_fixed, rank = "Species")

print(ps_filt_fixed_end_phylum)
print(ps_filt_fixed_end_genus)


summary_agg_end <- data.frame(Taxa = c(colnames(ps_filt_fixed_end@tax_table)[c(2:6)], "ASVs"),
                 number_of_taxa=c(dim(ps_filt_fixed_end_phylum@otu_table)[2], dim(ps_filt_fixed_end_class@otu_table)[2], dim(ps_filt_fixed_end_order@otu_table)[2], dim(ps_filt_fixed_end_family@otu_table)[2], dim(ps_filt_fixed_end_genus@otu_table)[2], dim(ps_filt_fixed_end@otu_table)[2]))
print(summary_agg_end)

## Visualize
create_end_barplots <- function(ps_obj, end_status, max_per_plot = 6, output_prefix = "plot") {
  ps_subset <- ps_obj %>% ps_filter(disease_status_end == end_status)
  
  # Get unique subjects
  subject_ids <- unique(sample_data(ps_subset)$host_subject_id)
  
  # Split into chunks of 'max_per_plot'
  subject_batches <- split(subject_ids, ceiling(seq_along(subject_ids) / max_per_plot))
  
  # Generate plots for each batch
  plots <- list()
  
  for (i in seq_along(subject_batches)) {
    subjects <- subject_batches[[i]]
    
    ps_batch <- ps_subset %>% ps_filter(host_subject_id %in% subjects)
    
    barplot <- ps_batch %>%
      ps_mutate(host_sex = as.factor(host_sex)) %>%
      ps_mutate(bioproject = as.factor(bioproject)) %>% 
      ps_arrange(disease_status_end, bioproject) %>%
      comp_barplot(group_by = "disease_status_end", tax_level = "Phylum",
                   tax_transform_for_plot = "compositional", n_taxa = 11,
                   bar_width = 0.8, merge_other = FALSE, label = "bioproject",
                   sample_order = "default")
    
    wrapped_plot <- wrap_plots(barplot) +
      plot_layout(guides = "collect") &
      facet_wrap("host_subject_id", scales = "free_x") &
      theme(axis.text.x = element_text(angle = 60, hjust = 1, size = 10),
            axis.ticks.y = element_blank(), legend.title = element_blank(),
            legend.position = "bottom", strip.text = element_text(size = 12),
            legend.key.size = unit(0.4, "cm"),
            legend.text = element_text(size = 10))
    
    plots[[i]] <- wrapped_plot
    
    # Save each plot separately
    ggsave(filename = paste0(output_prefix, "_batch_", i, ".png"), plot = wrapped_plot, 
           width = 10, height = 8, dpi = 300)
  }
  
  return(plots)
}

# Generate and save plots
plots_end_healthy <- create_end_barplots(ps_filt_fixed_end, "healthy", max_per_plot = 6, output_prefix = "_healthy")
plots_end_remission <- create_end_barplots(ps_filt_fixed_end, "remission", max_per_plot = 6, output_prefix = "_remission")
plots_end_active <- create_end_barplots(ps_filt_fixed_end, "active", max_per_plot = 6, output_prefix = "_active")

plots_end_healthy
plots_end_remission
plots_end_remission

create_genus_end_barplots <- function(ps_obj, end_status, max_per_plot = 6, output_prefix = "plot") {
  ps_subset <- ps_obj %>% ps_filter(disease_status_end == end_status)
  
  # Get unique subjects
  subject_ids <- unique(sample_data(ps_subset)$host_subject_id)
  
  # Split into chunks of 'max_per_plot'
  subject_batches <- split(subject_ids, ceiling(seq_along(subject_ids) / max_per_plot))
  
  # Generate plots for each batch
  plots <- list()
  
  for (i in seq_along(subject_batches)) {
    subjects <- subject_batches[[i]]
    
    ps_batch <- ps_subset %>% ps_filter(host_subject_id %in% subjects)
    
    barplot <- ps_batch %>%
      ps_mutate(host_sex = as.factor(host_sex)) %>%
      ps_mutate(bioproject = as.factor(bioproject)) %>% 
      ps_arrange(disease_status_end, bioproject) %>%
      comp_barplot(group_by = "disease_status_end", tax_level = "Genus",
                   tax_transform_for_plot = "compositional", n_taxa = 11,
                   bar_width = 0.8, merge_other = FALSE, label = "bioproject",
                   sample_order = "default")
    
    wrapped_plot <- wrap_plots(barplot) +
      plot_layout(guides = "collect") &
      facet_wrap("host_subject_id", scales = "free_x") &
      theme(axis.text.x = element_text(angle = 60, hjust = 1, size = 10),
            axis.ticks.y = element_blank(), legend.title = element_blank(),
            legend.position = "bottom", strip.text = element_text(size = 12),
            legend.key.size = unit(0.4, "cm"),
            legend.text = element_text(size = 10))
    
    plots[[i]] <- wrapped_plot
    
    # Save each plot separately
    ggsave(filename = paste0(output_prefix, "_genus_batch_", i, ".png"), plot = wrapped_plot, 
           width = 10, height = 8, dpi = 300)
  }
  
  return(plots)
}

# Generate and save plots
plots_genus_end_healthy <- create_genus_end_barplots(ps_filt_fixed_end, "healthy", max_per_plot = 6, output_prefix = "_healthy")
plots_genus_end_remission <- create_genus_end_barplots(ps_filt_fixed_end, "remission", max_per_plot = 6, output_prefix = "_remission")
plots_genus_end_active <- create_genus_end_barplots(ps_filt_fixed_end, "active", max_per_plot = 6, output_prefix = "_active")

plots_genus_end_healthy
plots_genus_end_remission
plots_genus_end_active

```


```{r}
## All end patient samples
ps_filt_fixed_end_pat_rel <- tax_transform(ps_filt_fixed_end_pat, trans = "compositional")

ps_filt_fixed_end_pat_rel_clr <- tax_transform(ps_filt_fixed_end_pat_rel, trans = "clr", chain = TRUE)

ps_filt_fixed_end_pat_phylum <- tax_agg(ps_filt_fixed_end_pat, rank = "Phylum")
ps_filt_fixed_end_pat_class <- tax_agg(ps_filt_fixed_end_pat, rank = "Class")
ps_filt_fixed_end_pat_order <- tax_agg(ps_filt_fixed_end_pat, rank = "Order")
ps_filt_fixed_end_pat_family <- tax_agg(ps_filt_fixed_end_pat, rank = "Family")
ps_filt_fixed_end_pat_genus <- tax_agg(ps_filt_fixed_end_pat, rank = "Genus")
#ps_filt_fixed_species <- tax_agg(ps_filt_fixed, rank = "Species")

print(ps_filt_fixed_end_pat_phylum)
print(ps_filt_fixed_end_pat_genus)


summary_agg_end_pat <- data.frame(Taxa = c(colnames(ps_filt_fixed_end_pat@tax_table)[c(2:6)], "ASVs"),
                              number_of_taxa=c(dim(ps_filt_fixed_end_pat_phylum@otu_table)[2], dim(ps_filt_fixed_end_pat_class@otu_table)[2], dim(ps_filt_fixed_end_pat_order@otu_table)[2], dim(ps_filt_fixed_end_pat_family@otu_table)[2], dim(ps_filt_fixed_end_pat_genus@otu_table)[2], dim(ps_filt_fixed_end_pat@otu_table)[2]))
print(summary_agg_end_pat)

## Visualize
create_end_pat_barplots <- function(ps_obj, end_status, max_per_plot = 6, output_prefix = "plot") {
  ps_subset <- ps_obj %>% ps_filter(disease_status_end == end_status)
  
  # Get unique subjects
  subject_ids <- unique(sample_data(ps_subset)$host_subject_id)
  
  # Split into chunks of 'max_per_plot'
  subject_batches <- split(subject_ids, ceiling(seq_along(subject_ids) / max_per_plot))
  
  # Generate plots for each batch
  plots <- list()
  
  for (i in seq_along(subject_batches)) {
    subjects <- subject_batches[[i]]
    
    ps_batch <- ps_subset %>% ps_filter(host_subject_id %in% subjects)
    
    barplot <- ps_batch %>%
      ps_mutate(host_sex = as.factor(host_sex)) %>%
      ps_mutate(bioproject = as.factor(bioproject)) %>% 
      ps_arrange(disease_status_end, bioproject) %>%
      comp_barplot(group_by = "disease_status_end", tax_level = "Phylum",
                   tax_transform_for_plot = "compositional", n_taxa = 11,
                   bar_width = 0.8, merge_other = FALSE, label = "bioproject",
                   sample_order = "default")
    
    wrapped_plot <- wrap_plots(barplot) +
      plot_layout(guides = "collect") &
      facet_wrap("host_subject_id", scales = "free_x") &
      theme(axis.text.x = element_text(angle = 60, hjust = 1, size = 10),
            axis.ticks.y = element_blank(), legend.title = element_blank(),
            legend.position = "bottom", strip.text = element_text(size = 12),
            legend.key.size = unit(0.4, "cm"),
            legend.text = element_text(size = 10))
    
    plots[[i]] <- wrapped_plot
    
    # Save each plot separately
    ggsave(filename = paste0(output_prefix, "_batch_", i, ".png"), plot = wrapped_plot, 
           width = 10, height = 8, dpi = 300)
  }
  
  return(plots)
}

# Generate and save plots
plots_end_pat_remission <- create_end_pat_barplots(ps_filt_fixed_end_pat, "remission", max_per_plot = 6, output_prefix = "_pat_remission")
plots_end_pat_active <- create_end_pat_barplots(ps_filt_fixed_end_pat, "active", max_per_plot = 6, output_prefix = "_pat_active")

plots_end_pat_remission
plots_end_pat_remission

create_genus_end_pat_barplots <- function(ps_obj, end_status, max_per_plot = 6, output_prefix = "plot") {
  ps_subset <- ps_obj %>% ps_filter(disease_status_end == end_status)
  
  # Get unique subjects
  subject_ids <- unique(sample_data(ps_subset)$host_subject_id)
  
  # Split into chunks of 'max_per_plot'
  subject_batches <- split(subject_ids, ceiling(seq_along(subject_ids) / max_per_plot))
  
  # Generate plots for each batch
  plots <- list()
  
  for (i in seq_along(subject_batches)) {
    subjects <- subject_batches[[i]]
    
    ps_batch <- ps_subset %>% ps_filter(host_subject_id %in% subjects)
    
    barplot <- ps_batch %>%
      ps_mutate(host_sex = as.factor(host_sex)) %>%
      ps_mutate(bioproject = as.factor(bioproject)) %>% 
      ps_arrange(disease_status_end, bioproject) %>%
      comp_barplot(group_by = "disease_status_end", tax_level = "Genus",
                   tax_transform_for_plot = "compositional", n_taxa = 11,
                   bar_width = 0.8, merge_other = FALSE, label = "bioproject",
                   sample_order = "default")
    
    wrapped_plot <- wrap_plots(barplot) +
      plot_layout(guides = "collect") &
      facet_wrap("host_subject_id", scales = "free_x") &
      theme(axis.text.x = element_text(angle = 60, hjust = 1, size = 10),
            axis.ticks.y = element_blank(), legend.title = element_blank(),
            legend.position = "bottom", strip.text = element_text(size = 12),
            legend.key.size = unit(0.4, "cm"),
            legend.text = element_text(size = 10))
    
    plots[[i]] <- wrapped_plot
    
    # Save each plot separately
    ggsave(filename = paste0(output_prefix, "_genus_batch_", i, ".png"), plot = wrapped_plot, 
           width = 10, height = 8, dpi = 300)
  }
  
  return(plots)
}

# Generate and save plots
plots_genus_end_pat_remission <- create_genus_end_pat_barplots(ps_filt_fixed_end_pat, "remission", max_per_plot = 6, output_prefix = "_remission")
plots_genus_end_pat_active <- create_genus_end_pat_barplots(ps_filt_fixed_end_pat, "active", max_per_plot = 6, output_prefix = "_active")

plots_genus_end_pat_remission
plots_genus_end_pat_active

beep(sound=2)
```


## Create trajectory plots for main taxa
```{r}
top10_sig_taxa_names <- top_10_sig_taxa$taxon

ps_ff_genus_melt <- ps_filt_fixed_genus_rel %>% psmelt()

ps_traj_taxa_filt <- ps_ff_genus_melt %>% filter(OTU %in% top10_sig_taxa_names)
ps_traj_taxa_filt <- ps_traj_taxa_filt %>% filter(!host_assigned_donor_subject_id == "is donor")

# Function to generate a plot for a given host_subject_id
plot_subject_trajectory <- function(df, group_name) {
  
  df_filt <- df %>% filter(group == group_name)
  
  ggplot(df_filt, aes(x = week, y = Abundance, group = OTU, color = OTU)) +
    geom_line(aes(linetype = OTU), size = 1) + # Line style per taxa
    geom_point(size = 3) + 
    scale_color_manual(values = distinct_palette(10, pal = "brewerPlus"))
  
  
}

plots_acac <- plot_subject_trajectory(ps_traj_taxa_filt, "active-active")

plots_acrem <- plot_subject_trajectory(ps_traj_taxa_filt, "active-remission")

plots_remrem <- plot_subject_trajectory(ps_traj_taxa_filt, "remission-remission")

plots_remac <- plot_subject_trajectory(ps_traj_taxa_filt, "remission-active")

wr_acac_lineplots <- wrap_plots(plots_acac) + plot_layout(guides = "collect") &
  facet_wrap("host_subject_id", scales = "free_x") &
  theme(axis.text.x = element_text(angle = 60, hjust = 1, size = 10),
        axis.ticks.y = element_blank(), legend.title = element_blank(),
        legend.position = "bottom", strip.text = element_text(size = 12),
        legend.key.size = unit(0.4, "cm"),
        legend.text = element_text(size = 14))
wr_acrem_lineplots <- wrap_plots(plots_acrem) + plot_layout(guides = "collect") &
  facet_wrap("host_subject_id", scales = "free_x") &
  theme(axis.text.x = element_text(angle = 60, hjust = 1, size = 10),
        axis.ticks.y = element_blank(), legend.title = element_blank(),
        legend.position = "bottom", strip.text = element_text(size = 12),
        legend.key.size = unit(0.4, "cm"),
        legend.text = element_text(size = 14))

wr_remrem_lineplots <- wrap_plots(plots_remrem) + plot_layout(guides = "collect") &
  facet_wrap("host_subject_id", scales = "free_x") &
  theme(axis.text.x = element_text(angle = 60, hjust = 1, size = 10),
        axis.ticks.y = element_blank(), legend.title = element_blank(),
        legend.position = "bottom", strip.text = element_text(size = 12),
        legend.key.size = unit(0.4, "cm"),
        legend.text = element_text(size = 14))

wr_remac_lineplots <- wrap_plots(plots_remac) + plot_layout(guides = "collect") &
  facet_wrap("host_subject_id", scales = "free_x") &
  theme(axis.text.x = element_text(angle = 60, hjust = 1, size = 10),
        axis.ticks.y = element_blank(), legend.title = element_blank(),
        legend.position = "bottom", strip.text = element_text(size = 12),
        legend.key.size = unit(0.4, "cm"),
        legend.text = element_text(size = 14))

ggsave(filename = "wr_acac_lineplots.png", plot = wr_acac_lineplots, 
       width = 14, height = 11, dpi = 320)

ggsave(filename = "wr_acrem_lineplots.png", plot = wr_acrem_lineplots, 
       width = 14, height = 11, dpi = 320)

ggsave(filename = "wr_remrem_lineplots.png", plot = wr_remrem_lineplots, 
       width = 14, height = 11, dpi = 320)

ggsave(filename = "wr_remac_lineplots.png", plot = wr_remac_lineplots, 
       width = 14, height = 11, dpi = 320)

```


## BIOPROJECT 1 WORKFLOW
```{r}

ps_1 <- ps %>% ps_filter(bioproject == "1")

meta_1 <- sample_data(ps_1)

sample_names_1 <- rownames(meta_1)
  
sample_names_1 

seqtab.nochim_737472 <- readRDS("seqtab.nochim_737472.rds")

rownames(seqtab.nochim_737472) <- sub("^\\d+\\.", "", rownames(seqtab.nochim_737472))

seqtab_filtered_1 <- seqtab.nochim_737472[sample_names_1, ]

seqs_1 <- getSequences(seqtab_filtered_1)
names(seqs_1) <- seqs_1

alignment_1 <- AlignSeqs(DNAStringSet(seqs_1), anchor=NA)
saveRDS(alignment_1, "alignment_1.rds")
# alignment_1 <- readRDS("alignment_1.rds")


phang_align_1 <- phyDat(as(alignment_1, "matrix"), type="DNA")
saveRDS(phang_align_1, "phang_align_1.rds")
# phang_align_1 <- readRDS("phang_align_1.rds")


## Check kind of distance
dm_1 <- dist.ml(phang_align_1)

saveRDS(dm_1, "dm_1.rds")
# dm_1 <- readRDS("dm_1.rds")


treeNJ_1 <- NJ(dm_1)
saveRDS(treeNJ_1, "treeNJ_1.rds")
# treeNJ_1 <- readRDS("treeNJ_1.rds")

```

```{r}

ps_sum_1 <- summarize_phyloseq(ps_1)
summary_ps_table_1 <- data.frame(do.call(rbind, ps_sum_1[c(1:10)]))

write.csv(summary_ps_table_1, "Results/summary_ps_table_1.csv", row.names = FALSE)

summary_ps_table_1

```

```{r}

head(ps_1@otu_table, n=10) %>% kbl() %>% 
  kableExtra::kable_paper(font_size = 12) %>% 
  scroll_box(width = "800px", height = "300px")

head(ps_1@tax_table, n=10) %>% kbl() %>%
  kableExtra::kable_paper(font_size = 12) %>%
  scroll_box(width = "800px", height = "300px")

metadata_x_1 <- ps_1 %>% meta()
head(metadata_x_1, n=10) %>% kbl() %>%
  kableExtra::kable_paper(font_size = 12) %>%
  scroll_box(width = "800px", height = "300px")

ggtree(ps_1@phy_tree, options(ignore.negative.edge=TRUE))

```


```{r}
Chloroplast_1 <- ntaxa(subset_taxa(ps_1, Order == "Chloroplast"))

Mitochondria_1 <- ntaxa(subset_taxa(ps_1, Family == "Mitochondria"))

chl_mt_df_1 <- data.frame(Taxa = c("Chloroplast", "Mitochondria"),
                       nASVs = c(Chloroplast_1, Mitochondria_1)) %>%
  mutate(fraction = round(nASVs/ntaxa(ps_1), 3))
chl_mt_df_1 

ps_filt_1 <- subset_taxa(ps_1, Order != "Chloroplast")
ps_filt_1 <- subset_taxa(ps_filt_1, Family != "Mitochondria")

ps_filt_1@tax_table %>% 
  data.frame() %>% 
  group_by(Kingdom) %>% 
  dplyr::count() %>%
  mutate(fraction = round(n/ntaxa(ps_filt_1), 3))

ps_filt_1 <- subset_taxa(ps_filt_1, Kingdom == "Bacteria")

phylum_filt_table_1 <- ps_filt_1@tax_table %>% 
  data.frame() %>% 
  group_by(Phylum) %>% 
  dplyr::count() %>% 
  arrange(desc(n))

phylum_filt_table_1

write.csv(phylum_filt_table_1, "Results/phylum_filt_table_1.csv", row.names = FALSE)

ps_filt_sum_1 <- summarize_phyloseq(ps_filt_1)
summary_ps_filt_table_1 <- data.frame(do.call(rbind, ps_filt_sum_1[c(1:10)]))
write.csv(summary_ps_filt_table_1, "Results/summary_ps_filt_table_1.csv", row.names = FALSE)

summary_ps_filt_table_1

```

```{r}

ps_filt_fixed_1 <- ps_filt_1 %>%
 tax_fix(
  min_length = 4,
  sep="_", anon_unique=TRUE, 
  suffix_rank="classified")

ps_filt_fixed_1@tax_table

ggtree(ps_filt_fixed_1@phy_tree, options(ignore.negative.edge=TRUE))

  
```


```{r}

sum_of_reads_1 <- as.data.frame(sample_sums(ps_filt_fixed_1@otu_table)) %>% rownames_to_column(var = "samples") %>%
 dplyr::rename(sum_of_reads_1 = `sample_sums(ps_filt_fixed_1@otu_table)`)%>%
  left_join((metadata_x_1 %>% rownames_to_column(var = "samples")), by = "samples")
head((sum_of_reads_1 %>% arrange(sum_of_reads_1)), n=20)%>% kbl() %>%
  kableExtra::kable_paper(font_size = 12) %>%
  scroll_box(width = "800px", height = "300px")

range(sum_of_reads_1$sum_of_reads)

```

```{r}
sum_of_reads_plot_1 <- sum_of_reads_1 %>% arrange(sum_of_reads_1) %>% 
  ggplot(aes(x = reorder(samples, sum_of_reads_1), y = sum_of_reads_1, fill = group)) +
  geom_point(shape = 21, size = 2, color = "black")+
  scale_fill_manual(values = c(
    "active-active" = "darkred",
    "active-remission" = "orange", 
    "healthy-healthy" = "darkgreen"
  ))+
  labs(x = "samples")+
  theme_bw()+
  theme(axis.text.x=element_text(angle=90))

ggsave("Results/SumOfReads_1_plot.png", sum_of_reads_plot_1, device = png, width = 10, height = 8)

sum_of_reads_plot_1
```


```{r}
ps_filt_fixed_gen_rel_1 <- ps_filt_fixed_1 %>% 
  tax_transform("compositional", rank = "Genus")

ps_filt_fixed_gen_rel_1@sam_data$group <- as.factor(ps_filt_fixed_gen_rel_1@sam_data$group)

ps_ff_gen_rel_end_1 <- ps_filt_fixed_gen_rel_1 %>% ps_filter(!is.na(disease_status_end))
top_taxa_end_1 <- tax_top(ps_filt_fixed_gen_rel_1, n=40)

melt_genus_rel_end_1 <- ps_filt_fixed_gen_rel_1 %>% psmelt() %>% 
  mutate(Genus_top = case_when(Genus %in% top_taxa_end_1 ~ Genus, 
                               TRUE ~ "other")) %>% 
  mutate(Genus_top = as.factor(Genus_top)) %>% 
  mutate(Genus_top = fct_relevel(Genus_top, c(top_taxa_end_1, "other"))) %>% 
  filter(!is.na(disease_status_end))

main_genus_end_1 <- melt_genus_rel_end_1 %>%
  ggplot(aes(x = Sample, y = Abundance, fill = Genus_top)) +
  geom_bar(stat = "identity", color = "black")+
  scale_fill_manual(values = distinct_palette(40, pal = "brewerPlus", add = "grey70"))+
  theme_bw()+
  labs(title= "Genus Bar plot for end samples") +
  theme(axis.title.x = element_blank(), legend.position = "right")

ps_ff_gen_rel_base_1 <- ps_filt_fixed_gen_rel_1 %>% ps_filter(!is.na(baseline_status))
top_taxa_base_1 <- tax_top(ps_ff_gen_rel_base_1, n=40)

melt_genus_rel_base_1 <- ps_filt_fixed_gen_rel_1 %>% psmelt() %>% 
  mutate(Genus_top = case_when(Genus %in% top_taxa_base_1~ Genus, 
                               TRUE ~ "other")) %>% 
  mutate(Genus_top = as.factor(Genus_top)) %>% 
  mutate(Genus_top = fct_relevel(Genus_top, c(top_taxa_base_1, "other"))) %>% 
  filter(!is.na(baseline_status))

main_genus_base_1 <- melt_genus_rel_base_1 %>%
  ggplot(aes(x = Sample, y = Abundance, fill = Genus_top)) +
  geom_bar(stat = "identity", color = "black")+
  scale_fill_manual(values = distinct_palette(40, pal = "brewerPlus", add = "grey70"))+
  theme_bw()+
  labs(title= "Genus Bar plot for baseline samples") +
  theme(axis.title.x = element_blank(), legend.position = "right")

ps_ff_pat_gen_rel_end_1 <- ps_filt_fixed_gen_rel_1 %>% ps_filter(!is.na(disease_status_end) & donor == "no")
top_taxa_pat_end_1 <- tax_top(ps_ff_pat_gen_rel_end_1, n=40)

melt_genus_rel_end_pat_1 <- ps_filt_fixed_gen_rel_1 %>% psmelt() %>% 
  mutate(Genus_top = case_when(Genus %in% top_taxa_pat_end_1 ~ Genus, 
                               TRUE ~ "other")) %>% 
  mutate(Genus_top = as.factor(Genus_top)) %>% 
  mutate(Genus_top = fct_relevel(Genus_top, c(top_taxa_pat_end_1, "other"))) %>% 
  filter(!is.na(disease_status_end)) %>% 
  filter(donor == "no")

main_genus_end_pat_1 <-melt_genus_rel_end_patient_1 %>%
  ggplot(aes(x = Sample, y = Abundance, fill = Genus_top)) +
  geom_bar(stat = "identity", color = "black")+
  scale_fill_manual(values = distinct_palette(40, pal = "brewerPlus", add = "grey70"))+
  theme_bw()+
  labs(title= "Genus Bar plot for end Patient samples") +
  theme(axis.title.x = element_blank(), legend.position = "right")

ps_ff_pat_gen_rel_base_1 <- ps_filt_fixed_gen_rel_1 %>% ps_filter(!is.na(baseline_status) & donor == "no")
top_taxa_pat_base_1 <- tax_top(ps_ff_pat_gen_rel_base_1, n=40)
melt_genus_rel_base_patient_1 <- ps_filt_fixed_gen_rel_1 %>% psmelt() %>% 
  mutate(Genus_top = case_when(Genus %in% top_taxa_pat_base_1 ~ Genus, 
                               TRUE ~ "other")) %>% 
  mutate(Genus_top = as.factor(Genus_top)) %>% 
  mutate(Genus_top = fct_relevel(Genus_top, c(top_taxa_pat_base_1, "other"))) %>% 
  filter(!is.na(baseline_status)) %>% 
  filter(donor == "no")

main_genus_base_pat_1 <- melt_genus_rel_base_patient_1 %>%
  ggplot(aes(x = Sample, y = Abundance, fill = Genus_top)) +
  geom_bar(stat = "identity", color = "black")+
  scale_fill_manual(values = distinct_palette(40, pal = "brewerPlus", add = "grey70"))+
  theme_bw()+
  labs(title= "Genus Bar plot for baseline Patient samples") +
  theme(axis.title.x = element_blank(), legend.position = "right")


ggsave("Results/main_genus_end_1_barplot.png", main_genus_end_1, device = png, width = 10, height = 8)
ggsave("Results/main_genus_base_1_barplot.png", main_genus_base_1, device = png, width = 10, height = 8)
ggsave("Results/main_genus_end_pat_1_barplot.png", main_genus_end_pat_1, device = png, width = 10, height = 8)
ggsave("Results/main_genus_base_pat_1_barplot.png", main_genus_base_pat_1, device = png, width = 10, height = 8)

main_genus_end_1
main_genus_base_1
main_genus_end_pat_1
main_genus_base_pat_1

```

## Group by donor, plot end  
```{r}
ps_ff_don_gen_rel_1 <- ps_filt_fixed_gen_rel_1 %>% ps_filter(donor == "yes")
top_taxa_don_1 <- tax_top(ps_ff_don_gen_rel_1, n=40)

melt_genus_rel_healthy_1 <- ps_filt_fixed_gen_rel_1 %>% psmelt() %>% 
  mutate(Genus_top = case_when(Genus %in% top_taxa_don_1 ~ Genus, 
                               TRUE ~ "other")) %>% 
  mutate(Genus_top = as.factor(Genus_top)) %>% 
  mutate(Genus_top = fct_relevel(Genus_top, c(top_taxa_don_1, "other"))) %>% 
  filter(donor == "yes")

genus_healthy_barplot_donorid <- melt_genus_rel_healthy_1 %>%
  ggplot(aes(x = Sample, y = Abundance, fill = Genus_top)) +
  geom_bar(stat = "identity", color = "black")+
  facet_wrap("host_subject_id", scales = "free_x") + 
  scale_fill_manual(values = distinct_palette(40, pal = "brewerPlus", add = "grey70"))+
  theme_bw()+
  labs(title= "Genus Bar plot for Healthy Donor samples grouped by DonorID") +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank(), legend.position = "right")

melt_genus_rel_end_pat_1 <- ps_filt_fixed_gen_rel_1 %>% psmelt() %>% 
  mutate(Genus_top = case_when(Genus %in% top_taxa_pat_end_1 ~ Genus, 
                               TRUE ~ "other")) %>% 
  mutate(Genus_top = as.factor(Genus_top)) %>% 
  mutate(host_assigned_donor_subject_id = as.factor(host_assigned_donor_subject_id)) %>% 
  mutate(Genus_top = fct_relevel(Genus_top, c(top_taxa_pat_end_1, "other"))) %>% 
  filter(!is.na(disease_status_end)) %>% 
  filter(donor == "no")

genus_pat_barplot_donorpaired <- melt_genus_rel_end_patient_1 %>%
  ggplot(aes(x = Sample, y = Abundance, fill = Genus_top)) +
  geom_bar(stat = "identity", color = "black")+
  facet_wrap("host_assigned_donor_subject_id", scales = "free_x") + 
  scale_fill_manual(values = distinct_palette(40, pal = "brewerPlus", add = "grey70"))+
  scale_x_discrete(guide = guide_axis(n.dodge = 2)) +
  theme_bw()+
  labs(title= "Genus Bar plot for end Patient samples grouped by Donor") +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank(), legend.position = "right")

ggsave("Results/genus_healthy_barplot_donorid.png", genus_healthy_barplot_donorid, width = 13, height = 8, dpi = 300)
ggsave("Results/genus_pat_barplot_donorpaired.png", genus_pat_barplot_donorpaired, width = 15, height = 8, dpi = 300)

genus_healthy_barplot_donorid
genus_pat_barplot_donorpaired

```




```{r}
# Define a function to create bar plots for each donor separately
create_donorpair_barplot <- function(ps_obj, donor_id) {
  # Subset phyloseq object for the specific donor
  ps_subset <- ps_obj %>% ps_filter(host_assigned_donor_subject_id == donor_id)

  # Extract metadata
  metadata <- sample_data(ps_subset) %>% as.data.frame()

  # Create new combined labels: "host_subject_id (group)"
  metadata$facet_label <- paste(metadata$host_subject_id, "(", metadata$group, ")", sep = "")

  # Update sample data with new facet labels
  sample_data(ps_subset) <- metadata

  # Generate barplot
  barplot <- ps_subset %>%
    ps_mutate(host_sex = as.factor(host_sex)) %>%
    ps_mutate(host_assigned_donor_subject_id = as.factor(host_assigned_donor_subject_id)) %>%
    ps_arrange(host_assigned_donor_subject_id, group, week) %>%
    comp_barplot(group_by = "host_assigned_donor_subject_id", tax_level = "Genus",
                 tax_transform_for_plot = "compositional", n_taxa = 11,
                 bar_width = 0.8, merge_other = FALSE, label = "week",
                 sample_order = "default")

  # Wrap the plot and apply facet labels
  wrapped_plot <- wrap_plots(barplot) +
    plot_layout(guides = "collect") &
    facet_wrap(~facet_label, scales = "free_x") &  # Use the new combined label
    theme(axis.text.x = element_text(angle = 60, hjust = 1, size = 10),
          axis.ticks.y = element_blank(), legend.title = element_blank(),
          legend.position = "bottom", strip.text = element_text(size = 12),
          legend.key.size = unit(0.4, "cm"),
          legend.text = element_text(size = 10))

  return(wrapped_plot)
}

# Create plots for each donor
plot_donorpaired_0044 <- create_donorpair_barplot(ps_filt_fixed_gen_rel_1, "D0044")
plot_donorpaired_0097 <- create_donorpair_barplot(ps_filt_fixed_gen_rel_1, "D0097")
plot_donorpaired_0485 <- create_donorpair_barplot(ps_filt_fixed_gen_rel_1, "D0485")

# Save plots with updated labels
ggsave("plot_donorpaired_0044_barplot.png", plot_donorpaired_0044, width = 10, height = 8, dpi = 300)
ggsave("plot_donorpaired_0097_barplot.png", plot_donorpaired_0097, width = 10, height = 8, dpi = 300)
ggsave("plot_donorpaired_0485_barplot.png", plot_donorpaired_0485, width = 10, height = 8, dpi = 300)

# Display plots
plot_donorpaired_0044
plot_donorpaired_0097
plot_donorpaired_0485

```


## Check for singletons and doubletons
```{r}
length(which(taxa_sums(ps_filt_fixed_1) <= 1))

length(which(taxa_sums(ps_filt_fixed_1) <= 2))

```


```{r}

taxa_sums(ps_filt_fixed_1) %>%
  data.frame() %>%
  dplyr::rename(sum = ".") %>%
  arrange(sum)%>%
  tail(n=20) %>% kbl() %>%
  kableExtra::kable_paper(font_size = 12) %>%
  scroll_box(width = "200px", height = "300px")

```

## Normalization and check distribution
```{r}

ps_filt_fixed_rel_1 <- tax_transform(ps_filt_fixed_1, trans = "compositional")
head(ps_filt_fixed_rel_1@otu_table, n=10) 

```


## Transformation
```{r}

ps_filt_fixed_rel_clr_1 <- tax_transform(ps_filt_fixed_rel_1, trans = "clr", chain = TRUE)
head(ps_filt_fixed_rel_clr_1@otu_table, n=10)


```

## Aggregate data at specific taxonomic rank: Phylum and Genus
```{r}

ps_filt_fixed_phylum_1 <- tax_agg(ps_filt_fixed_1, rank = "Phylum")
ps_filt_fixed_class_1 <- tax_agg(ps_filt_fixed_1, rank = "Class")
ps_filt_fixed_order_1 <- tax_agg(ps_filt_fixed_1, rank = "Order")
ps_filt_fixed_family_1 <- tax_agg(ps_filt_fixed_1, rank = "Family")
ps_filt_fixed_genus_1 <- tax_agg(ps_filt_fixed_1, rank = "Genus")
#ps_filt_fixed_species_1 <- tax_agg(ps_filt_fixed_1, rank = "Species")

print(ps_filt_fixed_phylum_1)
print(ps_filt_fixed_genus_1)


summary_agg_1 <- data.frame(Taxa = c(colnames(ps_filt_fixed_1@tax_table)[c(2:6)], "ASVs"),
                 number_of_taxa=c(dim(ps_filt_fixed_phylum_1@otu_table)[2], dim(ps_filt_fixed_class_1@otu_table)[2], dim(ps_filt_fixed_order_1@otu_table)[2], dim(ps_filt_fixed_family_1@otu_table)[2], dim(ps_filt_fixed_genus_1@otu_table)[2], dim(ps_filt_fixed_1@otu_table)[2]))
print(summary_agg_1)

```


## Data exploration
```{r}
barplots_group <- ps_filt_fixed_1 %>%
  ps_mutate(host_sex = as.factor(host_sex)) %>%
  ps_mutate(group = as.factor(group)) %>%
  ps_arrange(group, week) %>% 
  comp_barplot(group_by = "group", tax_level = "Phylum", 
               tax_transform_for_plot = "compositional", n_taxa = 11, 
               bar_width = 0.8, merge_other = FALSE, label = "week", 
               sample_order = "default")

barplots_group_wrapped <- wrap_plots(barplots_group) +
  plot_layout(guides = "collect", nrow=2)

groups_barplot <- barplots_group_wrapped&
  facet_wrap("host_subject_id", scales = "free_x")& 
  theme(axis.text.x = element_text(angle = 60, hjust = 1, size = 10), 
        axis.ticks.y = element_blank(), leg.end.title = element_blank(), 
        legend.position = "bottom", strip.text = element_text(size=12),
        legend.key.size = unit(0.4, "cm"),
        legend.text = element_text(size=10))

groups_barplot

ggsave("groups_barplot.png", groups_barplot, width = 10, height = 8, dpi = 300)
```


```{r}

barplots_genus_group <- ps_filt_fixed_1 %>%
  ps_mutate(host_sex = as.factor(host_sex)) %>%
  ps_mutate(group = as.factor(group)) %>%
  ps_arrange(group, week) %>% 
  comp_barplot(group_by = "group", tax_level = "Genus", 
               tax_transform_for_plot = "compositional", n_taxa = 11, 
               bar_width = 0.8, merge_other = FALSE, label = "week", 
               sample_order = "default")

barplots_genus_group_wrapped <- wrap_plots(barplots_genus_group) +
  plot_layout(guides = "collect", nrow=2)

groups_genus_barplot <- barplots_genus_group_wrapped&
  facet_wrap("host_subject_id", scales = "free_x")& 
  theme(axis.text.x = element_text(angle = 60, hjust = 1, size = 10), 
        axis.ticks.y = element_blank(), leg.end.title = element_blank(), 
        legend.position = "bottom", strip.text = element_text(size=12),
        legend.key.size = unit(0.4, "cm"),
        legend.text = element_text(size=10))

groups_genus_barplot

ggsave("groups_genus_barplot.png", groups_genus_barplot, width = 10, height = 8, dpi = 300)

```


```{r}
# Define a function to create bar plots for each group separately
create_group_barplot <- function(ps_obj, group_name) {
  ps_subset <- ps_obj %>% ps_filter(group == group_name)
  
  barplot <- ps_subset %>%
    ps_mutate(host_sex = as.factor(host_sex)) %>%
    ps_mutate(group = as.factor(group)) %>%
    ps_arrange(group, week) %>%
    comp_barplot(group_by = "group", tax_level = "Phylum",
                 tax_transform_for_plot = "compositional", n_taxa = 11,
                 bar_width = 0.8, merge_other = FALSE, label = "week",
                 sample_order = "default")
  
  wrapped_plot <- wrap_plots(barplot) +
    plot_layout(guides = "collect") &
    facet_wrap("host_subject_id", scales = "free_x") &
    theme(axis.text.x = element_text(angle = 60, hjust = 1, size = 10),
          axis.ticks.y = element_blank(), legend.title = element_blank(),
          legend.position = "bottom", strip.text = element_text(size = 12),
          legend.key.size = unit(0.4, "cm"),
          legend.text = element_text(size = 10))
  
  return(wrapped_plot)
}

# Create plots for each group
plot_1 <- create_group_barplot(ps_filt_fixed_1, "active-active")
plot_2 <- create_group_barplot(ps_filt_fixed_1, "active-remission")
plot_3 <- create_group_barplot(ps_filt_fixed_1, "healthy-healthy")

# Save plots as separate images
ggsave("ac-ac_barplot.png", plot_1, width = 10, height = 8, dpi = 300)
ggsave("ac-rem_barplot.png", plot_2, width = 10, height = 8, dpi = 300)
ggsave("healthy_barplot.png", plot_3, width = 10, height = 8, dpi = 300)

# If you want to visualize them:
plot_1
plot_2
plot_3

```


```{r}
create_group_barplots <- function(ps_obj, group_name, max_per_plot = 6, output_prefix = "plot") {
  ps_subset <- ps_obj %>% ps_filter(group == group_name)
  
  # Get unique subjects
  subject_ids <- unique(sample_data(ps_subset)$host_subject_id)
  
  # Split into chunks of 'max_per_plot'
  subject_batches <- split(subject_ids, ceiling(seq_along(subject_ids) / max_per_plot))
  
  # Generate plots for each batch
  plots <- list()
  
  for (i in seq_along(subject_batches)) {
    subjects <- subject_batches[[i]]
    
    ps_batch <- ps_subset %>% ps_filter(host_subject_id %in% subjects)
    
    barplot <- ps_batch %>%
      ps_mutate(host_sex = as.factor(host_sex)) %>%
      ps_mutate(group = as.factor(group)) %>%
      ps_arrange(group, week) %>%
      comp_barplot(group_by = "group", tax_level = "Phylum",
                   tax_transform_for_plot = "compositional", n_taxa = 11,
                   bar_width = 0.8, merge_other = FALSE, label = "week",
                   sample_order = "default")
    
    wrapped_plot <- wrap_plots(barplot) +
      plot_layout(guides = "collect") &
      facet_wrap("host_subject_id", scales = "free_x") &
      theme(axis.text.x = element_text(angle = 60, hjust = 1, size = 10),
            axis.ticks.y = element_blank(), legend.title = element_blank(),
            legend.position = "bottom", strip.text = element_text(size = 12),
            legend.key.size = unit(0.4, "cm"),
            legend.text = element_text(size = 10))
    
    plots[[i]] <- wrapped_plot
    
    # Save each plot separately
    ggsave(filename = paste0(output_prefix, "_1_batch_", i, ".png"), plot = wrapped_plot, 
           width = 10, height = 8, dpi = 300)
  }
  
  return(plots)
}

plots_ac_ac_1 <- create_group_barplots(ps_filt_fixed_1, "active-active", max_per_plot = 6, output_prefix = "ac_ac_barplot")
plots_ac_rem_1 <- create_group_barplots(ps_filt_fixed_1, "active-remission", max_per_plot = 6, output_prefix = "ac_rem_barplot")

plots_ac_ac_1

plots_ac_rem_1
```



```{r}
# Define a function to create bar plots for each group separately
create_genus_group_barplot <- function(ps_obj, group_name) {
  ps_subset <- ps_obj %>% ps_filter(group == group_name)
  
  barplot <- ps_subset %>%
    ps_mutate(host_sex = as.factor(host_sex)) %>%
    ps_mutate(group = as.factor(group)) %>%
    ps_arrange(group, week) %>%
    comp_barplot(group_by = "group", tax_level = "Genus",
                 tax_transform_for_plot = "compositional", n_taxa = 11,
                 bar_width = 0.8, merge_other = FALSE, label = "week",
                 sample_order = "default")
  
  wrapped_plot <- wrap_plots(barplot) +
    plot_layout(guides = "collect") &
    facet_wrap("host_subject_id", scales = "free_x") &
    theme(axis.text.x = element_text(angle = 60, hjust = 1, size = 10),
          axis.ticks.y = element_blank(), legend.title = element_blank(),
          legend.position = "bottom", strip.text = element_text(size = 12),
          legend.key.size = unit(0.4, "cm"),
          legend.text = element_text(size = 10))
  
  return(wrapped_plot)
}

# Create plots for each group
plot_genus_1 <- create_genus_group_barplot(ps_filt_fixed_1, "active-active")
plot_genus_2 <- create_genus_group_barplot(ps_filt_fixed_1, "active-remission")
plot_genus_3 <- create_genus_group_barplot(ps_filt_fixed_1, "healthy-healthy")

# Save plots as separate images
ggsave("ac-ac_genus_barplot.png", plot_genus_1, width = 10, height = 8, dpi = 300)
ggsave("ac-rem_genus_barplot.png", plot_genus_2, width = 10, height = 8, dpi = 300)
ggsave("healthy_genus_barplot.png", plot_genus_3, width = 10, height = 8, dpi = 300)

# If you want to visualize them:
plot_genus_1
plot_genus_2
plot_genus_3


```

```{r}
create_group_barplots <- function(ps_obj, group_name, max_per_plot = 6, output_prefix = "plot") {
  ps_subset <- ps_obj %>% ps_filter(group == group_name)
  
  # Get unique subjects
  subject_ids <- unique(sample_data(ps_subset)$host_subject_id)
  
  # Split into chunks of 'max_per_plot'
  subject_batches <- split(subject_ids, ceiling(seq_along(subject_ids) / max_per_plot))
  
  # Generate plots for each batch
  plots <- list()
  
  for (i in seq_along(subject_batches)) {
    subjects <- subject_batches[[i]]
    
    ps_batch <- ps_subset %>% ps_filter(host_subject_id %in% subjects)
    
    barplot <- ps_batch %>%
      ps_mutate(host_sex = as.factor(host_sex)) %>%
      ps_mutate(group = as.factor(group)) %>%
      ps_arrange(group, week) %>%
      comp_barplot(group_by = "group", tax_level = "Genus",
                   tax_transform_for_plot = "compositional", n_taxa = 11,
                   bar_width = 0.8, merge_other = FALSE, label = "week",
                   sample_order = "default")
    
    wrapped_plot <- wrap_plots(barplot) +
      plot_layout(guides = "collect") &
      facet_wrap("host_subject_id", scales = "free_x") &
      theme(axis.text.x = element_text(angle = 60, hjust = 1, size = 10),
            axis.ticks.y = element_blank(), legend.title = element_blank(),
            legend.position = "bottom", strip.text = element_text(size = 12),
            legend.key.size = unit(0.4, "cm"),
            legend.text = element_text(size = 10))
    
    plots[[i]] <- wrapped_plot
    
    # Save each plot separately
    ggsave(filename = paste0(output_prefix, "_1_genus_batch_", i, ".png"), plot = wrapped_plot, 
           width = 10, height = 8, dpi = 300)
  }
  
  return(plots)
}

# Generate and save plots
plots_genus_ac_ac_1 <- create_group_barplots(ps_filt_fixed_1, "active-active", max_per_plot = 6, output_prefix = "ac_ac_barplot")
plots_genus_ac_rem_1 <- create_group_barplots(ps_filt_fixed_1, "active-remission", max_per_plot = 6, output_prefix = "ac_rem_barplot")

plots_genus_ac_ac_1

plots_genus_ac_rem_1
```



```{r}
# cols_hsi<- pal_nejm("default")(17)
# names(cols_hsi) <- unique(samdat_tbl(ps_filt_fixed_1)$host_subject_id)
# cols_group <- c("darkred", "orange", "darkgreen")
# names(cols_group) <- unique(samdat_tbl(ps_filt_fixed_1)$group)
# cols_sex<- c("black", "grey", "lightgreen")
# names(cols_sex) <- unique(samdat_tbl(ps_filt_fixed_1)$host_sex)
# cols_week<- distinct_palette(22, pal = "brewerPlus")
# names(cols_week) <- unique(samdat_tbl(ps_filt_fixed_1)$week)
# 
# # sample_anno_test <- sampleAnnotation(
# #   Week = anno_sample("week"),
# #   Group = anno_sample("group"),
# #   Sex = anno_sample("host_sex"),
# #   HSI = anno_sample("host_subject_id"),
# #   col = list(HSI = cols_hsi, Group = cols_group, Sex = cols_sex, Week = cols_week),
# #   border = FALSE
# # )
# # 
# # print(sample_anno_test)
# 
# 
# ps_filt_fixed_1 %>%
#   tax_transform("compositional", rank = "Phylum") %>%
#   comp_heatmap(colors = heat_palette(sym = FALSE, "Blue-Red 3"),
#                tax_anno = taxAnnotation(
#       Prev. = anno_tax_prev(bar_width = 0.3, size = grid::unit(1, "cm"))
#     ),
#     sample_anno = sampleAnnotation(
#       Week = anno_sample(var = "week"),
#       Group = anno_sample(var = "group"),
#       Sex = anno_sample(var = "host_sex"),
#       HSI = anno_sample(var = "host_subject_id"),
#       col = list(HSI = cols_hsi, Group = cols_group, Sex = cols_sex, Week = cols_week), border = FALSE))


```



## Alpha diversity
```{r}

alpha_div_1 <- estimate_richness(ps_filt_fixed_1, measures = c("Observed", "Shannon", "InvSimpson", "Chao1"))
PD_1 <- calculatePD(ps_filt_fixed_1, justDF = TRUE, include_root = FALSE) %>% pull(PD)
alpha_div_1 <- alpha_div_1 %>% mutate(PD = PD_1)
rownames(alpha_div_1) <- gsub("^X", "", rownames(alpha_div_1))
head(alpha_div_1, n=20)

metadata_samples_alpha_1 <- bind_cols(meta_737472, alpha_div_1)
metadata_samples_long_1 <- metadata_samples_alpha_1 %>% pivot_longer(cols= Observed:PD, names_to = "alpha_div", values_to = "value")
kruskal_test_alpha_1 <- metadata_samples_long_1 %>% group_by(alpha_div) %>%
    rstatix::kruskal_test(value ~ group) %>% mutate(p_adj_BH = p.adjust(p, method = "BH"))

kruskal_test_alpha_1

```

```{r}
dunn_test_alpha_1 <- metadata_samples_long_1 %>% group_by(alpha_div) %>%
    rstatix::dunn_test(value ~ group, p.adjust.method = "BH") %>%
  add_xy_position(x = "group", fun = "mean_sd", step.increase = 0.05, scales = "free")
dunn_test_alpha_1 %>% kbl() %>%
  kableExtra::kable_paper(font_size = 12)


```

```{r}

ggplot(metadata_samples_long_1, aes(x = group, y = value)) +
  geom_boxplot() +
  geom_jitter(shape = 21, color = "black", size = 2, aes(fill = group)) +
  facet_wrap(~alpha_div, scales = "free", ncol = 3) +
  scale_fill_nejm() +
  stat_pvalue_manual(data = dunn_test_alpha_1, 
                     label = "p.adj.signif", tip.length = 0.01, bracket.shorten = 0.05,
                     hide.ns = TRUE, size = 4) +
  scale_x_discrete(guide = guide_axis(n.dodge = 2)) +
  theme_bw() +
  theme(axis.text.x = element_text(size = 10), legend.position = "none")

```



```{r}

metadata_samples_alpha_1

kruskal_chao1_1 <- kruskal.test(Chao1 ~ group, data = metadata_samples_alpha_1)
kruskal_shannon_1 <- kruskal.test(Shannon ~ group, data = metadata_samples_alpha_1)
kruskal_invsimps <- kruskal.test(InvSimpson ~ group, data = metadata_samples_alpha_1)
kruskal_obsv_1 <- kruskal.test(Observed ~ group, data = metadata_samples_alpha_1)
kruskal_pd_1 <- kruskal.test(PD ~ group, data = metadata_samples_alpha_1)


# Extract values
H_chao1_1 <- kruskal_chao1_1$statistic
p_chao1_1 <- kruskal_chao1_1$p.value

H_shannon_1 <- kruskal_shannon_1$statistic
p_shannon_1 <- kruskal_shannon_1$p.value

H_invsimps_1 <- kruskal_invsimps$statistic
p_invsimps_1 <- kruskal_invsimps$p.value

H_obsv_1 <- kruskal_obsv_1$statistic
p_obsv_1 <- kruskal_obsv_1$p.value

H_pd_1 <- kruskal_pd_1$statistic
p_pd_1 <- kruskal_pd_1$p.value

# Print results
cat("Kruskal-Wallis test for Chao1 index: H =", round(H_chao1_1, 3), "p =", round(p_chao1_1, 3), "\n")
cat("Kruskal-Wallis test for Shannon index: H =", round(H_shannon_1, 3), "p =", round(p_shannon_1, 3), "\n")
cat("Kruskal-Wallis test for Inverse Simpson index: H =", round(H_invsimps_1, 3), "p =", round(p_invsimps_1, 3), "\n")
cat("Kruskal-Wallis test for Observed Number of Species: H =", round(H_obsv_1, 3), "p =", round(p_obsv_1, 3), "\n")
cat("Kruskal-Wallis test for Faith's Phylogenetic Diversity: H =", round(H_pd_1, 3), "p =", round(p_pd_1, 3), "\n")


# Filter for Chao1 and Shannon indices
dunn_chao1_1 <- dunn_test_alpha_1 %>% filter(alpha_div == "Chao1")
dunn_shannon_1 <- dunn_test_alpha_1 %>% filter(alpha_div == "Shannon")
dunn_invsimps_1 <- dunn_test_alpha_1 %>% filter(alpha_div == "InvSimpson")
dunn_obsv_1 <- dunn_test_alpha_1 %>% filter(alpha_div == "Observed")
dunn_pd_1 <- dunn_test_alpha_1 %>% filter(alpha_div == "PD")

# Show significant results (p < 0.05)
significant_chao1_1 <- dunn_chao1_1 %>% filter(p.adj < 0.05)
significant_shannon_1 <- dunn_shannon_1 %>% filter(p.adj < 0.05)
significant_invsimps_1 <- dunn_invsimps_1 %>% filter(p.adj < 0.05)
significant_obsv_1 <- dunn_obsv_1 %>% filter(p.adj < 0.05)
significant_pd_1 <- dunn_pd_1 %>% filter(p.adj < 0.05)

# Print significant comparisons
print(significant_chao1_1)
print(significant_shannon_1)
print(significant_invsimps_1)
print(significant_obsv_1)
print(significant_pd_1)

alpha_summary_1 <- metadata_samples_alpha_1 %>%
  group_by(group) %>%
  summarise(
    Chao1_mean = mean(Chao1, na.rm = TRUE),
    Chao1_sd = sd(Chao1, na.rm = TRUE),
    Shannon_mean = mean(Shannon, na.rm = TRUE),
    Shannon_sd = sd(Shannon, na.rm = TRUE),
    InvSimps_mean = mean(InvSimpson, na.rm = TRUE),
    InvSimps_sd = sd(InvSimpson, na.rm = TRUE), 
    Obsv_mean = mean(Observed, na.rm = TRUE),
    Obsv_sd = sd(Observed, na.rm = TRUE),
    PD_mean = mean(PD, na.rm = TRUE),
    PD_sd = sd(PD, na.rm = TRUE)
  )

print(alpha_summary_1)

write.csv(alpha_summary_1, "Results/alpha_summary_1_table.csv", row.names = FALSE)

```


## Rarefied

```{r}
library(ranacapa); packageVersion("ranacapa")

ps_filt_fixed_1 %>% 
  ggrare(step = 50, color = "group", se = FALSE)

```


```{r}
set.seed(19950622)

ps_ff_rare_1 <- rarefy_even_depth(ps_filt_fixed_1, sample.size = 24190, replace = FALSE)

ps_ff_rare_1 %>% 
  ggrare(step = 50, color = "group", se = FALSE)

alpha_div_rare_1 <- estimate_richness(ps_ff_rare_1, measures = c("Observed", "Shannon", "InvSimpson", "Chao1"))
PD_rare_1 <- calculatePD(ps_ff_rare_1, justDF = TRUE, include_root = FALSE) %>% pull(PD)
alpha_div_rare_1 <- alpha_div_rare_1 %>% mutate(PD = PD_rare_1)
rownames(alpha_div_rare_1) <- gsub("^X", "", rownames(alpha_div_rare_1))
head(alpha_div_rare_1, n=20)

meta_rare_1 <- as.data.frame(as.matrix(sample_data(ps_ff_rare_1)))

metadata_samples_alpha_1_rare <- bind_cols(meta_rare_1, alpha_div_rare_1)
metadata_samples_long_rare_1 <- metadata_samples_alpha_1_rare %>% pivot_longer(cols= Observed:PD, names_to = "alpha_div", values_to = "value")
kruskal_test_alpha_rare_1 <- metadata_samples_long_rare_1 %>% group_by(alpha_div) %>%
  rstatix::kruskal_test(value ~ group) %>% mutate(p_adj_BH = p.adjust(p, method = "BH"))

kruskal_test_alpha_rare_1

write.csv(kruskal_test_alpha_rare_1, "Results/kruskal_test_alpha_rare_1.csv", row.names = FALSE)

dunn_test_alpha_rare_1 <- metadata_samples_long_rare_1 %>% group_by(alpha_div) %>%
  rstatix::dunn_test(value ~ group, p.adjust.method = "BH") %>%
  add_xy_position(x = "group", fun = "mean_sd", step.increase = 0.05, scales = "free")
dunn_test_alpha_rare_1 %>% kbl() %>%
  kableExtra::kable_paper(font_size = 12)

dunn_test_alpha_rare_table_1 <- apply(dunn_test_alpha_rare_1,2,as.character)
write.csv(dunn_test_alpha_rare_table_1, "Results/dunn_test_alpha_rare_table_1.csv", row.names = FALSE)

dunn_test_alpha_rare_plot_1 <- ggplot(metadata_samples_long_rare_1, aes(x = group, y = value)) +
  geom_boxplot() +
  geom_jitter(shape = 21, color = "black", size = 2, aes(fill = group)) +
  facet_wrap(~alpha_div, scales = "free", ncol = 3) +
  scale_fill_nejm() +
  stat_pvalue_manual(data = dunn_test_alpha_rare_1, 
                     label = "p.adj.signif", tip.length = 0.01, bracket.shorten = 0.05,
                     hide.ns = TRUE, size = 4) +
  scale_x_discrete(guide = guide_axis(n.dodge = 2)) +
  theme_bw() +
  theme(axis.text.x = element_text(size = 10), legend.position = "none")

ggsave("Results/dunn_test_alpha_rare_plot_1.png", dunn_test_alpha_rare_plot_1, width = 13, height = 8, dpi = 300)

dunn_test_alpha_rare_plot_1

```


```{r}

metadata_samples_alpha_1_rare

kruskal_chao1_1_rare <- kruskal.test(Chao1 ~ group, data = metadata_samples_alpha_1_rare)
kruskal_shannon_1_rare <- kruskal.test(Shannon ~ group, data = metadata_samples_alpha_1_rare)
kruskal_invsimps_rare <- kruskal.test(InvSimpson ~ group, data = metadata_samples_alpha_1_rare)
kruskal_obsv_1_rare <- kruskal.test(Observed ~ group, data = metadata_samples_alpha_1_rare)
kruskal_pd_1_rare <- kruskal.test(PD ~ group, data = metadata_samples_alpha_1_rare)


# Extract values
H_chao1_1_rare <- kruskal_chao1_1_rare$statistic
p_chao1_1_rare <- kruskal_chao1_1_rare$p.value

H_shannon_1_rare <- kruskal_shannon_1_rare$statistic
p_shannon_1_rare <- kruskal_shannon_1_rare$p.value

H_invsimps_1_rare <- kruskal_invsimps_rare$statistic
p_invsimps_1_rare <- kruskal_invsimps_rare$p.value

H_obsv_1_rare <- kruskal_obsv_1_rare$statistic
p_obsv_1_rare <- kruskal_obsv_1_rare$p.value

H_pd_1_rare <- kruskal_pd_1_rare$statistic
p_pd_1_rare <- kruskal_pd_1_rare$p.value

# Print results
cat("Kruskal-Wallis test for Chao1 index: H =", round(H_chao1_1_rare, 3), "p =", round(p_chao1_1_rare, 3), "\n")
cat("Kruskal-Wallis test for Shannon index: H =", round(H_shannon_1_rare, 3), "p =", round(p_shannon_1_rare, 3), "\n")
cat("Kruskal-Wallis test for Inverse Simpson index: H =", round(H_invsimps_1_rare, 3), "p =", round(p_invsimps_1_rare, 3), "\n")
cat("Kruskal-Wallis test for Observed Number of Species: H =", round(H_obsv_1_rare, 3), "p =", round(p_obsv_1_rare, 3), "\n")
cat("Kruskal-Wallis test for Faith's Phylogenetic Diversity: H =", round(H_pd_1_rare, 3), "p =", round(p_pd_1_rare, 3), "\n")


# Filter for Chao1 and Shannon indices
dunn_chao1_1_rare <- dunn_test_alpha_rare_1 %>% filter(alpha_div == "Chao1")
dunn_shannon_1_rare <- dunn_test_alpha_rare_1 %>% filter(alpha_div == "Shannon")
dunn_invsimps_1_rare <- dunn_test_alpha_rare_1 %>% filter(alpha_div == "InvSimpson")
dunn_obsv_1_rare <- dunn_test_alpha_rare_1 %>% filter(alpha_div == "Observed")
dunn_pd_1_rare <- dunn_test_alpha_rare_1 %>% filter(alpha_div == "PD")

# Show significant results (p < 0.05)
significant_chao1_1_rare <- dunn_chao1_1_rare %>% filter(p.adj < 0.05)
significant_shannon_1_rare <- dunn_shannon_1_rare %>% filter(p.adj < 0.05)
significant_invsimps_1_rare <- dunn_invsimps_1_rare %>% filter(p.adj < 0.05)
significant_obsv_1_rare <- dunn_obsv_1_rare %>% filter(p.adj < 0.05)
significant_pd_1_rare <- dunn_pd_1_rare %>% filter(p.adj < 0.05)

# Print significant comparisons
print(significant_chao1_1_rare)
print(significant_shannon_1_rare)
print(significant_invsimps_1_rare)
print(significant_obsv_1_rare)
print(significant_pd_1_rare)

alpha_rare_summary_1 <- metadata_samples_alpha_1_rare %>%
  group_by(group) %>%
  summarise(
    Chao1_mean = mean(Chao1, na.rm = TRUE),
    Chao1_sd = sd(Chao1, na.rm = TRUE),
    Shannon_mean = mean(Shannon, na.rm = TRUE),
    Shannon_sd = sd(Shannon, na.rm = TRUE),
    InvSimps_mean = mean(InvSimpson, na.rm = TRUE),
    InvSimps_sd = sd(InvSimpson, na.rm = TRUE), 
    Obsv_mean = mean(Observed, na.rm = TRUE),
    Obsv_sd = sd(Observed, na.rm = TRUE),
    PD_mean = mean(PD, na.rm = TRUE),
    PD_sd = sd(PD, na.rm = TRUE)
  )

print(alpha_rare_summary_1)
write.csv(alpha_rare_summary_1, "Results/alpha_rare_summary_table_1.csv", row.names = FALSE)

```


```{r}
PCoA_ASV_bray_1 <- ordinate(ps_filt_fixed_rel_1, distance = "bray", method = "PCoA")
beta_ASV_bray_1 <- phyloseq::distance(ps_filt_fixed_rel_1, "bray")
ado_bray_group_1 <- adonis2(beta_ASV_bray_1 ~ group, meta_737472)
print(ado_bray_group_1)

group.bd_1 <- betadisper(d = beta_ASV_bray_1, group = meta_737472$group, type="centroid")
bd_group_boxplot_1 <- boxplot(group.bd_1)

bd_group_boxplot_1

png("Results/beta_dispersion_bc_boxplot_1.png", width = 3500, height = 2000, res = 300)  # High resolution
boxplot(group.bd_1, main = "Beta Dispersion per Group (BC) BP1", cex.axis = 1.2, cex.lab = 1.5)
dev.off()

permutest(group.bd_1)


BC_pcoa_plot_1 <- plot_ordination(ps_filt_fixed_rel_1, PCoA_ASV_bray_1, axes = c(1,2))+
  stat_conf_ellipse(geom = "polygon", alpha = 0.1, aes(color = group,  fill = group))+
  geom_point(size = 3, shape = 21, aes(fill = group))+
  stat_mean(aes(fill = group), shape = 23, size = 4) + 
  annotate("text", x =- 0.3, y = 0.5, color = "black", label = paste0("p=", round(ado_bray_group_1$`Pr(>F)`[1], digits = 3)))+
  annotate("text", x = -0.3, y = 0.45, color = "black", label = paste0("R\u00b2=", round((ado_bray_group_1$R2[1])*100, digits = 2), "%"))+
  scale_fill_nejm(name = "Group")+
  scale_color_nejm(name = "Group")+
  theme_bw()+
  theme(legend.position = "right")

ggsave("Results/BC_pcoa_plot_1.png", BC_pcoa_plot_1, width = 10, height = 8, dpi = 300)

BC_pcoa_plot_1

ps_filt_fixed_rel@sam_data$host_sex <- as.factor(ps_filt_fixed_rel@sam_data$host_sex)

ado_bray_sex_1 <- adonis2(beta_ASV_bray_1 ~ host_sex, meta_737472)
print(ado_bray_sex_1)

BC_pcoa_sex_plot_1 <- plot_ordination(ps_filt_fixed_rel_1, PCoA_ASV_bray_1, axes = c(1,2))+
  stat_conf_ellipse(geom = "polygon", alpha = 0.1, aes(color = host_sex,  fill = host_sex))+
  geom_point(size = 3, shape = 21, aes(fill = host_sex))+
  stat_mean(aes(fill = host_sex), shape = 23, size = 4) + 
  annotate("text", x =- 0.3, y = 0.5, color = "black", label = paste0("p=", round(ado_bray_sex_1$`Pr(>F)`[1], digits = 3)))+
  annotate("text", x = -0.3, y = 0.45, color = "black", label = paste0("R\u00b2=", round((ado_bray_sex_1$R2[1])*100, digits = 2), "%"))+
  scale_fill_nejm(name = "Host sex")+
  scale_color_nejm(name = "Host Sex")+
  theme_bw()+
  theme(legend.position = "right")

ggsave("Results/BC_pcoa_sex_plot_1.png", BC_pcoa_sex_plot_1, width = 10, height = 8, dpi = 300)

BC_pcoa_sex_plot_1

bd_df_1 <- data.frame(samples = rownames(group.bd_1$vectors),
                    distances = group.bd_1$distances,
                    group = meta_737472$group)

kruskal_bc_test_1 <- bd_df_1 %>% rstatix::kruskal_test(distances ~ group)
print(kruskal_bc_test_1)

# Dunn’s test for pairwise comparisons with Benjamini-Hochberg correction
dunn_bc_test_1 <- bd_df_1 %>% rstatix::dunn_test(distances ~ group, p.adjust.method = "BH")
print(dunn_bc_test_1)

dunn_bc_test_table_1 <- apply(dunn_bc_test_1,2,as.character)
write.csv(dunn_bc_test_table_1, "Results/dunn_bc_test_table_1.csv", row.names = FALSE)

# Extract p-values for compact letter display
dunn_bc_pvalues_1 <- dunn_bc_test_1 %>% select(group1, group2, p.adj) %>%
  pivot_wider(names_from = group2, values_from = p.adj) %>%
  column_to_rownames(var = "group1")

# Convert to a matrix format for multcompLetters
dunn_bc_matrix_1 <- as.matrix(dunn_bc_pvalues_1)
# Fill the lower triangle with upper triangle values to make the matrix symmetric
dunn_bc_matrix_1[lower.tri(dunn_bc_matrix_1)] <- t(dunn_bc_matrix_1)[lower.tri(dunn_bc_matrix_1)]


# Generate compact letter display for groups
dunn_bc_labels_1 <- multcompLetters(dunn_bc_matrix_1)$Letters
print(dunn_bc_labels_1)

```

```{r}
## Set root for WU so it runs with the same root as previous wu analysis

# ASV5155

# Define the root sequence (use the exact sequence you mentioned before)
root_seq <- "ASV5155"

# Extract tree from phyloseq object
tree_1 <- phy_tree(ps_filt_fixed_rel_1)

# Set the root
tree_rooted_1 <- root(tree_1, outgroup = root_seq, resolve.root = TRUE)

# Replace the tree in the phyloseq object
phy_tree(ps_filt_fixed_rel_1) <- tree_rooted_1
# ggtree(ps_filt_fixed_rel@phy_tree, options(ignore.negative.edge=TRUE))


PCoA_ASV_wunifrac_1 <- ordinate(ps_filt_fixed_rel_1, distance = "wunifrac", method = "PCoA")
beta_ASV_wunifrac_1 <- phyloseq::distance(ps_filt_fixed_rel_1, "wunifrac")
ado_wunifrac_group_1 <- adonis2(beta_ASV_wunifrac_1 ~ group, meta_737472)
print(ado_wunifrac_group_1)


group_wunifrac.bd_1 <- betadisper(d = beta_ASV_wunifrac_1, group = meta_737472$group, type="centroid")
boxplot(group_wunifrac.bd_1)

png("Results/beta_dispersion_wu_boxplot_1.png", width = 3500, height = 2000, res = 300)  # High resolution
boxplot(group_wunifrac.bd_1, main = "Beta Dispersion per Group (WU) BP1", cex.axis = 1.2, cex.lab = 1.5)
dev.off()

permutest(group_wunifrac.bd_1)


```


```{r}
WU_pcoa_plot_1 <- plot_ordination(ps_filt_fixed_rel_1, PCoA_ASV_wunifrac_1, axes = c(1,2))+
  stat_conf_ellipse(geom = "polygon", alpha = 0.1, aes(color = group,  fill = group))+
  geom_point(size = 3, shape = 21, aes(fill = group))+
  stat_mean(aes(fill = group), shape = 23, size = 4) + 
  annotate("text", x = 0.1, y = -0.15, color = "black", label = paste0("p=", round(ado_wunifrac_group_1$`Pr(>F)`[1], digits = 3)))+
  annotate("text", x = 0.1, y = -0.13, color = "black", label = paste0("R\u00b2=", round((ado_wunifrac_group_1$R2[1])*100, digits = 2), "%"))+
  scale_fill_nejm(name = "Group")+
  scale_color_nejm(name = "Group")+
  theme_bw()+
  theme(legend.position = "right")

ggsave("Results/WU_pcoa_plot_1.png", WU_pcoa_plot_1, width = 10, height = 8, dpi = 300)


WU_pcoa_plot_1 

ado_wunifrac_sex_1 <- adonis2(beta_ASV_wunifrac_1 ~ host_sex, meta_737472)
print(ado_wunifrac_sex_1)

WU_pcoa_sex_plot_1 <- plot_ordination(ps_filt_fixed_rel_1, PCoA_ASV_wunifrac_1, axes = c(1,2))+
  stat_conf_ellipse(geom = "polygon", alpha = 0.1, aes(color = host_sex,  fill = host_sex))+
  geom_point(size = 3, shape = 21, aes(fill = host_sex))+
  stat_mean(aes(fill = host_sex), shape = 23, size = 4) + 
  annotate("text", x = 0.1, y = -0.15, color = "black", label = paste0("p=", round(ado_wunifrac_sex_1$`Pr(>F)`[1], digits = 3)))+
  annotate("text", x = 0.1, y = -0.13, color = "black", label = paste0("R\u00b2=", round((ado_wunifrac_sex_1$R2[1])*100, digits = 2), "%"))+
  scale_fill_nejm(name = "Host sex")+
  scale_color_nejm(name = "Host sex")+
  theme_bw()+
  theme(legend.position = "right")

ggsave("Results/WU_pcoa_sex_plot_1.png", WU_pcoa_sex_plot_1, width = 10, height = 8, dpi = 300)

WU_pcoa_sex_plot_1

wu_bd_df_1 <- data.frame(samples = rownames(group_wunifrac.bd_1$vectors),
                    distances = group_wunifrac.bd_1$distances,
                    group = meta_737472$group)

kruskal_wu_test_1 <- wu_bd_df %>% rstatix::kruskal_test(distances ~ group)
print(kruskal_wu_test_1)

# Dunn’s test for pairwise comparisons with Benjamini-Hochberg correction
dunn_wu_test_1 <- wu_bd_df_1 %>% rstatix::dunn_test(distances ~ group, p.adjust.method = "BH", detailed = TRUE)
print(dunn_wu_test_1)

dunn_wu_test_table_1 <- apply(dunn_wu_test_1,2,as.character)
write.csv(dunn_wu_test_table_1, "Results/dunn_wu_test_table_1.csv", row.names = FALSE)

# Extract p-values for compact letter display
dunn_wu_pvalues_1 <- dunn_wu_test_1 %>% select(group1, group2, p.adj) %>%
  pivot_wider(names_from = group2, values_from = p.adj) %>%
  column_to_rownames(var = "group1")

# Convert to a matrix format for multcompLetters
dunn_wu_matrix_1 <- as.matrix(dunn_wu_pvalues_1)
# Fill the lower triangle with upper triangle values to make the matrix symmetric
dunn_wu_matrix_1[lower.tri(dunn_wu_matrix_1)] <- t(dunn_wu_matrix_1)[lower.tri(dunn_wu_matrix_1)]


# Generate compact letter display for groups
dunn_wu_labels_1 <- multcompLetters(dunn_wu_matrix_1)$Letters
print(dunn_wu_labels_1)

```

## Same with baseline and end samples only
```{r}
## For baseline samples
ps_ff_rel_base_1 <- ps_filt_fixed_rel_1 %>% ps_filter(!is.na(baseline_status))

meta_beta_base_1 <- as.data.frame(as.matrix(sample_data(ps_ff_rel_base_1)))

PCoA_ASV_bray_base_1 <- ordinate(ps_ff_rel_base_1, distance = "bray", method = "PCoA")
PCoA_ASV_bray_base_1


beta_ASV_bray_base_1 <- phyloseq::distance(ps_ff_rel_base_1, "bray")
ado_bray_group_base_1 <- adonis2(beta_ASV_bray_base_1 ~ baseline_status, meta_beta_base_1)
print(ado_bray_group_base_1)

baseline.bd_1 <- betadisper(d = beta_ASV_bray_base_1, group = meta_beta_base_1$baseline_status, type="centroid")
bd_baseline_boxplot_1 <- boxplot(baseline.bd_1)

bd_baseline_boxplot_1

png("Results/betadisp_baseline_bc_boxplot_1.png", width = 3500, height = 2000, res = 300)  # High resolution
boxplot(baseline.bd_1, main = "Beta Dispersion per Baseline Status (BC) BP1", cex.axis = 1.2, cex.lab = 1.5)
dev.off()

permutest(baseline.bd_1)

BC_baseline_pcoa_plot_1 <- plot_ordination(ps_ff_rel_base_1, PCoA_ASV_bray_base_1, axes = c(1,2))+
  stat_conf_ellipse(geom = "polygon", alpha = 0.1, aes(color = baseline_status,  fill = baseline_status))+
  geom_point(size = 3, shape = 21, aes(fill = baseline_status))+
  stat_mean(aes(fill = baseline_status), shape = 23, size = 4) + 
  annotate("text", x =- 0.3, y = 0.5, color = "black", label = paste0("p=", round(ado_bray_group_base_1$`Pr(>F)`[1], digits = 3)))+
  annotate("text", x = -0.3, y = 0.45, color = "black", label = paste0("R\u00b2=", round((ado_bray_group_base_1$R2[1])*100, digits = 2), "%"))+
  scale_fill_nejm(name = "Baseline Status")+
  scale_color_nejm(name = "Baseline Status")+
  theme_bw()+
  theme(legend.position = "right")

ggsave("Results/BC_baseline_pcoa_plot_1.png", BC_baseline_pcoa_plot_1, width = 10, height = 8, dpi = 300)

BC_baseline_pcoa_plot_1


bd_base_df_1 <- data.frame(samples = rownames(baseline.bd_1$vectors),
                         distances = baseline.bd_1$distances,
                         baseline_status = meta_beta_base_1$baseline_status)

kruskal_bc_test_base_1 <- bd_base_df_1 %>% rstatix::kruskal_test(distances ~ baseline_status)
print(kruskal_bc_test_base_1)

# Dunn’s test for pairwise comparisons with Benjamini-Hochberg correction
dunn_bc_test_base_1 <- bd_base_df_1 %>% rstatix::dunn_test(distances ~ baseline_status, p.adjust.method = "BH")
print(dunn_bc_test_base_1)

dunn_bc_test_base_1_table <- apply(dunn_bc_test_base_1,2,as.character)
write.csv(dunn_bc_test_base_1_table, "Results/dunn_bc_test_base_1_table.csv", row.names = FALSE)

# Extract p-values for compact letter display
dunn_bc_pvalues_base_1 <- dunn_bc_test_base_1 %>% select(group1, group2, p.adj) %>%
  pivot_wider(names_from = group2, values_from = p.adj) %>%
  column_to_rownames(var = "group1")

# Convert to a matrix format for multcompLetters
dunn_bc_matrix_base_1 <- as.matrix(dunn_bc_pvalues_base_1)
# Fill the lower triangle with upper triangle values to make the matrix symmetric
dunn_bc_matrix_base_1[lower.tri(dunn_bc_matrix_base_1)] <- t(dunn_bc_matrix_base_1)[lower.tri(dunn_bc_matrix_base_1)]


# Generate compact letter display for groups
dunn_bc_labels_base_1 <- multcompLetters(dunn_bc_matrix_base_1)$Letters
print(dunn_bc_labels_base_1)

```


```{r}
## For end samples
ps_ff_rel_end_1 <- ps_filt_fixed_rel_1 %>% ps_filter(!is.na(disease_status_end))

meta_beta_end_1 <- as.data.frame(as.matrix(sample_data(ps_ff_rel_end_1)))

PCoA_ASV_bray_end_1 <- ordinate(ps_ff_rel_end_1, distance = "bray", method = "PCoA")
PCoA_ASV_bray_end_1


beta_ASV_bray_end_1 <- phyloseq::distance(ps_ff_rel_end_1, "bray")
ado_bray_group_end_1 <- adonis2(beta_ASV_bray_end_1 ~ disease_status_end, meta_beta_end_1)
print(ado_bray_group_end_1)

end.bd_1 <- betadisper(d = beta_ASV_bray_end_1, group = meta_beta_end_1$disease_status_end, type="centroid")
bd_end_boxplot_1 <- boxplot(end.bd_1)

bd_end_boxplot_1

png("Results/betadisp_end_bc_boxplot_1.png", width = 3500, height = 2000, res = 300)  # High resolution
boxplot(end.bd_1, main = "Beta Dispersion per End Status (BC) BP1", cex.axis = 1.2, cex.lab = 1.5)
dev.off()

permutest(end.bd_1)

BC_end_pcoa_plot_1 <- plot_ordination(ps_ff_rel_end_1, PCoA_ASV_bray_end_1, axes = c(1,2))+
  stat_conf_ellipse(geom = "polygon", alpha = 0.1, aes(color = disease_status_end,  fill = disease_status_end))+
  geom_point(size = 3, shape = 21, aes(fill = disease_status_end))+
  stat_mean(aes(fill = disease_status_end), shape = 23, size = 4) + 
  annotate("text", x =- 0.3, y = 0.5, color = "black", label = paste0("p=", round(ado_bray_group_end_1$`Pr(>F)`[1], digits = 3)))+
  annotate("text", x = -0.3, y = 0.45, color = "black", label = paste0("R\u00b2=", round((ado_bray_group_end_1$R2[1])*100, digits = 2), "%"))+
  scale_fill_nejm(name = "End Status")+
  scale_color_nejm(name = "End Status")+
  theme_bw()+
  theme(legend.position = "right")

ggsave("Results/BC_end_pcoa_plot_1.png", BC_end_pcoa_plot_1, width = 10, height = 8, dpi = 300)

BC_end_pcoa_plot_1

bd_end_df_1 <- data.frame(samples = rownames(end.bd_1$vectors),
                        distances = end.bd_1$distances,
                        disease_status_end = meta_beta_end_1$disease_status_end)

kruskal_bc_test_end_1 <- bd_end_df_1 %>% rstatix::kruskal_test(distances ~ disease_status_end)
print(kruskal_bc_test_end_1)

# Dunn’s test for pairwise comparisons with Benjamini-Hochberg correction
dunn_bc_test_end_1 <- bd_end_df_1 %>% rstatix::dunn_test(distances ~ disease_status_end, p.adjust.method = "BH")
print(dunn_bc_test_end_1)

dunn_bc_test_end_table_1 <- apply(dunn_bc_test_end_1,2,as.character)
write.csv(dunn_bc_test_end_table_1, "Results/dunn_bc_test_end_table_1.csv", row.names = FALSE)

# Extract p-values for compact letter display
dunn_bc_pvalues_end_1 <- dunn_bc_test_end_1 %>% select(group1, group2, p.adj) %>%
  pivot_wider(names_from = group2, values_from = p.adj) %>%
  column_to_rownames(var = "group1")

# Convert to a matrix format for multcompLetters
dunn_bc_matrix_end_1 <- as.matrix(dunn_bc_pvalues_end_1)
# Fill the lower triangle with upper triangle values to make the matrix symmetric
dunn_bc_matrix_end_1[lower.tri(dunn_bc_matrix_end_1)] <- t(dunn_bc_matrix_end_1)[lower.tri(dunn_bc_matrix_end_1)]


# Generate compact letter display for groups
dunn_bc_labels_end_1 <- multcompLetters(dunn_bc_matrix_end_1)$Letters
print(dunn_bc_labels_end_1)

```


## Check prevalence of taxa
```{r}

print(ps_filt_fixed_1)

ps_filt_fixed_prev_1 <- tax_filter(ps_filt_fixed_1,
  min_prevalence = 0.01)
  
print(ps_filt_fixed_prev_1)


prev_thresholds_1 <- c(0.05, 0.01, 0.001)

prev_filt_results_1 <- data.frame(min_prevalence = prev_thresholds_1, num_taxa_retained = NA, perc_taxa_removed = NA)

total_taxa_1 <- nrow(tax_table(ps_filt_fixed_1))

for (i in seq_along(prev_thresholds_1)) {
  ps_filt_fixed_samples_prev_all_1 <- ps_filt_fixed_1 %>%
    tax_filter(min_prevalence = prev_thresholds_1[i])
  
  num_taxa_retained <- nrow(tax_table(ps_filt_fixed_samples_prev_all_1))
  perc_taxa_removed <- 100 * (total_taxa_1 - num_taxa_retained) / total_taxa_1
  
  prev_filt_results_1$num_taxa_retained[i] <- num_taxa_retained
  prev_filt_results_1$perc_taxa_removed[i] <- perc_taxa_removed
}

print(prev_filt_results_1)

```

```{r}

ps_filt_fixed_genus_rel_1 <- tax_transform(ps_filt_fixed_genus_1, trans = "compositional")
input_sample_metadata_1 <- ps_filt_fixed_genus_rel_1 %>% meta
input_data_genus_Maaslin_1 <- as.data.frame(otu_table(ps_filt_fixed_genus_rel_1))

input_data_genus_Maaslin_1 <- input_data_genus_Maaslin_1[sapply(input_data_genus_Maaslin_1, is.numeric)]

output_directory_genus_1 <- "Maaslin2_output_genus_1"

fit_data_genus_1 <- Maaslin2::Maaslin2(
    input_data = input_data_genus_Maaslin_1,
    input_metadata = input_sample_metadata_1,
    output = output_directory_genus_1,
    analysis_method = "LM",
    fixed_effects = "group",
    normalization = "NONE", 
    correction = "BH",
    transform = "LOG",
    min_abundance = 0,
    min_prevalence = 0.01,
    reference = "group,active-active")

```

```{r}

head(fit_data_genus_1$results, n=20) %>% kbl() %>%
  kableExtra::kable_paper(font_size = 12) %>%
  scroll_box(width = "1000px", height = "500px")

```

## Now run DAA with Linda to compare
```{r}

## WRONG It's not the same output. I get samples in rownames instead of taxa. 
# asv.mat_1 <- otu_table(ps_filt_fixed_genus_rel_1) %>% as.matrix()
# 
# asv.mat_1 <- asv.mat_1[, colSums(asv.mat_1) > 0]
# 
# metadata.dat_1 <- meta(ps_filt_fixed_genus_rel_1) %>% as.data.frame()
# 
# metadata.dat_1 <- metadata.dat_1 %>% 
#   select(-c(collection_days_post_initial_fmt, median_fecal_calprot_arm)) %>% 
#   replace_na(list(host_age=0, disease_status_end="not end",                                     
#                                   baseline_status="not base",
#                                   disease_status="unknown"))
# 
# metadata.dat_1 <- metadata.dat_1 %>% mutate_if(is.factor, as.character)
# 
# all(rownames(metadata.dat_1) %in% rownames(asv.mat_1))
# all(rownames(asv.mat_1) %in% rownames(metadata.dat_1))
# 
# 
# linda.obj.genus_1 <- linda(feature.dat = asv.mat_1, meta.dat = metadata.dat_1,
#                          feature.dat.type = 'proportion',
#                          formula = '~group', 
#                          prev.filter = 0.01, 
#                          p.adj.method = "BH")

```
```{r}
# linda.out_1 <- as.data.frame(linda.obj.genus_1$output)
# 
# head(linda.out_1, n=20) %>% 
#   kbl() %>% 
#   kableExtra::kable_paper(font_size = 12) %>% 
#   scroll_box(width = "1000px", height="500px")
# 
# linda.res.plot.genus <- linda.plot(linda.obj.genus_1, 
#                                    c("groupactive-remission", "grouphealthy-healthy"),
#                                    legend=TRUE)
# 
# 
# lfc.plot1 <- linda.res.plot.genus$plot.lfc[[1]]
# volc.plot1 <- linda.res.plot.genus$plot.volcano[[1]]
# 
# ggsave("./linda_plots/lfc.plot1.png", lfc.plot1, device = png, width = 10, height = 7)
# ggsave("./linda_plots/volc.plot1.png", volc.plot1, device = png, width = 10, height = 7)
# 
# lfc.plot1
# volc.plot1

```


```{r}
taxa_abund_kw_1 <- as.data.frame(otu_table(ps_filt_fixed_genus_1))

# Convert row names to a column (sample IDs)
taxa_abund_kw_1 <- taxa_abund_kw_1 %>% 
  tibble::rownames_to_column(var = "samples")

# Convert sample_data from phyloseq object to a proper data frame
metadata_samples_kw_1 <- as(sample_data(ps_filt_fixed_genus_1), "data.frame") %>% 
  tibble::rownames_to_column(var = "samples") # Ensure sample IDs are a column

metadata_samples_kw_1 <- metadata_samples_kw_1 %>% 
  select(c(samples, group))


# Merge metadata with abundance data
taxa_data_kw_1 <- left_join(metadata_samples_kw_1, taxa_abund_kw_1, by = "samples") 

taxa_data_kw_tib_1 <- taxa_data_kw_1 %>%
  as_tibble()  # Convert to tibble for tidyverse functions

# Function to perform Kruskal-Wallis test on each taxon
kruskal_test_results_1 <- taxa_data_kw_tib_1 %>%
  pivot_longer(cols = -c(samples, group), names_to = "taxon", values_to = "abundance") %>%
  group_by(taxon) %>%
  rstatix::kruskal_test(abundance ~ group) %>%
  rstatix::adjust_pvalue(method = "BH") %>%  # Adjust for multiple testing (Benjamini-Hochberg)
  arrange(p.adj)

# View significant results (p.adj < 0.05)
significant_taxa_kw_1 <- kruskal_test_results_1 %>% filter(p.adj < 0.05)

# Display top results
significant_taxa_kw_1 %>% 
  arrange(p.adj) %>% 
  head(n=20)

top_10_sig_taxa_1 <- significant_taxa_kw_1 %>% arrange(p.adj) %>% head(n=10)

top_10_sig_taxa_1

```


```{r}

top10_sig_taxa_names_1 <- top_10_sig_taxa_1$taxon

ps_ff_genus_melt_1 <- ps_filt_fixed_genus_rel_1 %>% psmelt()


ps_traj_taxa_filt_1 <- ps_ff_genus_melt_1 %>% filter(OTU %in% top10_sig_taxa_names_1)
ps_traj_taxa_filt_1 <- ps_traj_taxa_filt_1 %>% filter(!host_assigned_donor_subject_id == "is donor")

# Function to generate a plot for a given host_subject_id
plot_subject_trajectory <- function(df, group_name) {
  
  df_filt <- df %>% filter(group == group_name)
  
  ggplot(df_filt, aes(x = week, y = Abundance, group = OTU, color = OTU)) +
    geom_line(aes(linetype = OTU), size = 1) + # Line style per taxa
    geom_point(size = 3) + 
    scale_color_manual(values = distinct_palette(10, pal = "brewerPlus"))
    
    
}

plots_acac_1 <- plot_subject_trajectory(ps_traj_taxa_filt_1, "active-active")

plots_acrem_1 <- plot_subject_trajectory(ps_traj_taxa_filt_1, "active-remission")

wr_acac_lineplots_1 <- wrap_plots(plots_acac_1) + plot_layout(guides = "collect") &
  facet_wrap("host_subject_id", scales = "free_x") &
  theme(axis.text.x = element_text(angle = 60, hjust = 1, size = 10),
        axis.ticks.y = element_blank(), legend.title = element_blank(),
        legend.position = "bottom", strip.text = element_text(size = 12),
        legend.key.size = unit(0.4, "cm"),
        legend.text = element_text(size = 14))
wr_acrem_lineplots_1 <- wrap_plots(plots_acrem_1) + plot_layout(guides = "collect") &
  facet_wrap("host_subject_id", scales = "free_x") &
  theme(axis.text.x = element_text(angle = 60, hjust = 1, size = 10),
        axis.ticks.y = element_blank(), legend.title = element_blank(),
        legend.position = "bottom", strip.text = element_text(size = 12),
        legend.key.size = unit(0.4, "cm"),
        legend.text = element_text(size = 14))

ggsave(filename = "wr_acac_lineplots_1.png", plot = wr_acac_lineplots_1, 
       width = 14, height = 11, dpi = 320)

ggsave(filename = "wr_acrem_lineplots_1.png", plot = wr_acrem_lineplots_1, 
       width = 14, height = 11, dpi = 320)


wr_acac_lineplots_1
wr_acrem_lineplots_1

```
## BIOPROJECT 2 WORKFLOW
```{r}

ps_2 <- ps %>% ps_filter(bioproject == "2")

meta_2 <- sample_data(ps_2)

sample_names_2 <- rownames(meta_2)

sample_names_2 

seqtab.nochim_951422 <- readRDS("seqtab.nochim_951422.rds")

rownames(seqtab.nochim_951422) <- sub("^\\d+\\.", "", rownames(seqtab.nochim_951422))

seqtab_filtered_2 <- seqtab.nochim_951422[sample_names_2, ]

seqs_2 <- getSequences(seqtab_filtered_2)
names(seqs_2) <- seqs_2

alignment_2 <- AlignSeqs(DNAStringSet(seqs_2), anchor=NA)
saveRDS(alignment_2, "alignment_2.rds")
# alignment_2 <- readRDS("alignment_2.rds")


phang_align_2 <- phyDat(as(alignment_2, "matrix"), type="DNA")
saveRDS(phang_align_2, "phang_align_2.rds")
# phang_align_2 <- readRDS("phang_align_2.rds")


## Check kind of distance
dm_2 <- dist.ml(phang_align_2)

saveRDS(dm_2, "dm_2.rds")
# dm_2 <- readRDS("dm_2.rds")


treeNJ_2 <- NJ(dm_2)
saveRDS(treeNJ_2, "treeNJ_2.rds")
# treeNJ_2 <- readRDS("treeNJ_2.rds")

```


```{r}
ps_sum_2 <- summarize_phyloseq(ps_2)
data.frame(do.call(rbind, ps_sum_2[c(1:10)]))

```

```{r}

head(ps_2@otu_table, n=10) %>% kbl() %>% 
  kableExtra::kable_paper(font_size = 12) %>% 
  scroll_box(width = "800px", height = "300px")

head(ps_2@tax_table, n=10) %>% kbl() %>%
  kableExtra::kable_paper(font_size = 12) %>%
  scroll_box(width = "800px", height = "300px")

metadata_x_2 <- ps_2 %>% meta()
head(metadata_x_2, n=10) %>% kbl() %>%
  kableExtra::kable_paper(font_size = 12) %>%
  scroll_box(width = "800px", height = "300px")

ggtree(ps_2@phy_tree, options(ignore.negative.edge=TRUE))

```


```{r}
# Chloroplast_2 <- ntaxa(subset_taxa(ps_2, Order == "Chloroplast"))
# 
# Mitochondria_2 <- ntaxa(subset_taxa(ps_2, Family == "Mitochondria"))
# 
# chl_mt_df_2 <- data.frame(Taxa = c("Chloroplast", "Mitochondria"),
#                        nASVs = c(Chloroplast_2, Mitochondria_2)) %>%
#   mutate(fraction = round(nASVs/ntaxa(ps_2), 3))
# chl_mt_df_2 
# 
# ps_filt_2 <- subset_taxa(ps_2, Order != "Chloroplast")
# ps_filt_2 <- subset_taxa(ps_filt_2, Family != "Mitochondria")

ps_2@tax_table %>% 
  data.frame() %>% 
  group_by(Kingdom) %>% 
  dplyr::count() %>%
  mutate(fraction = round(n/ntaxa(ps_filt_2), 3))

ps_filt_2 <- subset_taxa(ps_2, Kingdom == "Bacteria")

ps_filt_2@tax_table %>% 
  data.frame() %>% 
  group_by(Phylum) %>% 
  dplyr::count() %>% 
  arrange(desc(n))


```

```{r}

ps_filt_fixed_2 <- ps_filt_2 %>%
 tax_fix(
  min_length = 4,
  sep="_", anon_unique=TRUE, 
  suffix_rank="classified")

ps_filt_fixed_2@tax_table
  
```


```{r}

sum_of_reads_2 <- as.data.frame(sample_sums(ps_filt_fixed_2@otu_table)) %>% rownames_to_column(var = "samples") %>%
 dplyr::rename(sum_of_reads_2 = `sample_sums(ps_filt_fixed_2@otu_table)`)%>%
  left_join((metadata_x_2 %>% rownames_to_column(var = "samples")), by = "samples")
head((sum_of_reads_2 %>% arrange(sum_of_reads_2)), n=20)%>% kbl() %>%
  kableExtra::kable_paper(font_size = 12) %>%
  scroll_box(width = "800px", height = "300px")



```

```{r}
sum_of_reads_2 %>% arrange(sum_of_reads_2) %>% 
  ggplot(aes(x = reorder(samples, sum_of_reads_2), y = sum_of_reads_2, fill = group)) +
  geom_point(shape = 21, size = 2, color = "black")+
  scale_fill_manual(values = c(
    "remission-remission" = "orange",
    "remission-active" = "darkred"
    ))+
  labs(x = "samples")+
  theme_bw()+
  theme(axis.text.x=element_text(angle=90))


```


```{r}
ps_filt_fixed_gen_rel_2 <- ps_filt_fixed_2 %>% 
  tax_transform("compositional", rank = "Genus")

ps_filt_fixed_gen_rel_2@sam_data$group <- as.factor(ps_filt_fixed_gen_rel_2@sam_data$group)

top_taxa_2 <- tax_top(ps_filt_fixed_gen_rel_2, n=40)

ps_ff_gen_rel_end_2 <- ps_filt_fixed_gen_rel_2 %>% ps_filter(!is.na(disease_status_end))
top_taxa_end_2 <- tax_top(ps_ff_gen_rel_end_2, n=40)

melt_genus_rel_end_2 <- ps_filt_fixed_gen_rel_2 %>% psmelt() %>% 
  mutate(Genus_top = case_when(Genus %in% top_taxa_end_2 ~ Genus, 
                               TRUE ~ "other")) %>% 
  mutate(Genus_top = as.factor(Genus_top)) %>% 
  mutate(Genus_top = fct_relevel(Genus_top, c(top_taxa_end_2, "other"))) %>% 
  filter(!is.na(disease_status_end))

main_genus_end_2 <- melt_genus_rel_end_2 %>%
  ggplot(aes(x = Sample, y = Abundance, fill = Genus_top)) +
  geom_bar(stat = "identity", color = "black")+
  scale_fill_manual(values = distinct_palette(40, pal = "brewerPlus", add = "grey70"))+
  theme_bw()+
  labs(title= "Genus Bar plot for end samples") +
  theme(axis.title.x = element_blank(), legend.position = "right")

ps_ff_gen_rel_base_2 <- ps_filt_fixed_gen_rel_2 %>% ps_filter(!is.na(baseline_status))
top_taxa_base_2 <- tax_top(ps_ff_gen_rel_base_2, n=40)

melt_genus_rel_base_2 <- ps_filt_fixed_gen_rel_2 %>% psmelt() %>% 
  mutate(Genus_top = case_when(Genus %in% top_taxa_base_2 ~ Genus, 
                               TRUE ~ "other")) %>% 
  mutate(Genus_top = as.factor(Genus_top)) %>% 
  mutate(Genus_top = fct_relevel(Genus_top, c(top_taxa_base_2, "other"))) %>% 
  filter(!is.na(baseline_status))

main_genus_base_2 <- melt_genus_rel_base_2 %>%
  ggplot(aes(x = Sample, y = Abundance, fill = Genus_top)) +
  geom_bar(stat = "identity", color = "black")+
  scale_fill_manual(values = distinct_palette(40, pal = "brewerPlus", add = "grey70"))+
  theme_bw()+
  labs(title= "Genus Bar plot for baseline samples") +
  theme(axis.title.x = element_blank(), legend.position = "right")



ggsave("Results/main_genus_end_2_barplot.png", main_genus_end_2, device = png, width = 10, height = 8)
ggsave("Results/main_genus_base_2_barplot.png", main_genus_base_2, device = png, width = 10, height = 8)

main_genus_end_2
main_genus_base_2


```



## Check for singletons and doubletons
```{r}
length(which(taxa_sums(ps_filt_fixed_2) <= 1))

length(which(taxa_sums(ps_filt_fixed_2) <= 2))

```


```{r}

taxa_sums(ps_filt_fixed_2) %>%
  data.frame() %>%
  dplyr::rename(sum = ".") %>%
  arrange(sum)%>%
  tail(n=20) %>% kbl() %>%
  kableExtra::kable_paper(font_size = 12) %>%
  scroll_box(width = "200px", height = "300px")

```

## Normalization and check distribution
```{r}

ps_filt_fixed_rel_2 <- tax_transform(ps_filt_fixed_2, trans = "compositional")
head(ps_filt_fixed_rel_2@otu_table, n=10) 

```


## Transformation
```{r}

ps_filt_fixed_rel_clr_2 <- tax_transform(ps_filt_fixed_rel_2, trans = "clr", chain = TRUE)
head(ps_filt_fixed_rel_clr_2@otu_table, n=10)


```

## Aggregate data at specific taxonomic rank: Phylum and Genus
```{r}

ps_filt_fixed_phylum_2 <- tax_agg(ps_filt_fixed_2, rank = "Phylum")
ps_filt_fixed_class_2 <- tax_agg(ps_filt_fixed_2, rank = "Class")
ps_filt_fixed_order_2 <- tax_agg(ps_filt_fixed_2, rank = "Order")
ps_filt_fixed_family_2 <- tax_agg(ps_filt_fixed_2, rank = "Family")
ps_filt_fixed_genus_2 <- tax_agg(ps_filt_fixed_2, rank = "Genus")
#ps_filt_fixed_species_2 <- tax_agg(ps_filt_fixed_2, rank = "Species")

print(ps_filt_fixed_phylum_2)
print(ps_filt_fixed_genus_2)


summary_agg_2 <- data.frame(Taxa = c(colnames(ps_filt_fixed_2@tax_table)[c(2:6)], "ASVs"),
                 number_of_taxa=c(dim(ps_filt_fixed_phylum_2@otu_table)[2], dim(ps_filt_fixed_class_2@otu_table)[2], dim(ps_filt_fixed_order_2@otu_table)[2], dim(ps_filt_fixed_family_2@otu_table)[2], dim(ps_filt_fixed_genus_2@otu_table)[2], dim(ps_filt_fixed_2@otu_table)[2]))
print(summary_agg_2)

```


## Data exploration
```{r}
barplots_group <- ps_filt_fixed_2 %>%
  ps_mutate(host_sex = as.factor(host_sex)) %>%
  ps_mutate(group = as.factor(group)) %>%
  ps_arrange(group, week) %>% 
  comp_barplot(group_by = "group", tax_level = "Phylum", 
               tax_transform_for_plot = "compositional", n_taxa = 11, 
               bar_width = 0.8, merge_other = FALSE, label = "week", 
               sample_order = "default")

barplots_group_wrapped <- wrap_plots(barplots_group) +
  plot_layout(guides = "collect", nrow=2)

groups_barplot <- barplots_group_wrapped&
  facet_wrap("host_subject_id", scales = "free_x")& 
  theme(axis.text.x = element_text(angle = 60, hjust = 1, size = 10), 
        axis.ticks.y = element_blank(), leg.end.title = element_blank(), 
        legend.position = "bottom", strip.text = element_text(size=12),
        legend.key.size = unit(0.4, "cm"),
        legend.text = element_text(size=10))

groups_barplot

ggsave("groups_barplot.png", groups_barplot, width = 10, height = 8, dpi = 300)
```


```{r}

barplots_genus_group <- ps_filt_fixed_2 %>%
  ps_mutate(host_sex = as.factor(host_sex)) %>%
  ps_mutate(group = as.factor(group)) %>%
  ps_arrange(group, week) %>% 
  comp_barplot(group_by = "group", tax_level = "Genus", 
               tax_transform_for_plot = "compositional", n_taxa = 11, 
               bar_width = 0.8, merge_other = FALSE, label = "week", 
               sample_order = "default")

barplots_genus_group_wrapped <- wrap_plots(barplots_genus_group) +
  plot_layout(guides = "collect", nrow=2)

groups_genus_barplot <- barplots_genus_group_wrapped&
  facet_wrap("host_subject_id", scales = "free_x")& 
  theme(axis.text.x = element_text(angle = 60, hjust = 1, size = 10), 
        axis.ticks.y = element_blank(), leg.end.title = element_blank(), 
        legend.position = "bottom", strip.text = element_text(size=12),
        legend.key.size = unit(0.4, "cm"),
        legend.text = element_text(size=10))

groups_genus_barplot

ggsave("groups_genus_barplot.png", groups_genus_barplot, width = 10, height = 8, dpi = 300)

```


```{r}
# Define a function to create bar plots for each group separately
create_group_barplot <- function(ps_obj, group_name) {
  ps_subset <- ps_obj %>% ps_filter(group == group_name)
  
  barplot <- ps_subset %>%
    ps_mutate(host_sex = as.factor(host_sex)) %>%
    ps_mutate(group = as.factor(group)) %>%
    ps_arrange(group, week) %>%
    comp_barplot(group_by = "group", tax_level = "Phylum",
                 tax_transform_for_plot = "compositional", n_taxa = 11,
                 bar_width = 0.8, merge_other = FALSE, label = "week",
                 sample_order = "default")
  
  wrapped_plot <- wrap_plots(barplot) +
    plot_layout(guides = "collect") &
    facet_wrap("host_subject_id", scales = "free_x") &
    theme(axis.text.x = element_text(angle = 60, hjust = 1, size = 10),
          axis.ticks.y = element_blank(), legend.title = element_blank(),
          legend.position = "bottom", strip.text = element_text(size = 12),
          legend.key.size = unit(0.4, "cm"),
          legend.text = element_text(size = 10))
  
  return(wrapped_plot)
}

# Create plots for each group
plot_1 <- create_group_barplot(ps_filt_fixed_2, "remission-remission")
plot_2 <- create_group_barplot(ps_filt_fixed_2, "remission-active")

# Save plots as separate images
ggsave("rem-rem_barplot.png", plot_1, width = 10, height = 8, dpi = 300)
ggsave("rem-ac_barplot.png", plot_2, width = 10, height = 8, dpi = 300)

# If you want to visualize them:
plot_1
plot_2


```


```{r}
create_group_barplots <- function(ps_obj, group_name, max_per_plot = 6, output_prefix = "plot") {
  ps_subset <- ps_obj %>% ps_filter(group == group_name)
  
  # Get unique subjects
  subject_ids <- unique(sample_data(ps_subset)$host_subject_id)
  
  # Split into chunks of 'max_per_plot'
  subject_batches <- split(subject_ids, ceiling(seq_along(subject_ids) / max_per_plot))
  
  # Generate plots for each batch
  plots <- list()
  
  for (i in seq_along(subject_batches)) {
    subjects <- subject_batches[[i]]
    
    ps_batch <- ps_subset %>% ps_filter(host_subject_id %in% subjects)
    
    barplot <- ps_batch %>%
      ps_mutate(host_sex = as.factor(host_sex)) %>%
      ps_mutate(group = as.factor(group)) %>%
      ps_arrange(group, week) %>%
      comp_barplot(group_by = "group", tax_level = "Phylum",
                   tax_transform_for_plot = "compositional", n_taxa = 11,
                   bar_width = 0.8, merge_other = FALSE, label = "week",
                   sample_order = "default")
    
    wrapped_plot <- wrap_plots(barplot) +
      plot_layout(guides = "collect") &
      facet_wrap("host_subject_id", scales = "free_x") &
      theme(axis.text.x = element_text(angle = 60, hjust = 1, size = 10),
            axis.ticks.y = element_blank(), legend.title = element_blank(),
            legend.position = "bottom", strip.text = element_text(size = 12),
            legend.key.size = unit(0.4, "cm"),
            legend.text = element_text(size = 10))
    
    plots[[i]] <- wrapped_plot
    
    # Save each plot separately
    ggsave(filename = paste0(output_prefix, "_2_batch_", i, ".png"), plot = wrapped_plot, 
           width = 10, height = 8, dpi = 300)
  }
  
  return(plots)
}

# Generate and save plots
plots_rem_rem_2 <- create_group_barplots(ps_filt_fixed_2, "remission-remission", max_per_plot = 6, output_prefix = "rem_rem_barplot")
plots_rem_active_2 <- create_group_barplots(ps_filt_fixed_2, "remission-active", max_per_plot = 6, output_prefix = "rem_active_barplot")

plots_rem_rem_2

plots_rem_active_2

```



```{r}
# Define a function to create bar plots for each group separately
create_genus_group_barplot <- function(ps_obj, group_name) {
  ps_subset <- ps_obj %>% ps_filter(group == group_name)
  
  barplot <- ps_subset %>%
    ps_mutate(host_sex = as.factor(host_sex)) %>%
    ps_mutate(group = as.factor(group)) %>%
    ps_arrange(group, week) %>%
    comp_barplot(group_by = "group", tax_level = "Genus",
                 tax_transform_for_plot = "compositional", n_taxa = 11,
                 bar_width = 0.8, merge_other = FALSE, label = "week",
                 sample_order = "default")
  
  wrapped_plot <- wrap_plots(barplot) +
    plot_layout(guides = "collect") &
    facet_wrap("host_subject_id", scales = "free_x") &
    theme(axis.text.x = element_text(angle = 60, hjust = 1, size = 10),
          axis.ticks.y = element_blank(), legend.title = element_blank(),
          legend.position = "bottom", strip.text = element_text(size = 12),
          legend.key.size = unit(0.4, "cm"),
          legend.text = element_text(size = 10))
  
  return(wrapped_plot)
}

# Create plots for each group
plot_genus_1 <- create_genus_group_barplot(ps_filt_fixed_2, "remission-remission")
plot_genus_2 <- create_genus_group_barplot(ps_filt_fixed_2, "remission-active")

# Save plots as separate images
ggsave("rem-rem_genus_barplot.png", plot_genus_1, width = 10, height = 8, dpi = 300)
ggsave("rem-ac_genus_barplot.png", plot_genus_2, width = 10, height = 8, dpi = 300)

# If you want to visualize them:
plot_genus_1
plot_genus_2


```

```{r}
create_group_barplots <- function(ps_obj, group_name, max_per_plot = 6, output_prefix = "plot") {
  ps_subset <- ps_obj %>% ps_filter(group == group_name)
  
  # Get unique subjects
  subject_ids <- unique(sample_data(ps_subset)$host_subject_id)
  
  # Split into chunks of 'max_per_plot'
  subject_batches <- split(subject_ids, ceiling(seq_along(subject_ids) / max_per_plot))
  
  # Generate plots for each batch
  plots <- list()
  
  for (i in seq_along(subject_batches)) {
    subjects <- subject_batches[[i]]
    
    ps_batch <- ps_subset %>% ps_filter(host_subject_id %in% subjects)
    
    barplot <- ps_batch %>%
      ps_mutate(host_sex = as.factor(host_sex)) %>%
      ps_mutate(group = as.factor(group)) %>%
      ps_arrange(group, week) %>%
      comp_barplot(group_by = "group", tax_level = "Genus",
                   tax_transform_for_plot = "compositional", n_taxa = 11,
                   bar_width = 0.8, merge_other = FALSE, label = "week",
                   sample_order = "default")
    
    wrapped_plot <- wrap_plots(barplot) +
      plot_layout(guides = "collect") &
      facet_wrap("host_subject_id", scales = "free_x") &
      theme(axis.text.x = element_text(angle = 60, hjust = 1, size = 10),
            axis.ticks.y = element_blank(), legend.title = element_blank(),
            legend.position = "bottom", strip.text = element_text(size = 12),
            legend.key.size = unit(0.4, "cm"),
            legend.text = element_text(size = 10))
    
    plots[[i]] <- wrapped_plot
    
    # Save each plot separately
    ggsave(filename = paste0(output_prefix, "_2_genus_batch_", i, ".png"), plot = wrapped_plot, 
           width = 10, height = 8, dpi = 300)
  }
  
  return(plots)
}

# Generate and save plots
plots_genus_rem_rem_2 <- create_group_barplots(ps_filt_fixed_2, "remission-remission", max_per_plot = 6, output_prefix = "rem_rem_barplot")
plots_genus_rem_ac_2 <- create_group_barplots(ps_filt_fixed_2, "remission-active", max_per_plot = 6, output_prefix = "rem_active_barplot")

plots_genus_rem_rem_2

plots_genus_rem_ac_2


```


```{r}
# cols_hsi<- pal_nejm("default")(17)
# names(cols_hsi) <- unique(samdat_tbl(ps_filt_fixed_2)$host_subject_id)
# cols_group <- c("darkred", "orange", "darkgreen")
# names(cols_group) <- unique(samdat_tbl(ps_filt_fixed_2)$group)
# cols_sex<- c("black", "grey", "lightgreen")
# names(cols_sex) <- unique(samdat_tbl(ps_filt_fixed_2)$host_sex)
# cols_week<- distinct_palette(22, pal = "brewerPlus")
# names(cols_week) <- unique(samdat_tbl(ps_filt_fixed_2)$week)
# 
# # sample_anno_test <- sampleAnnotation(
# #   Week = anno_sample("week"),
# #   Group = anno_sample("group"),
# #   Sex = anno_sample("host_sex"),
# #   HSI = anno_sample("host_subject_id"),
# #   col = list(HSI = cols_hsi, Group = cols_group, Sex = cols_sex, Week = cols_week),
# #   border = FALSE
# # )
# # 
# # print(sample_anno_test)
# 
# 
# ps_filt_fixed_2 %>%
#   tax_transform("compositional", rank = "Phylum") %>%
#   comp_heatmap(colors = heat_palette(sym = FALSE, "Blue-Red 3"),
#                tax_anno = taxAnnotation(
#       Prev. = anno_tax_prev(bar_width = 0.3, size = grid::unit(1, "cm"))
#     ),
#     sample_anno = sampleAnnotation(
#       Week = anno_sample(var = "week"),
#       Group = anno_sample(var = "group"),
#       Sex = anno_sample(var = "host_sex"),
#       HSI = anno_sample(var = "host_subject_id"),
#       col = list(HSI = cols_hsi, Group = cols_group, Sex = cols_sex, Week = cols_week), border = FALSE))


```



## Alpha diversity
```{r}

alpha_div_2 <- estimate_richness(ps_filt_fixed_2, measures = c("Observed", "Shannon", "InvSimpson", "Chao1"))
PD_2 <- calculatePD(ps_filt_fixed_2, justDF = TRUE, include_root = FALSE) %>% pull(PD)
alpha_div_2 <- alpha_div_2 %>% mutate(PD = PD_2)
rownames(alpha_div_2) <- gsub("^X", "", rownames(alpha_div_2))
head(alpha_div_2, n=20)

metadata_samples_alpha_2 <- bind_cols(meta_951422, alpha_div_2)
metadata_samples_long_2 <- metadata_samples_alpha_2 %>% pivot_longer(cols= Observed:PD, names_to = "alpha_div", values_to = "value")
kruskal_test_alpha_2 <- metadata_samples_long_2 %>% group_by(alpha_div) %>%
    rstatix::kruskal_test(value ~ group) %>% mutate(p_adj_BH = p.adjust(p, method = "BH"))

kruskal_test_alpha_2

```

```{r}
dunn_test_alpha_2 <- metadata_samples_long_2 %>% group_by(alpha_div) %>%
    rstatix::dunn_test(value ~ group, p.adjust.method = "BH") %>%
  add_xy_position(x = "group", fun = "mean_sd", step.increase = 0.05, scales = "free")
dunn_test_alpha_2 %>% kbl() %>%
  kableExtra::kable_paper(font_size = 12)


```

```{r}

ggplot(metadata_samples_long_2, aes(x = group, y = value)) +
  geom_boxplot() +
  geom_jitter(shape = 21, color = "black", size = 2, aes(fill = group)) +
  facet_wrap(~alpha_div, scales = "free", ncol = 3) +
  scale_fill_nejm() +
  stat_pvalue_manual(data = dunn_test_alpha_2, 
                     label = "p.adj.signif", tip.length = 0.01, bracket.shorten = 0.05,
                     hide.ns = TRUE, size = 4) +
  scale_x_discrete(guide = guide_axis(n.dodge = 2)) +
  theme_bw() +
  theme(axis.text.x = element_text(size = 10), legend.position = "none")

```

```{r}

metadata_samples_alpha_2

kruskal_chao1_2 <- kruskal.test(Chao1 ~ group, data = metadata_samples_alpha_2)
kruskal_shannon_2 <- kruskal.test(Shannon ~ group, data = metadata_samples_alpha_2)
kruskal_invsimps <- kruskal.test(InvSimpson ~ group, data = metadata_samples_alpha_2)
kruskal_obsv_2 <- kruskal.test(Observed ~ group, data = metadata_samples_alpha_2)
kruskal_pd_2 <- kruskal.test(PD ~ group, data = metadata_samples_alpha_2)


# Extract values
H_chao1_2 <- kruskal_chao1_2$statistic
p_chao1_2 <- kruskal_chao1_2$p.value

H_shannon_2 <- kruskal_shannon_2$statistic
p_shannon_2 <- kruskal_shannon_2$p.value

H_invsimps_2 <- kruskal_invsimps$statistic
p_invsimps_2 <- kruskal_invsimps$p.value

H_obsv_2 <- kruskal_obsv_2$statistic
p_obsv_2 <- kruskal_obsv_2$p.value

H_pd_2 <- kruskal_pd_2$statistic
p_pd_2 <- kruskal_pd_2$p.value

# Print results
cat("Kruskal-Wallis test for Chao1 index: H =", round(H_chao1_2, 3), "p =", round(p_chao1_2, 3), "\n")
cat("Kruskal-Wallis test for Shannon index: H =", round(H_shannon_2, 3), "p =", round(p_shannon_2, 3), "\n")
cat("Kruskal-Wallis test for Inverse Simpson index: H =", round(H_invsimps_2, 3), "p =", round(p_invsimps_2, 3), "\n")
cat("Kruskal-Wallis test for Observed Number of Species: H =", round(H_obsv_2, 3), "p =", round(p_obsv_2, 3), "\n")
cat("Kruskal-Wallis test for Faith's Phylogenetic Diversity: H =", round(H_pd_2, 3), "p =", round(p_pd_2, 3), "\n")


# Filter for Chao1 and Shannon indices
dunn_chao1_2 <- dunn_test_alpha_2 %>% filter(alpha_div == "Chao1")
dunn_shannon_2 <- dunn_test_alpha_2 %>% filter(alpha_div == "Shannon")
dunn_invsimps_2 <- dunn_test_alpha_2 %>% filter(alpha_div == "InvSimpson")
dunn_obsv_2 <- dunn_test_alpha_2 %>% filter(alpha_div == "Observed")
dunn_pd_2 <- dunn_test_alpha_2 %>% filter(alpha_div == "PD")

# Show significant results (p < 0.05)
significant_chao1_2 <- dunn_chao1_2 %>% filter(p.adj < 0.05)
significant_shannon_2 <- dunn_shannon_2 %>% filter(p.adj < 0.05)
significant_invsimps_2 <- dunn_invsimps_2 %>% filter(p.adj < 0.05)
significant_obsv_2 <- dunn_obsv_2 %>% filter(p.adj < 0.05)
significant_pd_2 <- dunn_pd_2 %>% filter(p.adj < 0.05)

# Print significant comparisons
print(significant_chao1_2)
print(significant_shannon_2)
print(significant_invsimps_2)
print(significant_obsv_2)
print(significant_pd_2)

alpha_summary_2 <- metadata_samples_alpha_2 %>%
  group_by(group) %>%
  summarise(
    Chao1_mean = mean(Chao1, na.rm = TRUE),
    Chao1_sd = sd(Chao1, na.rm = TRUE),
    Shannon_mean = mean(Shannon, na.rm = TRUE),
    Shannon_sd = sd(Shannon, na.rm = TRUE),
    InvSimps_mean = mean(InvSimpson, na.rm = TRUE),
    InvSimps_sd = sd(InvSimpson, na.rm = TRUE), 
    Obsv_mean = mean(Observed, na.rm = TRUE),
    Obsv_sd = sd(Observed, na.rm = TRUE),
    PD_mean = mean(PD, na.rm = TRUE),
    PD_sd = sd(PD, na.rm = TRUE)
  )

print(alpha_summary_2)

write.csv(alpha_summary_2, "Results/alpha_summary_2_table.csv", row.names = FALSE)

```


## Rarefied

```{r}
library(ranacapa); packageVersion("ranacapa")

ps_filt_fixed_2 %>% 
  ggrare(step = 50, color = "group", se = FALSE)

```


```{r}
set.seed(19950622)

ps_ff_rare_2 <- rarefy_even_depth(ps_filt_fixed_2, sample.size = 10558, replace = FALSE)

ps_ff_rare_2 %>% 
  ggrare(step = 50, color = "group", se = FALSE)

alpha_div_rare_2 <- estimate_richness(ps_ff_rare_2, measures = c("Observed", "Shannon", "InvSimpson", "Chao1"))
PD_rare_2 <- calculatePD(ps_ff_rare_2, justDF = TRUE, include_root = FALSE) %>% pull(PD)
alpha_div_rare_2 <- alpha_div_rare_2 %>% mutate(PD = PD_rare_2)
rownames(alpha_div_rare_2) <- gsub("^X", "", rownames(alpha_div_rare_2))
head(alpha_div_rare_2, n=20)

meta_rare_2 <- as.data.frame(as.matrix(sample_data(ps_ff_rare_2)))

metadata_samples_alpha_2_rare <- bind_cols(meta_rare_2, alpha_div_rare_2)
metadata_samples_long_rare_2 <- metadata_samples_alpha_2_rare %>% pivot_longer(cols= Observed:PD, names_to = "alpha_div", values_to = "value")
kruskal_test_alpha_rare_2 <- metadata_samples_long_rare_2 %>% group_by(alpha_div) %>%
  rstatix::kruskal_test(value ~ group) %>% mutate(p_adj_BH = p.adjust(p, method = "BH"))

kruskal_test_alpha_rare_2

write.csv(kruskal_test_alpha_rare_2, "Results/kruskal_test_alpha_rare_2.csv", row.names = FALSE)

dunn_test_alpha_rare_2 <- metadata_samples_long_rare_2 %>% group_by(alpha_div) %>%
  rstatix::dunn_test(value ~ group, p.adjust.method = "BH") %>%
  add_xy_position(x = "group", fun = "mean_sd", step.increase = 0.05, scales = "free")
dunn_test_alpha_rare_2 %>% kbl() %>%
  kableExtra::kable_paper(font_size = 12)

dunn_test_alpha_rare_table_2 <- apply(dunn_test_alpha_rare_2,2,as.character)
write.csv(dunn_test_alpha_rare_table_2, "Results/dunn_test_alpha_rare_table_2.csv", row.names = FALSE)

dunn_test_alpha_rare_plot_2 <- ggplot(metadata_samples_long_rare_2, aes(x = group, y = value)) +
  geom_boxplot() +
  geom_jitter(shape = 21, color = "black", size = 2, aes(fill = group)) +
  facet_wrap(~alpha_div, scales = "free", ncol = 3) +
  scale_fill_nejm() +
  stat_pvalue_manual(data = dunn_test_alpha_rare_2, 
                     label = "p.adj.signif", tip.length = 0.01, bracket.shorten = 0.05,
                     hide.ns = TRUE, size = 4) +
  scale_x_discrete(guide = guide_axis(n.dodge = 2)) +
  theme_bw() +
  theme(axis.text.x = element_text(size = 10), legend.position = "none")

ggsave("Results/dunn_test_alpha_rare_plot_2.png", dunn_test_alpha_rare_plot_2, width = 13, height = 8, dpi = 300)

dunn_test_alpha_rare_plot_2

```


```{r}

metadata_samples_alpha_2_rare

kruskal_chao1_2_rare <- kruskal.test(Chao1 ~ group, data = metadata_samples_alpha_2_rare)
kruskal_shannon_2_rare <- kruskal.test(Shannon ~ group, data = metadata_samples_alpha_2_rare)
kruskal_invsimps_rare <- kruskal.test(InvSimpson ~ group, data = metadata_samples_alpha_2_rare)
kruskal_obsv_2_rare <- kruskal.test(Observed ~ group, data = metadata_samples_alpha_2_rare)
kruskal_pd_2_rare <- kruskal.test(PD ~ group, data = metadata_samples_alpha_2_rare)


# Extract values
H_chao1_2_rare <- kruskal_chao1_2_rare$statistic
p_chao1_2_rare <- kruskal_chao1_2_rare$p.value

H_shannon_2_rare <- kruskal_shannon_2_rare$statistic
p_shannon_2_rare <- kruskal_shannon_2_rare$p.value

H_invsimps_2_rare <- kruskal_invsimps_rare$statistic
p_invsimps_2_rare <- kruskal_invsimps_rare$p.value

H_obsv_2_rare <- kruskal_obsv_2_rare$statistic
p_obsv_2_rare <- kruskal_obsv_2_rare$p.value

H_pd_2_rare <- kruskal_pd_2_rare$statistic
p_pd_2_rare <- kruskal_pd_2_rare$p.value

# Print results
cat("Kruskal-Wallis test for Chao1 index: H =", round(H_chao1_2_rare, 3), "p =", round(p_chao1_2_rare, 3), "\n")
cat("Kruskal-Wallis test for Shannon index: H =", round(H_shannon_2_rare, 3), "p =", round(p_shannon_2_rare, 3), "\n")
cat("Kruskal-Wallis test for Inverse Simpson index: H =", round(H_invsimps_2_rare, 3), "p =", round(p_invsimps_2_rare, 3), "\n")
cat("Kruskal-Wallis test for Observed Number of Species: H =", round(H_obsv_2_rare, 3), "p =", round(p_obsv_2_rare, 3), "\n")
cat("Kruskal-Wallis test for Faith's Phylogenetic Diversity: H =", round(H_pd_2_rare, 3), "p =", round(p_pd_2_rare, 3), "\n")


# Filter for Chao1 and Shannon indices
dunn_chao1_2_rare <- dunn_test_alpha_rare_2 %>% filter(alpha_div == "Chao1")
dunn_shannon_2_rare <- dunn_test_alpha_rare_2 %>% filter(alpha_div == "Shannon")
dunn_invsimps_2_rare <- dunn_test_alpha_rare_2 %>% filter(alpha_div == "InvSimpson")
dunn_obsv_2_rare <- dunn_test_alpha_rare_2 %>% filter(alpha_div == "Observed")
dunn_pd_2_rare <- dunn_test_alpha_rare_2 %>% filter(alpha_div == "PD")

# Show significant results (p < 0.05)
significant_chao1_2_rare <- dunn_chao1_2_rare %>% filter(p.adj < 0.05)
significant_shannon_2_rare <- dunn_shannon_2_rare %>% filter(p.adj < 0.05)
significant_invsimps_2_rare <- dunn_invsimps_2_rare %>% filter(p.adj < 0.05)
significant_obsv_2_rare <- dunn_obsv_2_rare %>% filter(p.adj < 0.05)
significant_pd_2_rare <- dunn_pd_2_rare %>% filter(p.adj < 0.05)

# Print significant comparisons
print(significant_chao1_2_rare)
print(significant_shannon_2_rare)
print(significant_invsimps_2_rare)
print(significant_obsv_2_rare)
print(significant_pd_2_rare)

alpha_rare_summary_2 <- metadata_samples_alpha_2_rare %>%
  group_by(group) %>%
  summarise(
    Chao1_mean = mean(Chao1, na.rm = TRUE),
    Chao1_sd = sd(Chao1, na.rm = TRUE),
    Shannon_mean = mean(Shannon, na.rm = TRUE),
    Shannon_sd = sd(Shannon, na.rm = TRUE),
    InvSimps_mean = mean(InvSimpson, na.rm = TRUE),
    InvSimps_sd = sd(InvSimpson, na.rm = TRUE), 
    Obsv_mean = mean(Observed, na.rm = TRUE),
    Obsv_sd = sd(Observed, na.rm = TRUE),
    PD_mean = mean(PD, na.rm = TRUE),
    PD_sd = sd(PD, na.rm = TRUE)
  )

print(alpha_rare_summary_2)
write.csv(alpha_rare_summary_2, "Results/alpha_rare_summary_table_2.csv", row.names = FALSE)

```


```{r}
PCoA_ASV_bray_2 <- ordinate(ps_filt_fixed_rel_2, distance = "bray", method = "PCoA")
beta_ASV_bray_2 <- phyloseq::distance(ps_filt_fixed_rel_2, "bray")
ado_bray_group_2 <- adonis2(beta_ASV_bray_2 ~ group, meta_951422)
print(ado_bray_group_2)

group.bd_2 <- betadisper(d = beta_ASV_bray_2, group = meta_951422$group, type="centroid")
bd_group_boxplot_2 <- boxplot(group.bd_2)

bd_group_boxplot_2

png("Results/beta_dispersion_bc_boxplot_2.png", width = 3500, height = 2000, res = 300)  # High resolution
boxplot(group.bd_2, main = "Beta Dispersion per Group (BC) BP2", cex.axis = 1.2, cex.lab = 1.5)
dev.off()

permutest(group.bd_2)


BC_pcoa_plot_2 <- plot_ordination(ps_filt_fixed_rel_2, PCoA_ASV_bray_2, axes = c(1,2))+
  stat_conf_ellipse(geom = "polygon", alpha = 0.1, aes(color = group,  fill = group))+
  geom_point(size = 3, shape = 21, aes(fill = group))+
  stat_mean(aes(fill = group), shape = 23, size = 4) + 
  annotate("text", x =- 0.3, y = 0.5, color = "black", label = paste0("p=", round(ado_bray_group_2$`Pr(>F)`[1], digits = 3)))+
  annotate("text", x = -0.3, y = 0.45, color = "black", label = paste0("R\u00b2=", round((ado_bray_group_2$R2[1])*100, digits = 2), "%"))+
  scale_fill_nejm(name = "Group")+
  scale_color_nejm(name = "Group")+
  theme_bw()+
  theme(legend.position = "right")

ggsave("Results/BC_pcoa_plot_2.png", BC_pcoa_plot_2, width = 10, height = 8, dpi = 300)

BC_pcoa_plot_2

ps_filt_fixed_rel@sam_data$host_sex <- as.factor(ps_filt_fixed_rel@sam_data$host_sex)

ado_bray_sex_2 <- adonis2(beta_ASV_bray_2 ~ host_sex, meta_951422)
print(ado_bray_sex_2)

BC_pcoa_sex_plot_2 <- plot_ordination(ps_filt_fixed_rel_2, PCoA_ASV_bray_2, axes = c(1,2))+
  stat_conf_ellipse(geom = "polygon", alpha = 0.1, aes(color = host_sex,  fill = host_sex))+
  geom_point(size = 3, shape = 21, aes(fill = host_sex))+
  stat_mean(aes(fill = host_sex), shape = 23, size = 4) + 
  annotate("text", x =- 0.3, y = 0.5, color = "black", label = paste0("p=", round(ado_bray_sex_2$`Pr(>F)`[1], digits = 3)))+
  annotate("text", x = -0.3, y = 0.45, color = "black", label = paste0("R\u00b2=", round((ado_bray_sex_2$R2[1])*100, digits = 2), "%"))+
  scale_fill_nejm(name = "Host sex")+
  scale_color_nejm(name = "Host Sex")+
  theme_bw()+
  theme(legend.position = "right")

ggsave("Results/BC_pcoa_sex_plot_2.png", BC_pcoa_sex_plot_2, width = 10, height = 8, dpi = 300)

BC_pcoa_sex_plot_2

bd_df_2 <- data.frame(samples = rownames(group.bd_2$vectors),
                      distances = group.bd_2$distances,
                      group = meta_951422$group)

kruskal_bc_test_2 <- bd_df_2 %>% rstatix::kruskal_test(distances ~ group)
print(kruskal_bc_test_2)

# Dunn’s test for pairwise comparisons with Benjamini-Hochberg correction
dunn_bc_test_2 <- bd_df_2 %>% rstatix::dunn_test(distances ~ group, p.adjust.method = "BH")
print(dunn_bc_test_2)

dunn_bc_test_table_2 <- apply(dunn_bc_test_2,2,as.character)
write.csv(dunn_bc_test_table_2, "Results/dunn_bc_test_table_2.csv", row.names = FALSE)

# Extract p-values for compact letter display
dunn_bc_pvalues_2 <- dunn_bc_test_2 %>% select(group1, group2, p.adj) %>%
  pivot_wider(names_from = group2, values_from = p.adj) %>%
  column_to_rownames(var = "group1")

# Convert to a matrix format for multcompLetters
dunn_bc_matrix_2 <- as.matrix(dunn_bc_pvalues_2)
# Fill the lower triangle with upper triangle values to make the matrix symmetric
dunn_bc_matrix_2[lower.tri(dunn_bc_matrix_2)] <- t(dunn_bc_matrix_2)[lower.tri(dunn_bc_matrix_2)]


# Generate compact letter display for groups
dunn_bc_labels_2 <- multcompLetters(dunn_bc_matrix_2)$Letters
print(dunn_bc_labels_2)

```

```{r}
## Set root for WU so it runs with the same root as previous wu analysis

# ASV3504

# Define the root sequence (use the exact sequence you mentioned before)
root_seq <- "ASV3504"

# Extract tree from phyloseq object
tree_2 <- phy_tree(ps_filt_fixed_rel_2)

# Set the root
tree_rooted_2 <- root(tree_2, outgroup = root_seq, resolve.root = TRUE)

# Replace the tree in the phyloseq object
phy_tree(ps_filt_fixed_rel_2) <- tree_rooted_2
# ggtree(ps_filt_fixed_rel@phy_tree, options(ignore.negative.edge=TRUE))


PCoA_ASV_wunifrac_2 <- ordinate(ps_filt_fixed_rel_2, distance = "wunifrac", method = "PCoA")
beta_ASV_wunifrac_2 <- phyloseq::distance(ps_filt_fixed_rel_2, "wunifrac")
ado_wunifrac_group_2 <- adonis2(beta_ASV_wunifrac_2 ~ group, meta_951422)
print(ado_wunifrac_group_2)


group_wunifrac.bd_2 <- betadisper(d = beta_ASV_wunifrac_2, group = meta_951422$group, type="centroid")
boxplot(group_wunifrac.bd_2)

png("Results/beta_dispersion_wu_boxplot_2.png", width = 3500, height = 2000, res = 300)  # High resolution
boxplot(group_wunifrac.bd_2, main = "Beta Dispersion per Group (WU) BP2", cex.axis = 1.2, cex.lab = 1.5)
dev.off()

permutest(group_wunifrac.bd_2)


WU_pcoa_plot_2 <- plot_ordination(ps_filt_fixed_rel_2, PCoA_ASV_wunifrac_2, axes = c(1,2))+
  stat_conf_ellipse(geom = "polygon", alpha = 0.1, aes(color = group,  fill = group))+
  geom_point(size = 3, shape = 21, aes(fill = group))+
  stat_mean(aes(fill = group), shape = 23, size = 4) + 
  annotate("text", x = 0.1, y = -0.15, color = "black", label = paste0("p=", round(ado_wunifrac_group_2$`Pr(>F)`[1], digits = 3)))+
  annotate("text", x = 0.1, y = -0.13, color = "black", label = paste0("R\u00b2=", round((ado_wunifrac_group_2$R2[1])*100, digits = 2), "%"))+
  scale_fill_nejm(name = "Group")+
  scale_color_nejm(name = "Group")+
  theme_bw()+
  theme(legend.position = "right")

ggsave("Results/WU_pcoa_plot_2.png", WU_pcoa_plot_2, width = 10, height = 8, dpi = 300)


WU_pcoa_plot_2 

ado_wunifrac_sex_2 <- adonis2(beta_ASV_wunifrac_2 ~ host_sex, meta_951422)
print(ado_wunifrac_sex_2)

WU_pcoa_sex_plot_2 <- plot_ordination(ps_filt_fixed_rel_2, PCoA_ASV_wunifrac_2, axes = c(1,2))+
  stat_conf_ellipse(geom = "polygon", alpha = 0.1, aes(color = host_sex,  fill = host_sex))+
  geom_point(size = 3, shape = 21, aes(fill = host_sex))+
  stat_mean(aes(fill = host_sex), shape = 23, size = 4) + 
  annotate("text", x = 0.1, y = -0.15, color = "black", label = paste0("p=", round(ado_wunifrac_sex_2$`Pr(>F)`[1], digits = 3)))+
  annotate("text", x = 0.1, y = -0.13, color = "black", label = paste0("R\u00b2=", round((ado_wunifrac_sex_2$R2[1])*100, digits = 2), "%"))+
  scale_fill_nejm(name = "Host sex")+
  scale_color_nejm(name = "Host sex")+
  theme_bw()+
  theme(legend.position = "right")

ggsave("Results/WU_pcoa_sex_plot_2.png", WU_pcoa_sex_plot_2, width = 10, height = 8, dpi = 300)

WU_pcoa_sex_plot_2

wu_bd_df_2 <- data.frame(samples = rownames(group_wunifrac.bd_2$vectors),
                         distances = group_wunifrac.bd_2$distances,
                         group = meta_951422$group)

kruskal_wu_test_2 <- wu_bd_df %>% rstatix::kruskal_test(distances ~ group)
print(kruskal_wu_test_2)

# Dunn’s test for pairwise comparisons with Benjamini-Hochberg correction
dunn_wu_test_2 <- wu_bd_df_2 %>% rstatix::dunn_test(distances ~ group, p.adjust.method = "BH", detailed = TRUE)
print(dunn_wu_test_2)

dunn_wu_test_table_2 <- apply(dunn_wu_test_2,2,as.character)
write.csv(dunn_wu_test_table_2, "Results/dunn_wu_test_table_2.csv", row.names = FALSE)

# Extract p-values for compact letter display
dunn_wu_pvalues_2 <- dunn_wu_test_2 %>% select(group1, group2, p.adj) %>%
  pivot_wider(names_from = group2, values_from = p.adj) %>%
  column_to_rownames(var = "group1")

# Convert to a matrix format for multcompLetters
dunn_wu_matrix_2 <- as.matrix(dunn_wu_pvalues_2)
# Fill the lower triangle with upper triangle values to make the matrix symmetric
dunn_wu_matrix_2[lower.tri(dunn_wu_matrix_2)] <- t(dunn_wu_matrix_2)[lower.tri(dunn_wu_matrix_2)]


# Generate compact letter display for groups
dunn_wu_labels_2 <- multcompLetters(dunn_wu_matrix_2)$Letters
print(dunn_wu_labels_2)

```
## Not necessary to do it for baseline samples, they are all "remission"

```{r}
## For end samples
ps_ff_rel_end_2 <- ps_filt_fixed_rel_2 %>% ps_filter(!is.na(disease_status_end))

meta_beta_end_2 <- as.data.frame(as.matrix(sample_data(ps_ff_rel_end_2)))

PCoA_ASV_bray_end_2 <- ordinate(ps_ff_rel_end_2, distance = "bray", method = "PCoA")
PCoA_ASV_bray_end_2


beta_ASV_bray_end_2 <- phyloseq::distance(ps_ff_rel_end_2, "bray")
ado_bray_group_end_2 <- adonis2(beta_ASV_bray_end_2 ~ disease_status_end, meta_beta_end_2)
print(ado_bray_group_end_2)

end.bd_2 <- betadisper(d = beta_ASV_bray_end_2, group = meta_beta_end_2$disease_status_end, type="centroid")
bd_end_boxplot_2 <- boxplot(end.bd_2)

bd_end_boxplot_2

png("Results/betadisp_end_bc_boxplot_2.png", width = 3500, height = 2000, res = 300)  # High resolution
boxplot(end.bd_2, main = "Beta Dispersion per End Status (BC) BP1", cex.axis = 1.2, cex.lab = 1.5)
dev.off()

permutest(end.bd_2)

BC_end_pcoa_plot_2 <- plot_ordination(ps_ff_rel_end_2, PCoA_ASV_bray_end_2, axes = c(1,2))+
  stat_conf_ellipse(geom = "polygon", alpha = 0.1, aes(color = disease_status_end,  fill = disease_status_end))+
  geom_point(size = 3, shape = 21, aes(fill = disease_status_end))+
  stat_mean(aes(fill = disease_status_end), shape = 23, size = 4) + 
  annotate("text", x =- 0.3, y = 0.5, color = "black", label = paste0("p=", round(ado_bray_group_end_2$`Pr(>F)`[1], digits = 3)))+
  annotate("text", x = -0.3, y = 0.45, color = "black", label = paste0("R\u00b2=", round((ado_bray_group_end_2$R2[1])*100, digits = 2), "%"))+
  scale_fill_nejm(name = "End Status")+
  scale_color_nejm(name = "End Status")+
  theme_bw()+
  theme(legend.position = "right")

ggsave("Results/BC_end_pcoa_plot_2.png", BC_end_pcoa_plot_2, width = 10, height = 8, dpi = 300)

BC_end_pcoa_plot_2

bd_end_df_2 <- data.frame(samples = rownames(end.bd_2$vectors),
                        distances = end.bd_2$distances,
                        disease_status_end = meta_beta_end_2$disease_status_end)

kruskal_bc_test_end_2 <- bd_end_df_2 %>% rstatix::kruskal_test(distances ~ disease_status_end)
print(kruskal_bc_test_end_2)

# Dunn’s test for pairwise comparisons with Benjamini-Hochberg correction
dunn_bc_test_end_2 <- bd_end_df_2 %>% rstatix::dunn_test(distances ~ disease_status_end, p.adjust.method = "BH")
print(dunn_bc_test_end_2)

dunn_bc_test_end_table_2 <- apply(dunn_bc_test_end_2,2,as.character)
write.csv(dunn_bc_test_end_table_2, "Results/dunn_bc_test_end_table_2.csv", row.names = FALSE)

# Extract p-values for compact letter display
dunn_bc_pvalues_end_2 <- dunn_bc_test_end_2 %>% select(group1, group2, p.adj) %>%
  pivot_wider(names_from = group2, values_from = p.adj) %>%
  column_to_rownames(var = "group1")

# Convert to a matrix format for multcompLetters
dunn_bc_matrix_end_2 <- as.matrix(dunn_bc_pvalues_end_2)
# Fill the lower triangle with upper triangle values to make the matrix symmetric
dunn_bc_matrix_end_2[lower.tri(dunn_bc_matrix_end_2)] <- t(dunn_bc_matrix_end_2)[lower.tri(dunn_bc_matrix_end_2)]


# Generate compact letter display for groups
dunn_bc_labels_end_2 <- multcompLetters(dunn_bc_matrix_end_2)$Letters
print(dunn_bc_labels_end_2)

```


## Check prevalence of taxa
```{r}

print(ps_filt_fixed_2)

ps_filt_fixed_prev_2 <- tax_filter(ps_filt_fixed_2,
  min_prevalence = 0.01)
  
print(ps_filt_fixed_prev_2)


prev_thresholds_2 <- c(0.05, 0.01, 0.001)

prev_filt_results_2 <- data.frame(min_prevalence = prev_thresholds_2, num_taxa_retained = NA, perc_taxa_removed = NA)

total_taxa_2 <- nrow(tax_table(ps_filt_fixed_2))

for (i in seq_along(prev_thresholds_2)) {
  ps_filt_fixed_samples_prev_all_2 <- ps_filt_fixed_2 %>%
    tax_filter(min_prevalence = prev_thresholds_2[i])
  
  num_taxa_retained <- nrow(tax_table(ps_filt_fixed_samples_prev_all_2))
  perc_taxa_removed <- 100 * (total_taxa_2 - num_taxa_retained) / total_taxa_2
  
  prev_filt_results_2$num_taxa_retained[i] <- num_taxa_retained
  prev_filt_results_2$perc_taxa_removed[i] <- perc_taxa_removed
}

print(prev_filt_results_2)

```

```{r}

ps_filt_fixed_genus_rel_2 <- tax_transform(ps_filt_fixed_genus_2, trans = "compositional")
input_sample_metadata_2 <- ps_filt_fixed_genus_rel_2 %>% meta
input_data_genus_Maaslin_2 <- as.data.frame(otu_table(ps_filt_fixed_genus_rel_2))

input_data_genus_Maaslin_2 <- input_data_genus_Maaslin_2[sapply(input_data_genus_Maaslin_2, is.numeric)]

output_directory_genus_2 <- "Maaslin2_output_genus_2"

fit_data_genus_2 <- Maaslin2::Maaslin2(
    input_data = input_data_genus_Maaslin_2,
    input_metadata = input_sample_metadata_2,
    output = output_directory_genus_2,
    analysis_method = "LM",
    fixed_effects = "group",
    normalization = "NONE", 
    correction = "BH",
    transform = "LOG",
    min_abundance = 0,
    min_prevalence = 0.01,
    reference = "group,remission-remission")

```

```{r}

head(fit_data_genus_2$results, n=20) %>% kbl() %>%
  kableExtra::kable_paper(font_size = 12) %>%
  scroll_box(width = "1000px", height = "500px")

```


```{r}
taxa_abund_kw_2 <- as.data.frame(otu_table(ps_filt_fixed_genus_2))

# Convert row names to a column (sample IDs)
taxa_abund_kw_2 <- taxa_abund_kw_2 %>% 
  tibble::rownames_to_column(var = "samples")

# Convert sample_data from phyloseq object to a proper data frame
metadata_samples_kw_2 <- as(sample_data(ps_filt_fixed_genus_2), "data.frame") %>% 
  tibble::rownames_to_column(var = "samples") # Ensure sample IDs are a column

metadata_samples_kw_2 <- metadata_samples_kw_2 %>% 
  select(c(samples, group))


# Merge metadata with abundance data
taxa_data_kw_2 <- left_join(metadata_samples_kw_2, taxa_abund_kw_2, by = "samples") 

taxa_data_kw_tib_2 <- taxa_data_kw_2 %>%
  as_tibble()  # Convert to tibble for tidyverse functions

# Function to perform Kruskal-Wallis test on each taxon
kruskal_test_results_2 <- taxa_data_kw_tib_2 %>%
  pivot_longer(cols = -c(samples, group), names_to = "taxon", values_to = "abundance") %>%
  group_by(taxon) %>%
  rstatix::kruskal_test(abundance ~ group) %>%
  rstatix::adjust_pvalue(method = "BH") %>%  # Adjust for multiple testing (Benjamini-Hochberg)
  arrange(p.adj)

# View significant results (p.adj < 0.05)
significant_taxa_kw_2 <- kruskal_test_results_2 %>% filter(p.adj < 0.05)

# Display top results
significant_taxa_kw_2 %>% 
  arrange(p.adj) %>% 
  head(n=20)

top_10_sig_taxa_2 <- significant_taxa_kw_2 %>% arrange(p.adj) %>% head(n=10)

top_10_sig_taxa_2

```

```{r}
top10_sig_taxa_names_2 <- top_10_sig_taxa_2$taxon

ps_ff_genus_melt_2 <- ps_filt_fixed_genus_rel_2 %>% psmelt()

ps_traj_taxa_filt_2 <- ps_ff_genus_melt_2 %>% filter(OTU %in% top10_sig_taxa_names_2)

# Function to generate a plot for a given host_subject_id
plot_subject_trajectory <- function(df, group_name) {
  
  df_filt <- df %>% filter(group == group_name)
  
  ggplot(df_filt, aes(x = week, y = Abundance, group = OTU, color = OTU)) +
    geom_line(aes(linetype = OTU), size = 1) + # Line style per taxa
    geom_point(size = 3) + 
    scale_color_manual(values = distinct_palette(10, pal = "brewerPlus"))
  
  
}

plots_remrem_2 <- plot_subject_trajectory(ps_traj_taxa_filt_2, "remission-remission")

plots_remac_2 <- plot_subject_trajectory(ps_traj_taxa_filt_2, "remission-active")

wr_remrem_lineplots_2 <- wrap_plots(plots_remrem_2) + plot_layout(guides = "collect") &
  facet_wrap("host_subject_id", scales = "free_x") &
  theme(axis.text.x = element_text(angle = 60, hjust = 1, size = 10),
        axis.ticks.y = element_blank(), legend.title = element_blank(),
        legend.position = "bottom", strip.text = element_text(size = 12),
        legend.key.size = unit(0.4, "cm"),
        legend.text = element_text(size = 14))
wr_remac_lineplots_2 <- wrap_plots(plots_remac_2) + plot_layout(guides = "collect") &
  facet_wrap("host_subject_id", scales = "free_x") &
  theme(axis.text.x = element_text(angle = 60, hjust = 1, size = 10),
        axis.ticks.y = element_blank(), legend.title = element_blank(),
        legend.position = "bottom", strip.text = element_text(size = 12),
        legend.key.size = unit(0.4, "cm"),
        legend.text = element_text(size = 14))

ggsave(filename = "wr_remrem_lineplots_2.png", plot = wr_remrem_lineplots_2, 
       width = 14, height = 11, dpi = 320)

ggsave(filename = "wr_remac_lineplots_2.png", plot = wr_remac_lineplots_2, 
       width = 14, height = 11, dpi = 320)


wr_remrem_lineplots_2
wr_remac_lineplots_2

```
